<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.555">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>ALARMcoding Tutorial - Step 1: SeekDeep</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./2_MACSE.html" rel="next">
<link href="./Prepare_Sequencing.html" rel="prev">
<link href="./Images/ALARMcoding_Icon.png" rel="icon" type="image/png">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">ALARMcoding Tutorial</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./Index.qmd"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item compact">
    <a class="nav-link" href="./About.qmd"> 
<span class="menu-text"></span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools tools-wide">
    <a href="https://twitter.com/DionneArgy" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-twitter"></i></a>
    <div class="dropdown">
      <a href="" title="" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" aria-label=""><i class="bi bi-github"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item quarto-navbar-tools-item" href="https://github.com/dionnecargy">
            Source Code
            </a>
          </li>
          <li>
            <a class="dropdown-item quarto-navbar-tools-item" href="https://github.com/dionnecargy">
            Report a Bug
            </a>
          </li>
      </ul>
    </div>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./1_SeekDeep.html">Drug Resistance Pipeline</a></li><li class="breadcrumb-item"><a href="./1_SeekDeep.html">Step 1: SeekDeep</a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./Images/ALARMcoding_Icon.png" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Welcome</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">About ALARMcoding</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./About_Resistance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Antimalarial Drug Resistance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./About_Markers.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Marker Choice</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true">
 <span class="menu-text">PCR Protocol</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./PCR_Important.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PCR: Important Notes</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Multiplex_1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Multiplex 1: K13 and mdr1 (86, 184)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Multiplex_2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Multiplex 2: crt, mdr1 (1034, 1042, 1246)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Multiplex_3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Multiplex 3: dhfr, dhps</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Multiplex_4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Multiplex 4: aat1, pfs47</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./Prepare_Sequencing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Post-PCR: Prepare for Sequencing</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true">
 <span class="menu-text">Drug Resistance Pipeline</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./1_SeekDeep.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Step 1: SeekDeep</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./2_MACSE.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Step 2: MACSE</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./3_Custom.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Step 3: Processing in R</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#setuptarampanalysis" id="toc-setuptarampanalysis" class="nav-link active" data-scroll-target="#setuptarampanalysis">setupTarAmpAnalysis</a></li>
  <li><a href="#extra-commands" id="toc-extra-commands" class="nav-link" data-scroll-target="#extra-commands">Extra Commands</a></li>
  <li><a href="#output-interpretation" id="toc-output-interpretation" class="nav-link" data-scroll-target="#output-interpretation">Output Interpretation</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./1_SeekDeep.html">Drug Resistance Pipeline</a></li><li class="breadcrumb-item"><a href="./1_SeekDeep.html">Step 1: SeekDeep</a></li></ol></nav>
<div class="quarto-title">
<h1 class="title">Step 1: SeekDeep</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>Data processing of data using this SeekDeep made by <a href="https://github.com/bailey-lab/SeekDeep">Nick Nathaway</a>. Here is a link to a step-by-step guide for <a href="https://seekdeep.brown.edu/installingSeekDeep_Ubuntu.html">installation</a>.</p>
<p>We used Illumina sequencing data and would like to run all of the three pipelines. We therefore use the global function <a href="https://seekdeep.brown.edu/setupTarAmpAnalysis_usage.html">setupTarAmpAnalysis</a>. The data files were made in accordance to the <a href="https://seekdeep.brown.edu/tutorial_PairedEnd_withMIDs.html">Initial Set-Up Guidelines for Paired End Reads with MIDs</a>.</p>
<p><strong>STEP 1: Must run this before SeekDeep can work:</strong></p>
<p><code>export PATH=/home/software/SeekDeep-3.0.1/bin/:$PATH</code></p>
<p><strong>STEP 2: Run the SeekDeep command</strong></p>
<p>SeekDeep setupTarAmpAnalysis –samples sampleNamesP1.tab.txt –outDir output –inputDir . –idFile multipleGenePairs.id.txt –overlapStatusFnp overlapStatuses.txt –refSeqsDir markerRefSeqs/forSeekDeep/refSeqs/ –extraExtractorCmds=“–barcodeErrors 0 –midWithinStart 8 –primerWithinStart 8 –checkRevComplementForMids true –checkRevComplementForPrimers true” –extraProcessClusterCmds=“–fracCutOff 0.078 –sampleMinTotalReadCutOff 100”</p>
<p>This command was developed after testing for a wide number of parameters outlined here <a href="https://seekdeep.brown.edu/">SeekDeep</a> <span class="citation" data-cites="Hathaway2017">(<a href="#ref-Hathaway2017" role="doc-biblioref">Hathaway et al. 2017</a>)</span>. We used the controls in our pools to parameterise.</p>
<section id="setuptarampanalysis" class="level2">
<h2 class="anchored" data-anchor-id="setuptarampanalysis">setupTarAmpAnalysis</h2>
<p><strong>Sequencing Files</strong>: <code>--inputDir</code> Input Directory of raw data files; required. These files include:</p>
<ul>
<li><code>--fastq1</code> Input sequence filename, only need 1, fastq first mate text file; required</li>
<li><code>--fastq1gz</code> Input sequence filename, only need 1, fastq first mate gzipped file; required</li>
<li><code>--fastq2</code> Input sequence filename, only needed with –fastq1, fastq second mate text file; required</li>
<li><code>--fastq2gz</code> Input sequence filename, only needed with –fastq1gz, fastq second mate gzipped file; required</li>
</ul>
<p><strong>ID Files</strong>:</p>
<ul>
<li><code>--idFile</code> = ID file containing primer and possible additional MIDs; required.</li>
<li><code>--samples</code> = A file containing the samples names, columns should go target, sample, pcr1, pcr2 (optional) etc; required.</li>
</ul>
<p><strong>Illumina Stitching</strong>: <code>--overlapStatusFnp</code> A file with two columns, target,status; status column should contain 1 of 3 values (capitalization doesn’t matter): <code>r1BegOverR2End</code>,<code>r1EndOverR2Beg</code>,<code>NoOverlap</code>.</p>
<ul>
<li><code>r1BegOverR2End</code> = target size &lt; read length (causes read through),</li>
<li><code>r1EndOverR2Beg</code> = target size &gt; read length less than 2 x read length,</li>
<li><code>NoOverlap</code> = target size &gt; 2 x read length; required.</li>
</ul>
<p><strong>Output</strong>: <code>--outDir</code> Directory to setup for analysis; required.</p>
</section>
<section id="extra-commands" class="level2">
<h2 class="anchored" data-anchor-id="extra-commands">Extra Commands</h2>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true" aria-current="page">Add Reference</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false">extractor</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-3" role="tab" aria-controls="tabset-1-3" aria-selected="false">qluster</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-4" role="tab" aria-controls="tabset-1-4" aria-selected="false">processCluster</a></li></ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<p><strong>Step 1: Set up the reference genomes:</strong></p>
<p><code>export PATH=/home/software/samtools-1.15.1/bin:$PATH</code></p>
<p><code>export PATH=/home/software/bowtie2-2.4.5-linux-x86_64:$PATH</code></p>
<p>SeekDeep genTargetInfoFromGenomes –gffDir pfgenomes/gff/ –genomeDir pfgenomes/genomes/ –primers GeneID.tab.txt –pairedEndLength 250 –dout markerRefSeqs –numThreads 19 –overWriteDir true –verbose</p>
<p><strong>Step 2: Run the command</strong></p>
<p>SeekDeep setupTarAmpAnalysis –samples sampleNamesP8.tab.txt –outDir Run-Pool-8-WRefs –inputDir . –idFile multipleGenePairs.id.txt –overlapStatusFnp overlapStatuses.txt –refSeqsDir markerRefSeqs/markerFasta/ –extraExtractorCmds=“–midWithinStart 100 –primerWithinStart 100 –qualCheckCutOff .75 –qualCheckLevel 30 –checkRevComplementForMids true –checkRevComplementForPrimers true”</p>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<p>All of these commands and their uses are discussed in the <a href="https://seekdeep.brown.edu/extractor_usage.html">SeekDeep extractor Tutorial</a>. The aim of this pipeline is to extract sequences from various sequences input types (fastq,fasta,sff,etc.) with primers and barcodes plus some filtering.</p>
<p><strong>Barcodes</strong>:</p>
<p><code>--barcodeErrors 0</code> = To further reduce any ambiguity and specifically select for NO errors.</p>
<p><code>--midWithinStart 8</code> = By default the primer or barcodes are searched at the very beginning of seq, use this flag to extended the search, should be kept low to cut down on false positives; default=0; <code>--primerWithinStart 8</code> By default the primer or barcodes are searched at the very beginning of seq, use this flag to extended the search, should be kept low to cut down on false positives; default=0;</p>
<p>Roche MID 47, TGTGAGTAGT, is <strong>within</strong> the dhps-1 sequence and the output therefore being preferentially labelled as “MID47”. Because of this, we discovered that our midWithinStart and primerWithinStart commands could be adjusted. Reducing the parameter to 0 produced <strong>no</strong> data. This is because there are 7bp at the start of each sequence preceding the MID. By changing the midWithinStart and primerWithinStart parameters to a smaller value, we can use MID 47 for dhps-2.</p>
<p>Our optimal <strong>golidlocks</strong> value was –midWithinStart 8 –primerWithinStart 8.</p>
<p><code>--checkRevComplementForMids true</code> = Check the Reverse Complement of the Seqs As Well For MIDs <code>--checkRevComplementForPrimers true</code> = Check the Reverse Complement of the Seqs As Well For Primers</p>
</div>
<div id="tabset-1-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-3-tab">
<p>These commands and uses are discussed in the <a href="https://seekdeep.brown.edu/qluster_usage.html">SeekDeep qluster Tutorial</a>. The qluster pipeline aims to cluster input sequences by collapsing on set differences between the sequences. We did not modify anything from the default parameters.</p>
</div>
<div id="tabset-1-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-4-tab">
<p>The <a href="https://seekdeep.brown.edu/processClusters_usage.html">SeekDeep processClusters Tutorial</a> provides help for these commands and their uses. Here, we adjusted some <strong>filtering</strong> parameters to remove low-frequency variants based on preliminary analysis of mock mixed infections of controls.</p>
<p><code>--fracCutOff 0.078</code> = Final cluster Fraction Cut off; default=0.005;</p>
<p>Instead of specifying a hard read count cut-off e.g., <code>--clusterCutOff</code>, I opted to try and remove haplotypes that were represented at a frequency in the sample using the function <code>—-fracCutOff</code>:</p>
<ul>
<li>5% <span class="citation" data-cites="Boyce2019 Wamae2022a Wamae2022b">(<a href="#ref-Boyce2019" role="doc-biblioref">Boyce et al. 2019</a>; <a href="#ref-Wamae2022a" role="doc-biblioref">Wamae, Ndwiga, et al. 2022</a>; <a href="#ref-Wamae2022b" role="doc-biblioref">Wamae, Kimenyi, et al. 2022</a>)</span></li>
<li>3.5% <span class="citation" data-cites="Hemming2020">(<a href="#ref-Hemming2020" role="doc-biblioref">Hemming-Schroeder et al. 2020</a>)</span></li>
<li>1% <span class="citation" data-cites="Brazeau2019 Pringle2019 Osoti2022">(<a href="#ref-Brazeau2019" role="doc-biblioref">Brazeau et al. 2019</a>; <a href="#ref-Pringle2019" role="doc-biblioref">Pringle et al. 2019</a>; <a href="#ref-Osoti2022" role="doc-biblioref">Osoti et al. 2022</a>)</span></li>
<li>0.5% <span class="citation" data-cites="Aydemir2018 Bal2022 Topazian2022">(<a href="#ref-Aydemir2018" role="doc-biblioref">Aydemir et al. 2018</a>; <a href="#ref-Bal2022" role="doc-biblioref">Balasubramanian et al. 2022</a>; <a href="#ref-Topazian2022" role="doc-biblioref">Topazian et al. 2022</a>)</span></li>
</ul>
<p>For our data, a cut-off of 0.078 or 7.8% was used.</p>
<p><code>--sampleMinTotalReadCutOff 100</code> = Sample Minimum Total Read Cut Off, if the total read count for the sample is below this it will be thrown out; default=250;</p>
<p>This new parameter was added in the SeekDeep v. 3.0.1 that was used for the final analysis of the drug resistance dataset in 2023/2024. We realised that there was a new default parameter that removed isolates that had less than 250 total reads (i.e., 250X coverage) in the selected clusters output file.</p>
<p>We then amended this, based on a visualisation of the total read density per marker - this showed that for some markers like dhps-1/2, dhfr and mdr1-2 there were more isolates that had ~100X-250X coverage. We therefore created a threshold of 100X coverage.</p>
</div>
</div>
</div>
</section>
<section id="output-interpretation" class="level2">
<h2 class="anchored" data-anchor-id="output-interpretation">Output Interpretation</h2>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-2-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-1" role="tab" aria-controls="tabset-2-1" aria-selected="true">extractor</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-2-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-2-2" role="tab" aria-controls="tabset-2-2" aria-selected="false">processCluster</a></li></ul>
<div class="tab-content">
<div id="tabset-2-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-2-1-tab">
<ul>
<li><code>extractionProfile.tab.txt</code>: This breaks down the filtering per final extraction sequence file</li>
<li><code>extractionStats.tab.txt</code>: This has info how the whole extraction went</li>
</ul>
</div>
<div id="tabset-2-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-2-2-tab">
<p>All the data from the analysis done by <a href="https://seekdeep.brown.edu/processClusters_usage.html">processClusters</a> is located in the file: <strong>selectedClustersInfo.tab.txt</strong>.</p>
<p>Our main variables of interest are:</p>
<ul>
<li><code>s_Sample</code> = The name of the sample the haplotype appears in</li>
<li><code>h_popUID</code> = The population id given to haplotype. These are named with PopUID.XX where XX is the population rank number and this number is determined by the number of samples the haplotype appears in</li>
<li><code>c_clusterID</code> = The id given to the cluster in the sample, the lower the number the higher the relative abundance in the sample</li>
<li><code>c_ReadCnt</code> = The total number of reads associated with this cluster summed over all replicates</li>
<li><code>c_Consensus</code> = The consensus sequence for this cluster</li>
<li><code>c_bestExpected</code> = A string with the name of the closest reference sequence see below to see how to compare against expected references</li>
</ul>
</div>
</div>
</div>
</section>
<section id="references" class="level2 unnumbered">
<h2 class="unnumbered anchored" data-anchor-id="references">References</h2>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-Aydemir2018" class="csl-entry" role="listitem">
Aydemir, Ozkan, Mark Janko, Nick J Hathaway, Robert Verity, Melchior Kashamuka Mwandagalirwa, Antoinette K Tshefu, Sofonias K Tessema, et al. 2018. <span>“<span class="nocase">Drug-Resistance and Population Structure of Plasmodium falciparum Across the Democratic Republic of Congo Using High-Throughput Molecular Inversion Probes</span>.”</span> <em>The Journal of Infectious Diseases</em> 218 (6): 946–55. <a href="https://doi.org/10.1093/infdis/jiy223">https://doi.org/10.1093/infdis/jiy223</a>.
</div>
<div id="ref-Bal2022" class="csl-entry" role="listitem">
Balasubramanian, Sujata, Rachel Curtis-Robles, Bhagath Chirra, Lisa D. Auckland, Alan Mai, Virgilio Bocanegra-Garcia, Patti Clark, et al. 2022. <span>“<span class="nocase">Characterization of triatomine bloodmeal sources using direct Sanger sequencing and amplicon deep sequencing methods</span>.”</span> <em>Scientific Reports</em> 12 (1): 10234. <a href="https://doi.org/10.1038/s41598-022-14208-8">https://doi.org/10.1038/s41598-022-14208-8</a>.
</div>
<div id="ref-Boyce2019" class="csl-entry" role="listitem">
Boyce, Ross M, Nicholas Brazeau, Travis Fulton, Nick Hathaway, Michael Matte, Moses Ntaro, Edgar Mulogo, and Jonathan J Juliano. 2019. <span>“<span class="nocase">Prevalence of Molecular Markers of Antimalarial Drug Resistance across Altitudinal Transmission Zones in Highland Western Uganda</span>.”</span> <em>The American Journal of Tropical Medicine and Hygiene</em> 101 (4): 799–802. <a href="https://doi.org/10.4269/ajtmh.19-0081">https://doi.org/10.4269/ajtmh.19-0081</a>.
</div>
<div id="ref-Brazeau2019" class="csl-entry" role="listitem">
Brazeau, Nicholas F, Ashenafi Assefa, Hussein Mohammed, Heven Seme, Abeba G Tsadik, Jonathan B Parr, Corinna Keeler, et al. 2019. <span>“<span class="nocase">Pooled Deep Sequencing of Drug Resistance Loci from Plasmodium falciparum Parasites across Ethiopia</span>.”</span> <em>The American Journal of Tropical Medicine and Hygiene</em> 101 (5): 1139–43. <a href="https://doi.org/10.4269/ajtmh.19-0142">https://doi.org/10.4269/ajtmh.19-0142</a>.
</div>
<div id="ref-Hathaway2017" class="csl-entry" role="listitem">
Hathaway, Nicholas J., Christian M. Parobek, Jonathan J. Juliano, and Jeffrey A. Bailey. 2017. <span>“<span class="nocase">SeekDeep: single-base resolution de novo clustering for amplicon deep sequencing</span>.”</span> <em>Nucleic Acids Research</em> 46 (4): gkx1201–. <a href="https://doi.org/10.1093/nar/gkx1201">https://doi.org/10.1093/nar/gkx1201</a>.
</div>
<div id="ref-Hemming2020" class="csl-entry" role="listitem">
Hemming-Schroeder, Elizabeth, Daibin Zhong, Solomon Kibret, Amanda Chie, Ming-Chieh Lee, Guofa Zhou, Harrysone Atieli, Andrew Githeko, James W Kazura, and Guiyun Yan. 2020. <span>“<span class="nocase">Microgeographic Epidemiology of Malaria Parasites in an Irrigated Area of Western Kenya by Deep Amplicon Sequencing</span>.”</span> <em>The Journal of Infectious Diseases</em> 223 (8): 1456–65. <a href="https://doi.org/10.1093/infdis/jiaa520">https://doi.org/10.1093/infdis/jiaa520</a>.
</div>
<div id="ref-Osoti2022" class="csl-entry" role="listitem">
Osoti, Victor, Mercy Akinyi, Kevin Wamae, Kelvin M. Kimenyi, Zaydah de Laurent, Leonard Ndwiga, Paul Gichuki, et al. 2022. <span>“<span class="nocase">Targeted Amplicon Deep Sequencing for Monitoring Antimalarial Resistance Markers in Western Kenya</span>.”</span> <em>Antimicrobial Agents and Chemotherapy</em> 66 (4): e01945–21. <a href="https://doi.org/10.1128/aac.01945-21">https://doi.org/10.1128/aac.01945-21</a>.
</div>
<div id="ref-Pringle2019" class="csl-entry" role="listitem">
Pringle, Julia C., Amy Wesolowski, Sophie Berube, Tamaki Kobayashi, Mary E. Gebhardt, Modest Mulenga, Mike Chaponda, et al. 2019. <span>“<span class="nocase">High Plasmodium falciparum genetic diversity and temporal stability despite control efforts in high transmission settings along the international border between Zambia and the Democratic Republic of the Congo</span>.”</span> <em>Malaria Journal</em> 18 (1): 400. <a href="https://doi.org/10.1186/s12936-019-3023-4">https://doi.org/10.1186/s12936-019-3023-4</a>.
</div>
<div id="ref-Topazian2022" class="csl-entry" role="listitem">
Topazian, Hillary M., Kara A. Moser, Billy Ngasala, Peter O. Oluoch, Catherine S. Forconi, Lwidiko E. Mhamilawa, Ozkan Aydemir, et al. 2022. <span>“<span class="nocase">Low Complexity of Infection Is Associated With Molecular Persistence of Plasmodium falciparum in Kenya and Tanzania</span>.”</span> <em>Frontiers in Epidemiology</em> 2: 852237. <a href="https://doi.org/10.3389/fepid.2022.852237">https://doi.org/10.3389/fepid.2022.852237</a>.
</div>
<div id="ref-Wamae2022b" class="csl-entry" role="listitem">
Wamae, Kevin, Kelvin M. Kimenyi, Victor Osoti, Zaydah R. de Laurent, Leonard Ndwiga, Oksana Kharabora, Nicholas J. Hathaway, et al. 2022. <span>“<span class="nocase">Amplicon sequencing as a potential surveillance tool for complexity of infection and drug resistance markers in Plasmodium falciparum asymptomatic infections</span>.”</span> <em>Journal of Infectious Diseases</em>. <a href="https://doi.org/doi.org/10.1093/infdis/jiac144">https://doi.org/doi.org/10.1093/infdis/jiac144</a>.
</div>
<div id="ref-Wamae2022a" class="csl-entry" role="listitem">
Wamae, Kevin, Leonard Ndwiga, Oksana Kharabora, Kelvin Kimenyi, Victor Osoti, Zaydah de Laurent, Juliana Wambua, et al. 2022. <span>“<span class="nocase">Targeted Amplicon deep sequencing of ama1 and mdr1 to track within-host P. falciparum diversity throughout treatment in a clinical drug trial</span>.”</span> <em>Wellcome Open Research</em> 7: 95. <a href="https://doi.org/10.12688/wellcomeopenres.17736.1">https://doi.org/10.12688/wellcomeopenres.17736.1</a>.
</div>
</div>


</section>

<a onclick="window.scrollTo(0, 0); return false;" role="button" id="quarto-back-to-top"><i class="bi bi-arrow-up"></i> Back to top</a></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./Prepare_Sequencing.html" class="pagination-link" aria-label="Post-PCR: Prepare for Sequencing">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Post-PCR: Prepare for Sequencing</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./2_MACSE.html" class="pagination-link" aria-label="Step 2: MACSE">
        <span class="nav-page-text">Step 2: MACSE</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>Copyright 2024, Dionne C. Argyropoulos</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/dionnecargy">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="https://twitter.com/DionneArgy">
      <i class="bi bi-twitter" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>




</body></html>