---
title: "MACSE Processing"
output: html_notebook
---

# Before this stage: MACSE

To Run MACSE in Terminal, load: 
`export PATH=/home/argyropoulos/software/jre1.8.0_341/bin:$PATH`
`export PATH=/home/argyropoulos/software/macse_v2.06.jar/bin:$PATH`

Make file containing all fastq files: `cat /your/path/to/folder/*.fastq > newname.fastq`
Convert fastq to fasta: `paste - - - - < file.fastq | cut -f 1,2 | sed 's/^@/>/' | tr "\t" "\n" > file.fasta`
Add Ref seq: `cat ~/SeekDeep_DA/MACSE/forMACSE/dhfr_3D7_PlasmoDB.fasta P4dhfrFrac35.fasta > P4dhfrFrac35All.fasta`
Run MACSE: `java -jar macse_v2.06.jar -prog alignSequences -seq test.fasta`

Then copy across the files to the computer local drive for input into R. 

All MACSE runs were by **pool** by **marker** for each **Fraction Cut-Off** with respect to their SeekDeep runs. 
This means that the files will be input to the below code individually.

# MACSE Processing

This R Notebook contains the post-processing of MACSE data (.fasta) into a data frame (.csv) for manipulation in R. 

```{r Load packages, include=FALSE}
library(stringr)
library(dplyr)
library(tidyr)
library(janitor)
library(forcats)
library(data.table)
library(seqinr)
library(Biostrings)
```

## Nucleotide (NT) Processing

```{r}
filenames = Sys.glob(file.path("2_MACSE_Output_NT", pattern = "*.fasta"))

for (filename in filenames) {
  
  INPUT_NT = filename
  
  replace_strs <- c("2_MACSE_Output_NT" = "", ".fasta" = "")

  OUTPUT_1_NT = paste0("3_R_All_DrugR_Sequences_NT", str_replace_all(filename, replace_strs, ""), "_Aligned.csv")
  OUTPUT_2_NT = paste0("3_R_All_DrugR_Changes_NT", str_replace_all(filename, replace_strs, ""), "_Aligned_YorN.csv")
  OUTPUT_3_NT = paste0("3_R_All_DrugR_Changes_NT", str_replace_all(filename, replace_strs, ""), "_Aligned_NTchange.csv")
  
  ########## PART 1: General processing of input file  ########## 
  
  fastaFile <- readBStringSet(INPUT_NT)

  # Data manipulation 
    seq_name = names(fastaFile)
    sequence = paste(fastaFile)
    df_NT <- data.frame(seq_name, sequence)

  # Rename reference file
    df_NT1 <- df_NT %>% mutate(seq_name = ifelse(str_detect(seq_name, "Plasmodium falciparum 3D7"), "3D7_Ref",
                                             ifelse(str_detect(seq_name, "Plasmodium falciparum Dd2"), "Dd2_Ref",
                                                    ifelse(str_detect(seq_name, "Plasmodium falciparum HB3"), "HB3_Ref", 
                                                           df_NT$seq_name))))

    # Separate by codon
    n <- ((max(nchar(df_NT1$sequence)) - 1) %/% 3) + 1
    df_NT2 <- df_NT1 %>% separate(sequence, into=paste0("NT", seq(n)), sep=seq(n-1)*3, remove = FALSE)

    # Data Manipulation 2
    df_NT3 <- df_NT2 %>% dplyr::select(-sequence) %>% mutate(seq_name = gsub("-", "\\.",seq_name)) %>% mutate_all(~ str_replace_all(., "!", NA_character_)) 
    df_NT4 <- df_NT3 %>% separate(seq_name, c("Isolate", "MID", "Pool", "Date", "Haplotype"), sep="\\.", extra = "merge", fill = "right") %>% arrange(Isolate)

    # Transform and put 3D7_Ref to first column
    df_NT5 <- as.data.frame(t(df_NT3))
    df_NT6 <- df_NT5 %>% row_to_names(1) %>% dplyr::select(`3D7_Ref`, everything())

  ########## PART 2: Relabel codons based on reference 3D7 positions  ########## 
  # - Loop through rows of transposed df
  # - If find != “---”, label and increment position by 1
  # - If find “---”, label as something else to indicate alignment positions w/o ref in 3D7

    # Add initial NT value as column for unique values for merging
    df_NT6 <- df_NT6 %>% mutate(rownames=row.names(.))

    # Rearrange columns
    df_NT6 <- df_NT6 %>% select("rownames", "3D7_Ref", everything())

    # Initialise first positions and df object
    pos <- 0
    tmp <- 0
    new_df <- data.frame()

    # Loop every row to relabel 3D7 positions
for (row in 1:nrow(df_NT6)) {
  # get 3D7 codon triplet
  codon <- df_NT6[row, "3D7_Ref"]
  row_name <- row.names(df_NT6[row, ])
  # if 3D7 codon triplet is valid, relabel and increment count
  # if 3D7 codon triplet is invalid (i.e., "---"), relabel with appended temp value
  if (codon != "---") {
    pos <- pos + 1  # increments pos by 1 if codon is valid
    tmp <- 0  # resets tmp to 1 if the codon is valid
    label <- paste0("3D7_", pos)
  } else {
    tmp <- tmp + 1  # increments tmp by 1 if codon is invalid
    label <- paste0("3D7_", pos, ".", tmp)
  }
  new_df <- rbind(new_df, cbind(row_name, codon, label))
}

    # Rename new_df columns
    colnames(new_df) <- c("row_name", "codon", "3D7_Pos")

    # Join new labels with previous df with data, remove rownames column
    df_NT6 <- left_join(df_NT6, new_df, by=c("rownames" = "row_name", "3D7_Ref" = "codon"))
    df_NT6 <- df_NT6 %>% select("3D7_Pos", "3D7_Ref", everything()) %>% select(-c(rownames))

    # Remove temporary variables
    remove(row, pos, tmp, new_df, label)

    # Write data frame
    write.csv(df_NT6, OUTPUT_1_NT, row.names = FALSE)

  ########## PART 3: Compare isolates to 3D7 and generate multiple output files  ########## 
  # - Output 1: “Y/N” to refer to codons compared to ref
  # - Output 2: Nucleotide changes (e.g., ATC -> ATG): DA changed " > ", to substring(df_NT6$`3D7_Pos`, 5),

    df_NT7 <- df_NT6 # Y/N output
    df_NT8 <- df_NT6 # NT change output

    # Save list of isolates
    isolatelist <- df_NT6 %>% colnames() %>% data.frame() %>% filter(.!=c("3D7_Pos", "3D7_Ref"))

    # Loop through list of isolates (i.e., columns)
for (row in 1:nrow(isolatelist)) {
  
  # Get isolate ID
  id <- isolatelist[row, "."]
  
  ### Compare isolate codon triplets to 3D7_Ref codon triplets
  
  # "Y": same codons, "N": different codons, "---": gap in alignment
  # codon triplets are kept if 3D7_Ref is "---"
  df_NT7[[id]] <- ifelse(df_NT6[[id]] == "---", "-",
                         ifelse(df_NT6$`3D7_Ref` == "---", df_NT6[[id]],
                                ifelse(df_NT6[[id]] == df_NT6$`3D7_Ref`, "Y", "N")))
  
  # "NA": same codons, "xxx > xxx": different codons, "-": gap in alignment
  # codon triplets are kept if 3D7_Ref is "---"
  df_NT8[[id]] <- ifelse(df_NT6[[id]] == "---", "-",
                         ifelse(df_NT6$`3D7_Ref` == "---", paste0("- > ", df_NT6[[id]]),
                                ifelse(df_NT6[[id]] == df_NT6$`3D7_Ref`, NA, paste0(df_NT6$`3D7_Ref`, substring(df_NT6$`3D7_Pos`, 5), df_NT6[[id]]))))
}
    
    # Remove temporary variables
    remove(row, id)

    # Write data frame
    write.csv(df_NT7, OUTPUT_2_NT, row.names = FALSE)
    write.csv(df_NT8, OUTPUT_3_NT, row.names = FALSE)

} 

```

## Amino Acid (AA) Processing 

```{r}
filenames_AA = Sys.glob(file.path("2_MACSE_Output_AA", pattern = "*.fasta"))

for (filename in filenames_AA) {
  
  INPUT_AA = filename
  
  replace_strs <- c("2_MACSE_Output_AA" = "", ".fasta" = "")

  OUTPUT_1_AA = paste0("3_R_All_DrugR_Sequences_AA", str_replace_all(filename, replace_strs, ""), "_Aligned.csv")
  OUTPUT_2_AA = paste0("3_R_All_DrugR_Changes_AA", str_replace_all(filename, replace_strs, ""), "_Aligned_YorN.csv")
  OUTPUT_3_AA = paste0("3_R_All_DrugR_Changes_AA", str_replace_all(filename, replace_strs, ""), "_Aligned_AAchange.csv")

  ########## PART 1: General processing of input file  ########## 
  
    fastaFile <- readBStringSet(INPUT_AA)

    # Data manipulation 
    seq_name = names(fastaFile)
    sequence = paste(fastaFile)
    df_AA <- data.frame(seq_name, sequence)

    # Rename reference file
    df_AA1 <- df_AA %>% mutate(seq_name = ifelse(str_detect(seq_name, "Plasmodium falciparum 3D7"), "3D7_Ref",
                                             ifelse(str_detect(seq_name, "Plasmodium falciparum Dd2"), "Dd2_Ref",
                                                    ifelse(str_detect(seq_name, "Plasmodium falciparum HB3"), "HB3_Ref", 
                                                           df_AA$seq_name))))
    # Separate by codon
    df_AA2 <- df_AA1 %>% data.frame(do.call("rbind", strsplit(as.character(df_AA$sequence), "", fixed = TRUE)))

    # Data Manipulation 2
    df_AA3 <- df_AA2 %>% dplyr::select(-sequence) %>% mutate(seq_name = gsub("-", "\\.",seq_name)) %>% mutate_all(~ str_replace_all(., "!", NA_character_)) 
    df_AA4 <- df_AA3 %>% separate(seq_name, c("Isolate", "MID", "Pool", "Date", "Haplotype"), 
                              sep="\\.", extra = "merge", fill = "right") %>% arrange(Isolate)

    # Transform and put 3D7_Ref to first column
    df_AA5 <- as.data.frame(t(df_AA3))
    df_AA6 <- df_AA5 %>% row_to_names(1) %>% dplyr::select(`3D7_Ref`, everything())

      ########## PART 2: Relabel codons based on reference 3D7 positions  ########## 
      # - Loop through rows of transposed df
      # - If find != “---”, label and increment position by 1
      # - If find “---”, label as something else to indicate alignment positions w/o ref in 3D7
    
    # Add initial AA value as column for unique values for merging
    df_AA6 <- df_AA6 %>% mutate(rownames=row.names(.))

    # Rearrange columns
    df_AA6 <- df_AA6 %>% select("rownames", "3D7_Ref", everything())

    # Initialise first positions and df object
    pos <- 0
    tmp <- 0
    new_df <- data.frame()

# Loop every row to relabel 3D7 positions
for (row in 1:nrow(df_AA6)) {
  # get 3D7 codon triplet
  codon <- df_AA6[row, "3D7_Ref"]
  row_name <- row.names(df_AA6[row, ])
  # if 3D7 codon triplet is valid, relabel and increment count
  # if 3D7 codon triplet is invalid (i.e., "-"), relabel with appended temp value ## DCA CHANGED "---" to "-"
  if (codon != "-") { ## DCA CHANGED "---" to "-"
    pos <- pos + 1  # increments pos by 1 if codon is valid
    tmp <- 0  # resets tmp to 1 if the codon is valid
    label <- paste0("3D7_", pos)
  } else {
    tmp <- tmp + 1  # increments tmp by 1 if codon is invalid
    label <- paste0("3D7_", pos, ".", tmp)
  }
  new_df <- rbind(new_df, cbind(row_name, codon, label))
}

    # Rename new_df columns
    colnames(new_df) <- c("row_name", "codon", "3D7_Pos")

    # Join new labels with previous df with data, remove rownames column
    df_AA6 <- left_join(df_AA6, new_df, by=c("rownames" = "row_name", "3D7_Ref" = "codon"))
    df_AA6 <- df_AA6 %>% select("3D7_Pos", "3D7_Ref", everything()) %>% select(-c(rownames))

    # Remove temporary variables
    remove(row, pos, tmp, new_df, label)

    # Write data frame
    write.csv(df_AA6, OUTPUT_1_AA, row.names = FALSE)

    ########## PART 3: Compare isolates to 3D7 and generate multiple output files  ########## 
    # - Output 1: “Y/N” to refer to codons compared to ref
    # - Output 2: Nucleotide changes (e.g., ATC -> ATG): DA changed " > ", to substring(df_AA6$`3D7_Pos`, 5),
    df_AA7 <- df_AA6 # Y/N output
    df_AA8 <- df_AA6 # AA change output

    # Save list of isolates
    isolatelist <- df_AA6 %>% colnames() %>% data.frame() %>% filter(.!=c("3D7_Pos", "3D7_Ref"))

    # Loop through list of isolates (i.e., columns)
    for (row in 1:nrow(isolatelist)) {
  
      # Get isolate ID
      id <- isolatelist[row, "."]
  
      ### Compare isolate codon triplets to 3D7_Ref codon triplets
  
      # "Y": same codons, "N": different codons, "---": gap in alignment
      # codon triplets are kept if 3D7_Ref is "---"; ## DCA CHANGED "---" to "-"
      df_AA7[[id]] <- ifelse(df_AA6[[id]] == "-", "-", ## DCA CHANGED "---" to "-"
                         ifelse(df_AA6$`3D7_Ref` == "-", df_AA6[[id]], ## DCA CHANGED "---" to "-"
                                ifelse(df_AA6[[id]] == df_AA6$`3D7_Ref`, "Y", "N")))

      # "NA": same codons, "xxx > xxx": different codons, "-": gap in alignment ## This is where I think we need to modify! 
      # codon triplets are kept if 3D7_Ref is "---"
      df_AA8[[id]] <- ifelse(df_AA6[[id]] == "-", "-",
                         ifelse(df_AA6$`3D7_Ref` == "-", paste0("- > ", df_AA6[[id]]),
                                ifelse(df_AA6[[id]] == df_AA6$`3D7_Ref`, NA, paste0(df_AA6$`3D7_Ref`, substring(df_NT6$`3D7_Pos`, 5), df_AA6[[id]]))))  
}

      # Remove temporary variables
    remove(row, id)

    # Write data frame
    write.csv(df_AA7, OUTPUT_2_AA, row.names = FALSE)
    write.csv(df_AA8, OUTPUT_3_AA, row.names = FALSE)
} 

```

# Join by Marker 

## Function 
Help from ChatGPT: 
```{r}
# Load necessary libraries
library(dplyr)
library(readr)

# Define a function to merge files based on the specified pattern
merge_files <- function(directory, pattern, common_columns) {
  file_paths <- list.files(path = directory, pattern = pattern, full.names = TRUE)
  merged_data <- read_csv(file_paths[[1]])
  
  # Use a for loop to left join each file to the merged data
  for (i in 2:length(file_paths)) {
    current_file <-  read_csv(file_paths[[i]], col_types = cols(.default = col_character()))

    merged_data <- left_join(merged_data, current_file, by = common_columns)
  }
  
  return(merged_data)
}

```

## Frac 3.5% 
```{r}
# Merge files with pattern "[marker]" and common columns "3D7_Pos" and "3D7_Ref"
merged_crt_data <- merge_files("3_R_All_DrugR_Sequences_NT", "crt.*Frac35_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_dhfr_data <- merge_files("3_R_All_DrugR_Sequences_NT", "dhfr.*Frac35_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_dhps1_data <- merge_files("3_R_All_DrugR_Sequences_NT", "dhps1.*Frac35_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_dhps2_data <- merge_files("3_R_All_DrugR_Sequences_NT", "dhps2.*Frac35_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_K131_data <- merge_files("3_R_All_DrugR_Sequences_NT", "K131.*Frac35_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_K132_data <- merge_files("3_R_All_DrugR_Sequences_NT", "K132.*Frac35_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_mdr11_data <- merge_files("3_R_All_DrugR_Sequences_NT", "mdr11.*Frac35_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_mdr12_data <- merge_files("3_R_All_DrugR_Sequences_NT", "mdr12.*Frac35_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_mdr13_data <- merge_files("3_R_All_DrugR_Sequences_NT", "mdr13.*Frac35_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
#merged_aat1_data <- merge_files("3_R_All_DrugR_Sequences_NT", "aat1.*Frac35_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
#merged_pfs471_data <- merge_files("3_R_All_DrugR_Sequences_NT", "pfs471.*Frac35_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
#merged_pfs472_data <- merge_files("3_R_All_DrugR_Sequences_NT", "pfs472.*Frac35_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))

merged_crt_YN <- merge_files("3_R_All_DrugR_Changes_NT", "crt.*Frac35_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_dhfr_YN <- merge_files("3_R_All_DrugR_Changes_NT", "dhfr.*Frac35_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_dhps1_YN <- merge_files("3_R_All_DrugR_Changes_NT", "dhps1.*Frac35_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_dhps2_YN <- merge_files("3_R_All_DrugR_Changes_NT", "dhps2.*Frac35_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_K131_YN <- merge_files("3_R_All_DrugR_Changes_NT", "K131.*Frac35_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_K132_YN <- merge_files("3_R_All_DrugR_Changes_NT", "K132.*Frac35_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_mdr11_YN <- merge_files("3_R_All_DrugR_Changes_NT", "mdr11.*Frac35_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_mdr12_YN <- merge_files("3_R_All_DrugR_Changes_NT", "mdr12.*Frac35_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_mdr13_YN <- merge_files("3_R_All_DrugR_Changes_NT", "mdr13.*Frac35_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
#merged_aat1_YN <- merge_files("3_R_All_DrugR_Changes_NT", "aat1.*Frac35_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
#merged_pfs471_YN <- merge_files("3_R_All_DrugR_Changes_NT", "pfs471.*Frac35_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
#merged_pfs472_YN <- merge_files("3_R_All_DrugR_Changes_NT", "pfs472.*Frac35_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))

merged_crt_Change <- merge_files("3_R_All_DrugR_Changes_NT", "crt.*Frac35_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_dhfr_Change <- merge_files("3_R_All_DrugR_Changes_NT", "dhfr.*Frac35_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_dhps1_Change <- merge_files("3_R_All_DrugR_Changes_NT", "dhps1.*Frac35_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_dhps2_Change <- merge_files("3_R_All_DrugR_Changes_NT", "dhps2.*Frac35_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_K131_Change <- merge_files("3_R_All_DrugR_Changes_NT", "K131.*Frac35_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_K132_Change <- merge_files("3_R_All_DrugR_Changes_NT", "K132.*Frac35_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_mdr11_Change <- merge_files("3_R_All_DrugR_Changes_NT", "mdr11.*Frac35_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_mdr12_Change <- merge_files("3_R_All_DrugR_Changes_NT", "mdr12.*Frac35_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_mdr13_Change <- merge_files("3_R_All_DrugR_Changes_NT", "mdr13.*Frac35_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
#merged_aat1_Change <- merge_files("3_R_All_DrugR_Changes_NT", "aat1.*Frac35_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
#merged_pfs471_Change <- merge_files("3_R_All_DrugR_Changes_NT", "pfs471.*Frac35_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
#merged_pfs472_Change <- merge_files("3_R_All_DrugR_Changes_NT", "pfs472.*Frac35_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))

merged_AA_crt_data <- merge_files("3_R_All_DrugR_Sequences_AA", "crt.*Frac35_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_dhfr_data <- merge_files("3_R_All_DrugR_Sequences_AA", "dhfr.*Frac35_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_dhps1_data <- merge_files("3_R_All_DrugR_Sequences_AA", "dhps1.*Frac35_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_dhps2_data <- merge_files("3_R_All_DrugR_Sequences_AA", "dhps2.*Frac35_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_K131_data <- merge_files("3_R_All_DrugR_Sequences_AA", "K131.*Frac35_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_K132_data <- merge_files("3_R_All_DrugR_Sequences_AA", "K132.*Frac35_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_mdr11_data <- merge_files("3_R_All_DrugR_Sequences_AA", "mdr11.*Frac35_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_mdr12_data <- merge_files("3_R_All_DrugR_Sequences_AA", "mdr12.*Frac35_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_mdr13_data <- merge_files("3_R_All_DrugR_Sequences_AA", "mdr13.*Frac35_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
#merged_AA_aat1_data <- merge_files("3_R_All_DrugR_Sequences_AA", "aat1.*Frac35_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
#merged_AA_pfs471_data <- merge_files("3_R_All_DrugR_Sequences_AA", "pfs471.*Frac35_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
#merged_AA_pfs472_data <- merge_files("3_R_All_DrugR_Sequences_AA", "pfs472.*Frac35_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))

merged_AA_crt_YN <- merge_files("3_R_All_DrugR_Changes_AA", "crt.*Frac35_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_dhfr_YN <- merge_files("3_R_All_DrugR_Changes_AA", "dhfr.*Frac35_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_dhps1_YN <- merge_files("3_R_All_DrugR_Changes_AA", "dhps1.*Frac35_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_dhps2_YN <- merge_files("3_R_All_DrugR_Changes_AA", "dhps2.*Frac35_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_K131_YN <- merge_files("3_R_All_DrugR_Changes_AA", "K131.*Frac35_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_K132_YN <- merge_files("3_R_All_DrugR_Changes_AA", "K132.*Frac35_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_mdr11_YN <- merge_files("3_R_All_DrugR_Changes_AA", "mdr11.*Frac35_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_mdr12_YN <- merge_files("3_R_All_DrugR_Changes_AA", "mdr12.*Frac35_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_mdr13_YN <- merge_files("3_R_All_DrugR_Changes_AA", "mdr13.*Frac35_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
#merged_AA_aat1_YN <- merge_files("3_R_All_DrugR_Changes_AA", "aat1.*Frac35_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
#merged_AA_pfs471_YN <- merge_files("3_R_All_DrugR_Changes_AA", "pfs471.*Frac35_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
#merged_AA_pfs472_YN <- merge_files("3_R_All_DrugR_Changes_AA", "pfs472.*Frac35_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))

merged_AA_crt_Change <- merge_files("3_R_All_DrugR_Changes_AA", "crt.*Frac35_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_dhfr_Change <- merge_files("3_R_All_DrugR_Changes_AA", "dhfr.*Frac35_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_dhps1_Change <- merge_files("3_R_All_DrugR_Changes_AA", "dhps1.*Frac35_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_dhps2_Change <- merge_files("3_R_All_DrugR_Changes_AA", "dhps2.*Frac35_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_K131_Change <- merge_files("3_R_All_DrugR_Changes_AA", "K131.*Frac35_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_K132_Change <- merge_files("3_R_All_DrugR_Changes_AA", "K132.*Frac35_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_mdr11_Change <- merge_files("3_R_All_DrugR_Changes_AA", "mdr11.*Frac35_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_mdr12_Change <- merge_files("3_R_All_DrugR_Changes_AA", "mdr12.*Frac35_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_AA_mdr13_Change <- merge_files("3_R_All_DrugR_Changes_AA", "mdr13.*Frac35_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
#merged_AA_aat1_Change <- merge_files("3_R_All_DrugR_Changes_AA", "aat1.*Frac35_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
#merged_AA_pfs471_Change <- merge_files("3_R_All_DrugR_Changes_AA", "pfs471.*Frac35_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
#merged_AA_pfs472_Change <- merge_files("3_R_All_DrugR_Changes_AA", "pfs472.*Frac35_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
```

## Frac 5% 
```{r}
# Merge files with pattern "[marker]" and common columns "3D7_Pos" and "3D7_Ref"
merged_F5_crt_data <- merge_files("3_R_All_DrugR_Sequences_NT", "crt.*Frac5_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_dhfr_data <- merge_files("3_R_All_DrugR_Sequences_NT", "dhfr.*Frac5_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_dhps1_data <- merge_files("3_R_All_DrugR_Sequences_NT", "dhps1.*Frac5_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_dhps2_data <- merge_files("3_R_All_DrugR_Sequences_NT", "dhps2.*Frac5_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_K131_data <- merge_files("3_R_All_DrugR_Sequences_NT", "K131.*Frac5_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_K132_data <- merge_files("3_R_All_DrugR_Sequences_NT", "K132.*Frac5_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_mdr11_data <- merge_files("3_R_All_DrugR_Sequences_NT", "mdr11.*Frac5_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_mdr12_data <- merge_files("3_R_All_DrugR_Sequences_NT", "mdr12.*Frac5_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_mdr13_data <- merge_files("3_R_All_DrugR_Sequences_NT", "mdr13.*Frac5_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
#merged_F5_aat1_data <- merge_files("3_R_All_DrugR_Sequences_NT", "aat1.*Frac5_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
#merged_F5_pfs471_data <- merge_files("3_R_All_DrugR_Sequences_NT", "pfs471.*Frac5_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
#merged_F5_pfs472_data <- merge_files("3_R_All_DrugR_Sequences_NT", "pfs472.*Frac5_All_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))

merged_F5_crt_YN <- merge_files("3_R_All_DrugR_Changes_NT", "crt.*Frac5_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_dhfr_YN <- merge_files("3_R_All_DrugR_Changes_NT", "dhfr.*Frac5_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_dhps1_YN <- merge_files("3_R_All_DrugR_Changes_NT", "dhps1.*Frac5_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_dhps2_YN <- merge_files("3_R_All_DrugR_Changes_NT", "dhps2.*Frac5_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_K131_YN <- merge_files("3_R_All_DrugR_Changes_NT", "K131.*Frac5_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_K132_YN <- merge_files("3_R_All_DrugR_Changes_NT", "K132.*Frac5_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_mdr11_YN <- merge_files("3_R_All_DrugR_Changes_NT", "mdr11.*Frac5_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_mdr12_YN <- merge_files("3_R_All_DrugR_Changes_NT", "mdr12.*Frac5_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_mdr13_YN <- merge_files("3_R_All_DrugR_Changes_NT", "mdr13.*Frac5_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
#merged_F5_aat1_YN <- merge_files("3_R_All_DrugR_Changes_NT", "aat1.*Frac5_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
#merged_F5_pfs471_YN <- merge_files("3_R_All_DrugR_Changes_NT", "pfs471.*Frac5_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
#merged_F5_pfs472_YN <- merge_files("3_R_All_DrugR_Changes_NT", "pfs472.*Frac5_All_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))

merged_F5_crt_Change <- merge_files("3_R_All_DrugR_Changes_NT", "crt.*Frac5_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_dhfr_Change <- merge_files("3_R_All_DrugR_Changes_NT", "dhfr.*Frac5_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_dhps1_Change <- merge_files("3_R_All_DrugR_Changes_NT", "dhps1.*Frac5_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_dhps2_Change <- merge_files("3_R_All_DrugR_Changes_NT", "dhps2.*Frac5_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_K131_Change <- merge_files("3_R_All_DrugR_Changes_NT", "K131.*Frac5_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_K132_Change <- merge_files("3_R_All_DrugR_Changes_NT", "K132.*Frac5_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_mdr11_Change <- merge_files("3_R_All_DrugR_Changes_NT", "mdr11.*Frac5_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_mdr12_Change <- merge_files("3_R_All_DrugR_Changes_NT", "mdr12.*Frac5_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_mdr13_Change <- merge_files("3_R_All_DrugR_Changes_NT", "mdr13.*Frac5_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
#merged_F5_aat1_Change <- merge_files("3_R_All_DrugR_Changes_NT", "aat1.*Frac5_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
#merged_F5_pfs471_Change <- merge_files("3_R_All_DrugR_Changes_NT", "pfs471.*Frac5_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
#merged_F5_pfs472_Change <- merge_files("3_R_All_DrugR_Changes_NT", "pfs472.*Frac5_All_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))

merged_F5_AA_crt_data <- merge_files("3_R_All_DrugR_Sequences_AA", "crt.*Frac5_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_dhfr_data <- merge_files("3_R_All_DrugR_Sequences_AA", "dhfr.*Frac5_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_dhps1_data <- merge_files("3_R_All_DrugR_Sequences_AA", "dhps1.*Frac5_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_dhps2_data <- merge_files("3_R_All_DrugR_Sequences_AA", "dhps2.*Frac5_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_K131_data <- merge_files("3_R_All_DrugR_Sequences_AA", "K131.*Frac5_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_K132_data <- merge_files("3_R_All_DrugR_Sequences_AA", "K132.*Frac5_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_mdr11_data <- merge_files("3_R_All_DrugR_Sequences_AA", "mdr11.*Frac5_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_mdr12_data <- merge_files("3_R_All_DrugR_Sequences_AA", "mdr12.*Frac5_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_mdr13_data <- merge_files("3_R_All_DrugR_Sequences_AA", "mdr13.*Frac5_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
#merged_F5_AA_aat1_data <- merge_files("3_R_All_DrugR_Sequences_AA", "aat1.*Frac5_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
#merged_F5_AA_pfs471_data <- merge_files("3_R_All_DrugR_Sequences_AA", "pfs471.*Frac5_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
#merged_F5_AA_pfs472_data <- merge_files("3_R_All_DrugR_Sequences_AA", "pfs472.*Frac5_All_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))

merged_F5_AA_crt_YN <- merge_files("3_R_All_DrugR_Changes_AA", "crt.*Frac5_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_dhfr_YN <- merge_files("3_R_All_DrugR_Changes_AA", "dhfr.*Frac5_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_dhps1_YN <- merge_files("3_R_All_DrugR_Changes_AA", "dhps1.*Frac5_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_dhps2_YN <- merge_files("3_R_All_DrugR_Changes_AA", "dhps2.*Frac5_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_K131_YN <- merge_files("3_R_All_DrugR_Changes_AA", "K131.*Frac5_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_K132_YN <- merge_files("3_R_All_DrugR_Changes_AA", "K132.*Frac5_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_mdr11_YN <- merge_files("3_R_All_DrugR_Changes_AA", "mdr11.*Frac5_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_mdr12_YN <- merge_files("3_R_All_DrugR_Changes_AA", "mdr12.*Frac5_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_mdr13_YN <- merge_files("3_R_All_DrugR_Changes_AA", "mdr13.*Frac5_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
#merged_F5_AA_aat1_YN <- merge_files("3_R_All_DrugR_Changes_AA", "aat1.*Frac5_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
#merged_F5_AA_pfs471_YN <- merge_files("3_R_All_DrugR_Changes_AA", "pfs471.*Frac5_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
#merged_F5_AA_pfs472_YN <- merge_files("3_R_All_DrugR_Changes_AA", "pfs472.*Frac5_All_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))

merged_F5_AA_crt_Change <- merge_files("3_R_All_DrugR_Changes_AA", "crt.*Frac5_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_dhfr_Change <- merge_files("3_R_All_DrugR_Changes_AA", "dhfr.*Frac5_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_dhps1_Change <- merge_files("3_R_All_DrugR_Changes_AA", "dhps1.*Frac5_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_dhps2_Change <- merge_files("3_R_All_DrugR_Changes_AA", "dhps2.*Frac5_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_K131_Change <- merge_files("3_R_All_DrugR_Changes_AA", "K131.*Frac5_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_K132_Change <- merge_files("3_R_All_DrugR_Changes_AA", "K132.*Frac5_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_mdr11_Change <- merge_files("3_R_All_DrugR_Changes_AA", "mdr11.*Frac5_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_mdr12_Change <- merge_files("3_R_All_DrugR_Changes_AA", "mdr12.*Frac5_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
merged_F5_AA_mdr13_Change <- merge_files("3_R_All_DrugR_Changes_AA", "mdr13.*Frac5_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
#merged_F5_AA_aat1_Change <- merge_files("3_R_All_DrugR_Changes_AA", "aat1.*Frac5_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
#merged_F5_AA_pfs471_Change <- merge_files("3_R_All_DrugR_Changes_AA", "pfs471.*Frac5_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
#merged_F5_AA_pfs472_Change <- merge_files("3_R_All_DrugR_Changes_AA", "pfs472.*Frac5_All_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
```

## Write .csv 

### Frac 3.5% 
```{r}
write.csv(merged_crt_data, "4_R_Compiled_DrugR_Sequences_NT/crt_Frac35_All_NT_Aligned.csv", row.names = FALSE)
write.csv(merged_dhfr_data, "4_R_Compiled_DrugR_Sequences_NT/dhfr_Frac35_All_NT_Aligned.csv", row.names = FALSE)
write.csv(merged_dhps1_data, "4_R_Compiled_DrugR_Sequences_NT/dhps1_Frac35_All_NT_Aligned.csv", row.names = FALSE)
write.csv(merged_dhps2_data, "4_R_Compiled_DrugR_Sequences_NT/dhps2_Frac35_All_NT_Aligned.csv", row.names = FALSE)
write.csv(merged_K131_data, "4_R_Compiled_DrugR_Sequences_NT/K131_Frac35_All_NT_Aligned.csv", row.names = FALSE)
write.csv(merged_K132_data, "4_R_Compiled_DrugR_Sequences_NT/K132_Frac35_All_NT_Aligned.csv", row.names = FALSE)
write.csv(merged_mdr11_data, "4_R_Compiled_DrugR_Sequences_NT/mdr11_Frac35_All_NT_Aligned.csv", row.names = FALSE)
write.csv(merged_mdr12_data, "4_R_Compiled_DrugR_Sequences_NT/mdr12_Frac35_All_NT_Aligned.csv", row.names = FALSE)
write.csv(merged_mdr13_data, "4_R_Compiled_DrugR_Sequences_NT/mdr13_Frac35_All_NT_Aligned.csv", row.names = FALSE)
#write.csv(merged_aat1_data, "4_R_Compiled_DrugR_Sequences_NT/aat1_Frac35_All_NT_Aligned.csv", row.names = FALSE)
#write.csv(merged_pfs471_data, "4_R_Compiled_DrugR_Sequences_NT/pfs471_Frac35_All_NT_Aligned.csv", row.names = FALSE)
#write.csv(merged_pfs472_data, "4_R_Compiled_DrugR_Sequences_NT/pfs472_Frac35_All_NT_Aligned.csv", row.names = FALSE)

write.csv(merged_crt_YN, "4_R_Compiled_DrugR_Sequences_NT/crt_Frac35_All_NT_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_dhfr_YN, "4_R_Compiled_DrugR_Sequences_NT/dhfr_Frac35_All_NT_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_dhps1_YN, "4_R_Compiled_DrugR_Sequences_NT/dhps1_Frac35_All_NT_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_dhps2_YN, "4_R_Compiled_DrugR_Sequences_NT/dhps2_Frac35_All_NT_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_K131_YN, "4_R_Compiled_DrugR_Sequences_NT/K131_Frac35_All_NT_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_K132_YN, "4_R_Compiled_DrugR_Sequences_NT/K132_Frac35_All_NT_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_mdr11_YN, "4_R_Compiled_DrugR_Sequences_NT/mdr11_Frac35_All_NT_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_mdr12_YN, "4_R_Compiled_DrugR_Sequences_NT/mdr12_Frac35_All_NT_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_mdr13_YN, "4_R_Compiled_DrugR_Sequences_NT/mdr13_Frac35_All_NT_Aligned_YorN.csv", row.names = FALSE)
#write.csv(merged_aat1_YN, "4_R_Compiled_DrugR_Sequences_NT/aat1_Frac35_All_NT_Aligned_YorN.csv", row.names = FALSE)
#write.csv(merged_pfs471_YN, "4_R_Compiled_DrugR_Sequences_NT/pfs471_Frac35_All_NT_Aligned_YorN.csv", row.names = FALSE)
#write.csv(merged_pfs472_YN, "4_R_Compiled_DrugR_Sequences_NT/pfs472_Frac35_All_NT_Aligned_YorN.csv", row.names = FALSE)

write.csv(merged_crt_Change, "4_R_Compiled_DrugR_Sequences_NT/crt_Frac35_All_NT_Aligned_NTchange.csv", row.names = FALSE)
write.csv(merged_dhfr_Change, "4_R_Compiled_DrugR_Sequences_NT/dhfr_Frac35_All_NT_Aligned_NTchange.csv", row.names = FALSE)
write.csv(merged_dhps1_Change, "4_R_Compiled_DrugR_Sequences_NT/dhps1_Frac35_All_NT_Aligned_NTchange.csv", row.names = FALSE)
write.csv(merged_dhps2_Change, "4_R_Compiled_DrugR_Sequences_NT/dhps2_Frac35_All_NT_Aligned_NTchange.csv", row.names = FALSE)
write.csv(merged_K131_Change, "4_R_Compiled_DrugR_Sequences_NT/K131_Frac35_All_NT_Aligned_NTchange.csv", row.names = FALSE)
write.csv(merged_K132_Change, "4_R_Compiled_DrugR_Sequences_NT/K132_Frac35_All_NT_Aligned_NTchange.csv", row.names = FALSE)
write.csv(merged_mdr11_Change, "4_R_Compiled_DrugR_Sequences_NT/mdr11_Frac35_All_NT_Aligned_NTchange.csv", row.names = FALSE)
write.csv(merged_mdr12_Change, "4_R_Compiled_DrugR_Sequences_NT/mdr12_Frac35_All_NT_Aligned_NTchange.csv", row.names = FALSE)
write.csv(merged_mdr13_Change, "4_R_Compiled_DrugR_Sequences_NT/mdr13_Frac35_All_NT_Aligned_NTchange.csv", row.names = FALSE)
#write.csv(merged_aat1_Change, "4_R_Compiled_DrugR_Sequences_NT/aat1_Frac35_All_NT_Aligned_NTchange.csv", row.names = FALSE)
#write.csv(merged_pfs471_Change, "4_R_Compiled_DrugR_Sequences_NT/pfs471_Frac35_All_NT_Aligned_NTchange.csv", row.names = FALSE)
#write.csv(merged_pfs472_Change, "4_R_Compiled_DrugR_Sequences_NT/pfs472_Frac35_All_NT_Aligned_NTchange.csv", row.names = FALSE)

write.csv(merged_AA_crt_data, "4_R_Compiled_DrugR_Sequences_AA/crt_Frac35_All_AA_Aligned.csv", row.names = FALSE)
write.csv(merged_AA_dhfr_data, "4_R_Compiled_DrugR_Sequences_AA/dhfr_Frac35_All_AA_Aligned.csv", row.names = FALSE)
write.csv(merged_AA_dhps1_data, "4_R_Compiled_DrugR_Sequences_AA/dhps1_Frac35_All_AA_Aligned.csv", row.names = FALSE)
write.csv(merged_AA_dhps2_data, "4_R_Compiled_DrugR_Sequences_AA/dhps2_Frac35_All_AA_Aligned.csv", row.names = FALSE)
write.csv(merged_AA_K131_data, "4_R_Compiled_DrugR_Sequences_AA/K131_Frac35_All_AA_Aligned.csv", row.names = FALSE)
write.csv(merged_AA_K132_data, "4_R_Compiled_DrugR_Sequences_AA/K132_Frac35_All_AA_Aligned.csv", row.names = FALSE)
write.csv(merged_AA_mdr11_data, "4_R_Compiled_DrugR_Sequences_AA/mdr11_Frac35_All_AA_Aligned.csv", row.names = FALSE)
write.csv(merged_AA_mdr12_data, "4_R_Compiled_DrugR_Sequences_AA/mdr12_Frac35_All_AA_Aligned.csv", row.names = FALSE)
write.csv(merged_AA_mdr13_data, "4_R_Compiled_DrugR_Sequences_AA/mdr13_Frac35_All_AA_Aligned.csv", row.names = FALSE)
#write.csv(merged_AA_aat1_data, "4_R_Compiled_DrugR_Sequences_AA/aat1_Frac35_All_AA_Aligned.csv", row.names = FALSE)
#write.csv(merged_AA_pfs471_data, "4_R_Compiled_DrugR_Sequences_AA/pfs471_Frac35_All_AA_Aligned.csv", row.names = FALSE)
#write.csv(merged_AA_pfs472_data, "4_R_Compiled_DrugR_Sequences_AA/pfs472_Frac35_All_AA_Aligned.csv", row.names = FALSE)

write.csv(merged_AA_crt_YN, "4_R_Compiled_DrugR_Sequences_AA/crt_Frac35_All_AA_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_AA_dhfr_YN, "4_R_Compiled_DrugR_Sequences_AA/dhfr_Frac35_All_AA_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_AA_dhps1_YN, "4_R_Compiled_DrugR_Sequences_AA/dhps1_Frac35_All_AA_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_AA_dhps2_YN, "4_R_Compiled_DrugR_Sequences_AA/dhps2_Frac35_All_AA_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_AA_K131_YN, "4_R_Compiled_DrugR_Sequences_AA/K131_Frac35_All_AA_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_AA_K132_YN, "4_R_Compiled_DrugR_Sequences_AA/K132_Frac35_All_AA_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_AA_mdr11_YN, "4_R_Compiled_DrugR_Sequences_AA/mdr11_Frac35_All_AA_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_AA_mdr12_YN, "4_R_Compiled_DrugR_Sequences_AA/mdr12_Frac35_All_AA_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_AA_mdr13_YN, "4_R_Compiled_DrugR_Sequences_AA/mdr13_Frac35_All_AA_Aligned_YorN.csv", row.names = FALSE)
#write.csv(merged_AA_aat1_YN, "4_R_Compiled_DrugR_Sequences_AA/aat1_Frac35_All_AA_Aligned_YorN.csv", row.names = FALSE)
#write.csv(merged_AA_pfs471_YN, "4_R_Compiled_DrugR_Sequences_AA/pfs471_Frac35_All_AA_Aligned_YorN.csv", row.names = FALSE)
#write.csv(merged_AA_pfs472_YN, "4_R_Compiled_DrugR_Sequences_AA/pfs472_Frac35_All_AA_Aligned_YorN.csv", row.names = FALSE)

write.csv(merged_AA_crt_Change, "4_R_Compiled_DrugR_Sequences_AA/crt_Frac35_All_AA_Aligned_AAchange.csv", row.names = FALSE)
write.csv(merged_AA_dhfr_Change, "4_R_Compiled_DrugR_Sequences_AA/dhfr_Frac35_All_AA_Aligned_AAchange.csv", row.names = FALSE)
write.csv(merged_AA_dhps1_Change, "4_R_Compiled_DrugR_Sequences_AA/dhps1_Frac35_All_AA_Aligned_AAchange.csv", row.names = FALSE)
write.csv(merged_AA_dhps2_Change, "4_R_Compiled_DrugR_Sequences_AA/dhps2_Frac35_All_AA_Aligned_AAchange.csv", row.names = FALSE)
write.csv(merged_AA_K131_Change, "4_R_Compiled_DrugR_Sequences_AA/K131_Frac35_All_AA_Aligned_AAchange.csv", row.names = FALSE)
write.csv(merged_AA_K132_Change, "4_R_Compiled_DrugR_Sequences_AA/K132_Frac35_All_AA_Aligned_AAchange.csv", row.names = FALSE)
write.csv(merged_AA_mdr11_Change, "4_R_Compiled_DrugR_Sequences_AA/mdr11_Frac35_All_AA_Aligned_AAchange.csv", row.names = FALSE)
write.csv(merged_AA_mdr12_Change, "4_R_Compiled_DrugR_Sequences_AA/mdr12_Frac35_All_AA_Aligned_AAchange.csv", row.names = FALSE)
write.csv(merged_AA_mdr13_Change, "4_R_Compiled_DrugR_Sequences_AA/mdr13_Frac35_All_AA_Aligned_AAchange.csv", row.names = FALSE)
#write.csv(merged_AA_aat1_Change, "4_R_Compiled_DrugR_Sequences_AA/aat1_Frac35_All_AA_Aligned_AAchange.csv", row.names = FALSE)
#write.csv(merged_AA_pfs471_Change, "4_R_Compiled_DrugR_Sequences_AA/pfs471_Frac35_All_AA_Aligned_AAchange.csv", row.names = FALSE)
#write.csv(merged_AA_pfs472_Change, "4_R_Compiled_DrugR_Sequences_AA/pfs472_Frac35_All_AA_Aligned_AAchange.csv", row.names = FALSE)
```

### Frac 5% 
```{r}
# Write files 
write.csv(merged_F5_crt_data, "4_R_Compiled_DrugR_Sequences_NT/crt_Frac5_All_NT_Aligned.csv", row.names = FALSE)
write.csv(merged_F5_dhfr_data, "4_R_Compiled_DrugR_Sequences_NT/dhfr_Frac5_All_NT_Aligned.csv", row.names = FALSE)
write.csv(merged_F5_dhps1_data, "4_R_Compiled_DrugR_Sequences_NT/dhps1_Frac5_All_NT_Aligned.csv", row.names = FALSE)
write.csv(merged_F5_dhps2_data, "4_R_Compiled_DrugR_Sequences_NT/dhps2_Frac5_All_NT_Aligned.csv", row.names = FALSE)
write.csv(merged_F5_K131_data, "4_R_Compiled_DrugR_Sequences_NT/K131_Frac5_All_NT_Aligned.csv", row.names = FALSE)
write.csv(merged_F5_K132_data, "4_R_Compiled_DrugR_Sequences_NT/K132_Frac5_All_NT_Aligned.csv", row.names = FALSE)
write.csv(merged_F5_mdr11_data, "4_R_Compiled_DrugR_Sequences_NT/mdr11_Frac5_All_NT_Aligned.csv", row.names = FALSE)
write.csv(merged_F5_mdr12_data, "4_R_Compiled_DrugR_Sequences_NT/mdr12_Frac5_All_NT_Aligned.csv", row.names = FALSE)
write.csv(merged_F5_mdr13_data, "4_R_Compiled_DrugR_Sequences_NT/mdr13_Frac5_All_NT_Aligned.csv", row.names = FALSE)
#write.csv(merged_F5_aat1_data, "4_R_Compiled_DrugR_Sequences_NT/aat1_Frac5_All_NT_Aligned.csv", row.names = FALSE)
#write.csv(merged_F5_pfs471_data, "4_R_Compiled_DrugR_Sequences_NT/pfs471_Frac5_All_NT_Aligned.csv", row.names = FALSE)
#write.csv(merged_F5_pfs472_data, "4_R_Compiled_DrugR_Sequences_NT/pfs472_Frac5_All_NT_Aligned.csv", row.names = FALSE)

write.csv(merged_F5_crt_YN, "4_R_Compiled_DrugR_Sequences_NT/crt_Frac5_All_NT_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_F5_dhfr_YN, "4_R_Compiled_DrugR_Sequences_NT/dhfr_Frac5_All_NT_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_F5_dhps1_YN, "4_R_Compiled_DrugR_Sequences_NT/dhps1_Frac5_All_NT_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_F5_dhps2_YN, "4_R_Compiled_DrugR_Sequences_NT/dhps2_Frac5_All_NT_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_F5_K131_YN, "4_R_Compiled_DrugR_Sequences_NT/K131_Frac5_All_NT_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_F5_K132_YN, "4_R_Compiled_DrugR_Sequences_NT/K132_Frac5_All_NT_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_F5_mdr11_YN, "4_R_Compiled_DrugR_Sequences_NT/mdr11_Frac5_All_NT_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_F5_mdr12_YN, "4_R_Compiled_DrugR_Sequences_NT/mdr12_Frac5_All_NT_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_F5_mdr13_YN, "4_R_Compiled_DrugR_Sequences_NT/mdr13_Frac5_All_NT_Aligned_YorN.csv", row.names = FALSE)
#write.csv(merged_F5_aat1_YN, "4_R_Compiled_DrugR_Sequences_NT/aat1_Frac5_All_NT_Aligned_YorN.csv", row.names = FALSE)
#write.csv(merged_F5_pfs471_YN, "4_R_Compiled_DrugR_Sequences_NT/pfs471_Frac5_All_NT_Aligned_YorN.csv", row.names = FALSE)
#write.csv(merged_F5_pfs472_YN, "4_R_Compiled_DrugR_Sequences_NT/pfs472_Frac5_All_NT_Aligned_YorN.csv", row.names = FALSE)

write.csv(merged_F5_crt_Change, "4_R_Compiled_DrugR_Sequences_NT/crt_Frac5_All_NT_Aligned_NTchange.csv", row.names = FALSE)
write.csv(merged_F5_dhfr_Change, "4_R_Compiled_DrugR_Sequences_NT/dhfr_Frac5_All_NT_Aligned_NTchange.csv", row.names = FALSE)
write.csv(merged_F5_dhps1_Change, "4_R_Compiled_DrugR_Sequences_NT/dhps1_Frac5_All_NT_Aligned_NTchange.csv", row.names = FALSE)
write.csv(merged_F5_dhps2_Change, "4_R_Compiled_DrugR_Sequences_NT/dhps2_Frac5_All_NT_Aligned_NTchange.csv", row.names = FALSE)
write.csv(merged_F5_K131_Change, "4_R_Compiled_DrugR_Sequences_NT/K131_Frac5_All_NT_Aligned_NTchange.csv", row.names = FALSE)
write.csv(merged_F5_K132_Change, "4_R_Compiled_DrugR_Sequences_NT/K132_Frac5_All_NT_Aligned_NTchange.csv", row.names = FALSE)
write.csv(merged_F5_mdr11_Change, "4_R_Compiled_DrugR_Sequences_NT/mdr11_Frac5_All_NT_Aligned_NTchange.csv", row.names = FALSE)
write.csv(merged_F5_mdr12_Change, "4_R_Compiled_DrugR_Sequences_NT/mdr12_Frac5_All_NT_Aligned_NTchange.csv", row.names = FALSE)
write.csv(merged_F5_mdr13_Change, "4_R_Compiled_DrugR_Sequences_NT/mdr13_Frac5_All_NT_Aligned_NTchange.csv", row.names = FALSE)
#write.csv(merged_F5_aat1_Change, "4_R_Compiled_DrugR_Sequences_NT/aat1_Frac5_All_NT_Aligned_NTchange.csv", row.names = FALSE)
#write.csv(merged_F5_pfs471_Change, "4_R_Compiled_DrugR_Sequences_NT/pfs471_Frac5_All_NT_Aligned_NTchange.csv", row.names = FALSE)
#write.csv(merged_F5_pfs472_Change, "4_R_Compiled_DrugR_Sequences_NT/pfs472_Frac5_All_NT_Aligned_NTchange.csv", row.names = FALSE)

write.csv(merged_F5_AA_crt_data, "4_R_Compiled_DrugR_Sequences_AA/crt_Frac5_All_AA_Aligned.csv", row.names = FALSE)
write.csv(merged_F5_AA_dhfr_data, "4_R_Compiled_DrugR_Sequences_AA/dhfr_Frac5_All_AA_Aligned.csv", row.names = FALSE)
write.csv(merged_F5_AA_dhps1_data, "4_R_Compiled_DrugR_Sequences_AA/dhps1_Frac5_All_AA_Aligned.csv", row.names = FALSE)
write.csv(merged_F5_AA_dhps2_data, "4_R_Compiled_DrugR_Sequences_AA/dhps2_Frac5_All_AA_Aligned.csv", row.names = FALSE)
write.csv(merged_F5_AA_K131_data, "4_R_Compiled_DrugR_Sequences_AA/K131_Frac5_All_AA_Aligned.csv", row.names = FALSE)
write.csv(merged_F5_AA_K132_data, "4_R_Compiled_DrugR_Sequences_AA/K132_Frac5_All_AA_Aligned.csv", row.names = FALSE)
write.csv(merged_F5_AA_mdr11_data, "4_R_Compiled_DrugR_Sequences_AA/mdr11_Frac5_All_AA_Aligned.csv", row.names = FALSE)
write.csv(merged_F5_AA_mdr12_data, "4_R_Compiled_DrugR_Sequences_AA/mdr12_Frac5_All_AA_Aligned.csv", row.names = FALSE)
write.csv(merged_F5_AA_mdr13_data, "4_R_Compiled_DrugR_Sequences_AA/mdr13_Frac5_All_AA_Aligned.csv", row.names = FALSE)
#write.csv(merged_F5_AA_aat1_data, "4_R_Compiled_DrugR_Sequences_AA/aat1_Frac5_All_AA_Aligned.csv", row.names = FALSE)
#write.csv(merged_F5_AA_pfs471_data, "4_R_Compiled_DrugR_Sequences_AA/pfs471_Frac5_All_AA_Aligned.csv", row.names = FALSE)
#write.csv(merged_F5_AA_pfs472_data, "4_R_Compiled_DrugR_Sequences_AA/pfs472_Frac5_All_AA_Aligned.csv", row.names = FALSE)

write.csv(merged_F5_AA_crt_YN, "4_R_Compiled_DrugR_Sequences_AA/crt_Frac5_All_AA_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_F5_AA_dhfr_YN, "4_R_Compiled_DrugR_Sequences_AA/dhfr_Frac5_All_AA_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_F5_AA_dhps1_YN, "4_R_Compiled_DrugR_Sequences_AA/dhps1_Frac5_All_AA_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_F5_AA_dhps2_YN, "4_R_Compiled_DrugR_Sequences_AA/dhps2_Frac5_All_AA_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_F5_AA_K131_YN, "4_R_Compiled_DrugR_Sequences_AA/K131_Frac5_All_AA_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_F5_AA_K132_YN, "4_R_Compiled_DrugR_Sequences_AA/K132_Frac5_All_AA_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_F5_AA_mdr11_YN, "4_R_Compiled_DrugR_Sequences_AA/mdr11_Frac5_All_AA_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_F5_AA_mdr12_YN, "4_R_Compiled_DrugR_Sequences_AA/mdr12_Frac5_All_AA_Aligned_YorN.csv", row.names = FALSE)
write.csv(merged_F5_AA_mdr13_YN, "4_R_Compiled_DrugR_Sequences_AA/mdr13_Frac5_All_AA_Aligned_YorN.csv", row.names = FALSE)
#write.csv(merged_F5_AA_aat1_YN, "4_R_Compiled_DrugR_Sequences_AA/aat1_Frac5_All_AA_Aligned_YorN.csv", row.names = FALSE)
#write.csv(merged_F5_AA_pfs471_YN, "4_R_Compiled_DrugR_Sequences_AA/pfs471_Frac5_All_AA_Aligned_YorN.csv", row.names = FALSE)
#write.csv(merged_F5_AA_pfs472_YN, "4_R_Compiled_DrugR_Sequences_AA/pfs472_Frac5_All_AA_Aligned_YorN.csv", row.names = FALSE)

write.csv(merged_F5_AA_crt_Change, "4_R_Compiled_DrugR_Sequences_AA/crt_Frac5_All_AA_Aligned_AAchange.csv", row.names = FALSE)
write.csv(merged_F5_AA_dhfr_Change, "4_R_Compiled_DrugR_Sequences_AA/dhfr_Frac5_All_AA_Aligned_AAchange.csv", row.names = FALSE)
write.csv(merged_F5_AA_dhps1_Change, "4_R_Compiled_DrugR_Sequences_AA/dhps1_Frac5_All_AA_Aligned_AAchange.csv", row.names = FALSE)
write.csv(merged_F5_AA_dhps2_Change, "4_R_Compiled_DrugR_Sequences_AA/dhps2_Frac5_All_AA_Aligned_AAchange.csv", row.names = FALSE)
write.csv(merged_F5_AA_K131_Change, "4_R_Compiled_DrugR_Sequences_AA/K131_Frac5_All_AA_Aligned_AAchange.csv", row.names = FALSE)
write.csv(merged_F5_AA_K132_Change, "4_R_Compiled_DrugR_Sequences_AA/K132_Frac5_All_AA_Aligned_AAchange.csv", row.names = FALSE)
write.csv(merged_F5_AA_mdr11_Change, "4_R_Compiled_DrugR_Sequences_AA/mdr11_Frac5_All_AA_Aligned_AAchange.csv", row.names = FALSE)
write.csv(merged_F5_AA_mdr12_Change, "4_R_Compiled_DrugR_Sequences_AA/mdr12_Frac5_All_AA_Aligned_AAchange.csv", row.names = FALSE)
write.csv(merged_F5_AA_mdr13_Change, "4_R_Compiled_DrugR_Sequences_AA/mdr13_Frac5_All_AA_Aligned_AAchange.csv", row.names = FALSE)
#write.csv(merged_F5_AA_aat1_Change, "4_R_Compiled_DrugR_Sequences_AA/aat1_Frac5_All_AA_Aligned_AAchange.csv", row.names = FALSE)
#write.csv(merged_F5_AA_pfs471_Change, "4_R_Compiled_DrugR_Sequences_AA/pfs471_Frac5_All_AA_Aligned_AAchange.csv", row.names = FALSE)
#write.csv(merged_F5_AA_pfs472_Change, "4_R_Compiled_DrugR_Sequences_AA/pfs472_Frac5_All_AA_Aligned_AAchange.csv", row.names = FALSE)
```

