---
title: "Data manipulation of MACSE result to Haplotypes and Microhaplotypes"
output: html_notebook
---


```{r Load packages, include=FALSE}
library(stringr)
library(tidyverse)
library(janitor)
library(forcats)
library(data.table)
library(naniar)
library(readr) #necessary for function created (+ dplyr)
```

# Step 1: Process_Seq

**INPUTS**

**Folder**: 4_R_Compiled_DrugR_Sequences_NT
**Files**:
*_All_NT_Aligned.csv                    Use as the sequence file and create microhaplotypes (e.g., CVMNT)

Aims: 
* Remove symbols: `*!-`
* Separate SeqID code into "Isolate", "MID", "MID_N", "Pool", "Date", "Haplotype"
* Create microhaplotype specific to marker 

**OUTPUTS**
**Folder**: 5_R_Final_DrugR_Sequences_NT
  /Controls
  /Isolates
  /Repeats
**Files**:
S7_Controls*_Final.csv
S7_Isolates*_Final.csv
S7_Repeats*_Final.csv

## Input File

```{r}
filenames_NT =          Sys.glob(file.path("4_R_Compiled_DrugR_Sequences_NT/", pattern = "*NT_Aligned.csv"))
filenames_AA =          Sys.glob(file.path("4_R_Compiled_DrugR_Sequences_AA/", pattern = "*AA_Aligned.csv"))
```

## Processing Code 

```{r}
for (filename in filenames_NT) {
  
  INPUT_1_NT = filename

  replace_strs <- c("4_R_Compiled_DrugR_Sequences_NT" = "", ".csv" = "")
  
  OUTPUT_1_NT = paste0("5_R_Final_DrugR_Sequences_NT/Isolates", str_replace_all(filename, replace_strs, ""), "_Isolates_MH.csv")
  OUTPUT_2_NT = paste0("5_R_Final_DrugR_Sequences_NT/Controls", str_replace_all(filename, replace_strs, ""), "_Controls_MH.csv")
  OUTPUT_3_NT = paste0("5_R_Final_DrugR_Sequences_NT/Repeats", str_replace_all(filename, replace_strs, ""), "_Repeats_MH.csv")
  OUTPUT_4_NT = paste0("5_R_Final_DrugR_MH_Info/temp", str_replace_all(filename, replace_strs, ""), "_MH_only.csv")
  
  ######## Part 1: Data transformation ######## 
  
  df <- read.csv(INPUT_1_NT)
  df_1 <- as.data.frame(t(df))
  df_2 <- df_1 %>% row_to_names(row_number = 1) %>% tibble::rownames_to_column("SeqID")

  ######## Part 2: Marker Information ######## 
  
  # Vector of strings to search for
  search_strings <- c("crt", "dhfr", "dhps1", "dhps2", "K131", "K132", "mdr11", "mdr12", "mdr13")

  # Create an empty column called "Marker" in the data frame
  df_3 <- df_2 %>% mutate(FileNames = INPUT_1_NT, Marker = NA)

  # Loop through each search string and update the "Marker" column based on the presence of the string in file names

for (marker in search_strings) {
  df_3$Marker[grepl(marker, df_3$FileNames, ignore.case = TRUE)] <- marker
}

  df_4 <- df_3 %>% filter(!str_detect(SeqID, "Ref")) %>% dplyr::select(-FileNames)
  df_5 <- df_4 %>% 
    separate(SeqID, 
             c("StudyID", "MID", "MID_N", "Pool", "Date", "Haplotype", "Frequency"),  
             sep="[\\._]", extra = "merge", fill = "right", remove = F) %>%
    mutate(SeqID = gsub("X", "", SeqID), 
           StudyID = gsub("X", "", StudyID), 
           Frequency = gsub("f", "", Frequency)) %>% 
    dplyr::select(-c(MID, Pool, Date)) %>%
    mutate_all(~ str_replace_all(., "-", NA_character_)) %>%
    dplyr::select(-where(~ all(is.na(.)))) %>% 
 #   mutate(Reference = ifelse(str_detect(SeqID, "CS7") | str_detect(SeqID, "CS8") | str_detect(SeqID, "CS10"), "Control", 
#                       ifelse(str_detect(SeqID, "RS7") | str_detect(SeqID, "RS8") | str_detect(SeqID, "RS10"), "Repeat", "MRS_Isolate"))) %>% 
    mutate(Reference = ifelse(str_detect(SeqID, "3D7") | str_detect(SeqID, "Dd2") | str_detect(SeqID, "HB3"), "Control", 
                       ifelse(str_detect(SeqID, "RS7"), "Repeat", "MRS_Isolate"))) %>% 
    dplyr::select(SeqID:Frequency, Marker, Reference, everything())

  ######## Part 3: Codon Key: Define codons per marker to then stratify ####### 
df_6 <- df_5 %>% 
    mutate(MH_NT = 
         #    ifelse(Marker=="aat1", paste(`3D7_258`, `3D7_313`, sep = "-"),
             ifelse(Marker=="crt", paste(`3D7_72`, `3D7_73`, `3D7_74`, `3D7_75`, `3D7_76`, sep = "-"),
             ifelse(Marker=="dhfr", paste(`3D7_16`, `3D7_50`, `3D7_51`, `3D7_59`, `3D7_108`, `3D7_164`, sep = "-"),
             ifelse(Marker=="dhps1", paste(`3D7_431`, `3D7_436`, `3D7_437`, sep = "-"),
             ifelse(Marker=="dhps2", paste(`3D7_540`, `3D7_581`, `3D7_613`, sep = "-"),
             ifelse(Marker=="K132", paste(`3D7_441`, `3D7_446`, `3D7_449`, `3D7_454`, `3D7_458`,
                                          `3D7_469`, `3D7_476`, `3D7_481`, `3D7_493`, `3D7_515`, 
                                          `3D7_527`, `3D7_537`, `3D7_538`, `3D7_539`, `3D7_543`, # Change to K13-1 when using for real #
                                          sep = "-"),
             ifelse(Marker=="K131", paste(`3D7_553`, `3D7_561`, `3D7_568`, `3D7_574`,
                                          `3D7_578`, `3D7_580`, `3D7_603`, `3D7_622`,
                                          sep = "-"),
             ifelse(Marker=="mdr11", paste(`3D7_86`, `3D7_184`, sep = "-"),
             ifelse(Marker=="mdr12", paste(`3D7_1034`, `3D7_1042`, sep = "-"),  
             ifelse(Marker=="mdr13", paste(`3D7_1246`, sep = "-"),
       #      ifelse(Marker=="pfs471", paste(`3D7_27`, `3D7_28`, `3D7_49`, `3D7_55`, sep = "-"),       
        #     ifelse(Marker=="pfs472", paste(`3D7_186`, `3D7_188`, `3D7_194`, `3D7_224`, `3D7_236`, `3D7_240`, `3D7_242`, `3D7_248`, `3D7_272`, sep="-"), 
                    Marker)))))))))) %>%
    dplyr::select(SeqID:Reference, MH_NT, everything()) %>%  
    mutate(Haplotype = ifelse(Haplotype==0, "Major", 
                       ifelse(Haplotype==1, "Minor_1", 
                       ifelse(Haplotype==2, "Minor_2", 
                       ifelse(Haplotype==3, "Minor_3",
                       ifelse(Haplotype==4, "Minor_4", 
                       ifelse(Haplotype==5, "Minor_5", Haplotype))))))) %>% 
    arrange(Reference, SeqID)
  
  ######## Part 4: Separate data.frames into "Controls", "Repeats" and "Isolates" 

  df_isolates <- df_6 %>% 
    filter(Reference!="Control") %>% # Remove CONTROL sequences
    filter(Reference!="Repeat") %>%  # Remove REPEATS sequences 
    filter(!str_detect(StudyID, "SY")) %>%  # Remove SYMPTOMATIC sequences 
    separate(StudyID, c("Survey", "MRS", "StudyID"), sep = c(2, 5)) %>% 
  #  mutate(StudyID = as.numeric(StudyID)) %>% 
    dplyr::select(-c(MRS, MID_N)) 
  
  df_controls <- df_6 %>% 
    filter(Reference=="Control") %>% 
    dplyr::select(Control = StudyID, everything()) %>% dplyr::select(-MID_N)
  
  df_repeats <- df_6 %>%
    filter(Reference=="Repeat") %>% 
    separate(StudyID, c("Survey", "MRS", "StudyID"), sep = c(2, 5)) #%>% 
   # mutate(StudyID = as.numeric(StudyID))
  
  df_MH_only <- df_6 %>% dplyr::select(SeqID, Marker, Reference, MH_NT) 
  
  # Write NT data frame 
  
  write.csv(df_isolates, OUTPUT_1_NT, row.names = FALSE)
  write.csv(df_controls, OUTPUT_2_NT, row.names = FALSE)
  write.csv(df_repeats, OUTPUT_3_NT, row.names = FALSE)
  write.csv(df_MH_only, OUTPUT_4_NT, row.names = FALSE)

  for(filename_AA in filenames_AA){
    
  INPUT_1_AA = filename_AA
  
  replace_strs_AA <- c("4_R_Compiled_DrugR_Sequences_AA" = "", ".csv" = "")

  OUTPUT_1_AA = paste0("5_R_Final_DrugR_Sequences_AA/Isolates/", str_replace_all(filename_AA, replace_strs_AA, ""), "_Isolates_MH.csv")
  OUTPUT_2_AA = paste0("5_R_Final_DrugR_Sequences_AA/Controls/", str_replace_all(filename_AA, replace_strs_AA, ""), "_Controls_MH.csv")
  OUTPUT_3_AA = paste0("5_R_Final_DrugR_Sequences_AA/Repeats/", str_replace_all(filename_AA, replace_strs_AA, ""), "_Repeats_MH.csv")
  OUTPUT_4_AA = paste0("5_R_Final_DrugR_MH_Info/temp", str_replace_all(filename_AA, replace_strs_AA, ""), "_MH_only.csv")

  ######## Part 1: Data transformation ######## 
  
  df_AA <- read.csv(INPUT_1_AA)
  df_AA_1 <- as.data.frame(t(df_AA))
  df_AA_2 <- df_AA_1 %>% row_to_names(row_number = 1) %>% tibble::rownames_to_column("SeqID")

  ######## Part 2: Marker Information ######## 
  
  # Vector of strings to search for
  search_strings <- c("crt", "dhfr", "dhps1", "dhps2", "K131", "K132", "mdr11", "mdr12", "mdr13")

  # Create an empty column called "Marker" in the data frame
  df_AA_3 <- df_AA_2 %>% mutate(FileNames = INPUT_1_AA, Marker = NA)

  # Loop through each search string and update the "Marker" column based on the presence of the string in file names

for (marker in search_strings) {
  df_AA_3$Marker[grepl(marker, df_AA_3$FileNames, ignore.case = TRUE)] <- marker
}
  
  df_AA_4 <- df_AA_3 %>% filter(!str_detect(SeqID, "Ref"))
  df_AA_5 <- df_AA_4 %>% 
    separate(SeqID, 
             c("StudyID", "MID", "MID_N", "Pool", "Date", "Haplotype", "Frequency"),  
             sep="[\\._]", extra = "merge", fill = "right", remove = F) %>%
    mutate(SeqID = gsub("X", "", SeqID), 
           StudyID = gsub("X", "", StudyID), 
           Frequency = gsub("f", "", Frequency)) %>% 
    dplyr::select(-c(MID, Pool, Date)) %>%
    mutate_all(~ str_replace_all(., "-", NA_character_)) %>%
    dplyr::select(-where(~ all(is.na(.)))) %>% 
 #   mutate(Reference = ifelse(str_detect(SeqID, "CS7") | str_detect(SeqID, "CS8") | str_detect(SeqID, "CS10"), "Control", 
#                       ifelse(str_detect(SeqID, "RS7") | str_detect(SeqID, "RS8") | str_detect(SeqID, "RS10"), "Repeat", "MRS_Isolate"))) %>% 
    mutate(Reference = ifelse(str_detect(SeqID, "3D7") | str_detect(SeqID, "Dd2") | str_detect(SeqID, "HB3"), "Control", 
                       ifelse(str_detect(SeqID, "RS7"), "Repeat", "MRS_Isolate"))) %>% 
    dplyr::select(SeqID:Frequency, Marker, Reference, everything())
  
  df_AA_6 <- df_AA_5 %>% 
    mutate(MH_AA = 
            # ifelse(Marker=="aat1", paste(`3D7_258`, `3D7_313`, sep=""),
             ifelse(Marker=="crt", paste(`3D7_72`, `3D7_73`, `3D7_74`, `3D7_75`, `3D7_76`, sep=""),
             ifelse(Marker=="dhfr", paste(`3D7_16`, `3D7_50`, `3D7_51`, `3D7_59`, `3D7_108`, `3D7_164`, sep=""),
             ifelse(Marker=="dhps1", paste(`3D7_431`, `3D7_436`, `3D7_437`, sep=""),
             ifelse(Marker=="dhps2", paste(`3D7_540`, `3D7_581`, `3D7_613`, sep=""),
             ifelse(Marker=="K132", paste(`3D7_441`, `3D7_446`, `3D7_449`, `3D7_454`, `3D7_458`, # Change to K13-1 when using for real #
                                          `3D7_469`, `3D7_476`, `3D7_481`, `3D7_493`, `3D7_515`, 
                                          `3D7_527`, `3D7_537`, `3D7_538`, `3D7_539`, `3D7_543`, sep=""),
             ifelse(Marker=="K131", paste(`3D7_553`, `3D7_561`, `3D7_568`, `3D7_574`, # Change to K13-2 when using for real #
                                          `3D7_578`, `3D7_580`, `3D7_603`, `3D7_622`, sep=""),
             ifelse(Marker=="mdr11", paste(`3D7_86`, `3D7_184`, sep=""),
             ifelse(Marker=="mdr12", paste(`3D7_1034`, `3D7_1042`, sep=""),  
             ifelse(Marker=="mdr13", paste(`3D7_1246`),
       #      ifelse(Marker=="pfs471", paste(`3D7_27`, `3D7_28`, `3D7_49`, `3D7_55`, sep=""),       
        #     ifelse(Marker=="pfs472", paste(`3D7_186`, `3D7_188`, `3D7_194`, `3D7_224`, `3D7_236`, `3D7_240`, `3D7_242`, `3D7_248`, `3D7_272`, sep=""), 
                    Marker)))))))))) %>%
    dplyr::select(SeqID:Reference, MH_AA, everything()) %>%  
    mutate(Haplotype = ifelse(Haplotype==0, "Major", 
                       ifelse(Haplotype==1, "Minor_1", 
                       ifelse(Haplotype==2, "Minor_2", 
                       ifelse(Haplotype==3, "Minor_3",
                       ifelse(Haplotype==4, "Minor_4", 
                       ifelse(Haplotype==5, "Minor_5", Haplotype))))))) %>% 
    arrange(Reference, SeqID)
  
  ######## Part 4: Separate data.frames into "Controls", "Repeats" and "Isolates" 

  df_AA_isolates <- df_AA_6 %>% 
    filter(Reference!="Control") %>% # Remove CONTROL sequences
    filter(Reference!="Repeat") %>%  # Remove REPEATS sequences 
    filter(!str_detect(StudyID, "SY")) %>%  # Remove SYMPTOMATIC sequences 
    separate(StudyID, c("Survey", "MRS", "StudyID"), sep = c(2, 5)) %>% 
 #   mutate(StudyID = as.numeric(StudyID)) %>% 
    dplyr::select(-c(MRS, MID_N)) 
  
  df_AA_controls <- df_AA_6 %>% 
    filter(Reference=="Control") %>% 
    dplyr::select(Control = StudyID, everything()) %>% dplyr::select(-MID_N)
  
  df_AA_repeats <- df_AA_6 %>%
    filter(Reference=="Repeat") %>% 
    separate(StudyID, c("Survey", "MRS", "StudyID"), sep = c(2, 5)) #%>% 
   # mutate(StudyID = as.numeric(StudyID))
  
  df_AA_MH_only <- df_AA_6 %>% dplyr::select(SeqID, Marker, Reference, MH_AA) 
  
  # Write AA data frame 
  
  write.csv(df_AA_isolates, OUTPUT_1_AA, row.names = FALSE)
  write.csv(df_AA_controls, OUTPUT_2_AA, row.names = FALSE)
  write.csv(df_AA_repeats, OUTPUT_3_AA, row.names = FALSE)
  write.csv(df_AA_MH_only, OUTPUT_4_AA, row.names = FALSE)
  
  }
}

```

# Step 2: Process_YN

*_All_NT_Aligned_YorN.csv               Use to determine how many changes from 3D7 reference (e.g., 1 change)

## Input File
```{r}
filenames_NT_YN =       Sys.glob(file.path("4_R_Compiled_DrugR_Sequences_NT/", pattern = "*YorN.csv"))
filenames_AA_YN =       Sys.glob(file.path("4_R_Compiled_DrugR_Sequences_AA/", pattern = "*YorN.csv"))
```

## Processing Code 
```{r}
for (filename_NT_YN in filenames_NT_YN) {
  
  INPUT_YN_NT = filename_NT_YN
  
  replace_strs <- c("4_R_Compiled_DrugR_Sequences_NT" = "", ".csv" = "")
  
  OUTPUT_YN_NT = paste0("5_R_Final_DrugR_MH_Info/temp/", str_replace_all(filename_NT_YN, replace_strs, ""), "_Count.csv")
  
  ################### Nucleotide (NT) Data Processing   ################### 
  ######## Part 1: Data transformation ######## 
    
  df_YN <- read.csv(INPUT_YN_NT)
  df_YN_1 <- as.data.frame(t(df_YN))
  df_YN_2 <- df_YN_1 %>% row_to_names(row_number = 1) %>% tibble::rownames_to_column("SeqID") %>% filter(!str_detect(SeqID, "Ref"))

  # Vector of strings to search for
  search_strings <- c("crt", "dhfr", "dhps1", "dhps2", "K131", "K132", "mdr11", "mdr12", "mdr13")

  # Create an empty column called "Marker" in the data frame
  df_YN_3 <- df_YN_2 %>% mutate(FileNames = INPUT_YN_NT, Marker = NA)

  # Loop through each search string and update the "Marker" column based on the presence of the string in file names

for (marker in search_strings) {
  df_YN_3$Marker[grepl(marker, df_YN_3$FileNames, ignore.case = TRUE)] <- marker
}
  df_YN_4 <- df_YN_3 %>% filter(!str_detect(SeqID, "Ref"))
  df_YN_5 <- df_YN_4 %>% 
      mutate(SeqID = gsub("X", "", SeqID)) %>% 
      mutate_all(~ str_replace_all(., "[-!]", NA_character_)) %>%
      dplyr::select(SeqID, Marker, everything())

  df_YN_6 <- df_YN_5 %>% filter(!str_detect(SeqID, "Ref"))
  
  ######## Part 2: Calculate Number of Changes from Reference ####### 

  columns_to_recode <- setdiff(names(df_YN_6), c("SeqID", "Marker"))                                  # Define the columns to be recoded (exclude SeqID & Marker)
  
  df_YN_7 <- df_YN_6 %>% mutate_at(vars(columns_to_recode), ~ ifelse(. == "N", 1, 0))                 # Recode "Y" to 1 and "N" to 0 for the selected column
  df_YN_8 <- df_YN_7 %>% mutate(NT_Diffs = rowSums(select(., all_of(columns_to_recode)), na.rm = T))  # Calculate the total per row and add a new column "NT_Diffs" per SeqID
  df_YN_9 <- df_YN_8 %>% dplyr::select(c(SeqID, Marker, NT_Diffs)) 
  yn_NT_final_df <- df_YN_9 %>% 
    mutate(Reference = ifelse(str_detect(SeqID, "3D7") | str_detect(SeqID, "Dd2") | str_detect(SeqID, "HB3"), "Control", 
                       ifelse(str_detect(SeqID, "RS7"), "Repeat", "MRS_Isolate")))

  remove(columns_to_recode)

  write.csv(yn_NT_final_df, OUTPUT_YN_NT, row.names = FALSE)
  
  for (filename_AA_YN in filenames_AA_YN) {
  
  INPUT_YN_AA = filename_AA_YN
  
  replace_strs_AA <- c("4_R_Compiled_DrugR_Sequences_AA" = "", ".csv" = "")
  
  OUTPUT_YN_AA = paste0("5_R_Final_DrugR_MH_Info/temp/", str_replace_all(filename_AA_YN, replace_strs_AA, ""), "_Count.csv")

  ################### Amino Acid Data Processing   ################### 
  ######## Part 1: Data transformation ######## 
  
  df_YN_AA <- read.csv(INPUT_YN_AA)
  df_YN_AA_1 <- as.data.frame(t(df_YN_AA))
  df_YN_AA_2 <- df_YN_AA_1 %>% row_to_names(row_number = 1) %>% tibble::rownames_to_column("SeqID") %>% filter(!str_detect(SeqID, "Ref"))
  
  # Vector of strings to search for
  search_strings <- c("crt", "dhfr", "dhps1", "dhps2", "K131", "K132", "mdr11", "mdr12", "mdr13")

  # Create an empty column called "Marker" in the data frame
  df_YN_AA_3 <- df_YN_AA_2 %>% mutate(FileNames = INPUT_YN_AA, Marker = NA)

  # Loop through each search string and update the "Marker" column based on the presence of the string in file names

for (marker in search_strings) {
  df_YN_AA_3$Marker[grepl(marker, df_YN_AA_3$FileNames, ignore.case = TRUE)] <- marker
}

  df_YN_AA_4 <- df_YN_AA_3 %>% filter(!str_detect(SeqID, "Ref"))
  df_YN_AA_5 <- df_YN_AA_4 %>% 
      mutate(SeqID = gsub("X", "", SeqID)) %>% 
      mutate_all(~ str_replace_all(., "-", NA_character_)) %>%
      dplyr::select(SeqID,Marker, everything())

  ######## Part 2: Calculate Number of Changes from Reference ####### 

  columns_to_recode <- setdiff(names(df_YN_AA_5), c("SeqID", "Marker"))                                     # Define the columns to be recoded (exclude SeqID & Marker)
  
  df_YN_AA_6 <- df_YN_AA_5 %>% mutate_at(vars(columns_to_recode), ~ ifelse(. == "N", 1, 0))                 # Recode "Y" to 1 and "N" to 0 for the selected column
  df_YN_AA_7 <- df_YN_AA_6 %>% mutate(AA_Diffs = rowSums(select(., all_of(columns_to_recode)), na.rm = T))  # Calculate the total per row and add a new column "AA_Diffs" per SeqID
  df_YN_AA_8 <- df_YN_AA_7 %>% dplyr::select(c(SeqID, Marker, AA_Diffs)) 
  yn_AA_final_df <- df_YN_AA_8 %>% 
    mutate(Reference = ifelse(str_detect(SeqID, "3D7") | str_detect(SeqID, "Dd2") | str_detect(SeqID, "HB3"), "Control", 
                       ifelse(str_detect(SeqID, "RS7"), "Repeat", "MRS_Isolate")))
  
  remove(columns_to_recode)
  
  write.csv(yn_AA_final_df, OUTPUT_YN_AA, row.names = FALSE)
  
  }
  
}

```


# Step 3: Process_Change

*_All_NT_Aligned_NTchange.csv           Use to determine the actual change from 3D7 reference (e.g., 76T)

## Input File
```{r}
filenames_NT_NSYN =     Sys.glob(file.path("4_R_Compiled_DrugR_Sequences_NT/", pattern = "*NTchange.csv"))
filenames_AA_NSYN =     Sys.glob(file.path("4_R_Compiled_DrugR_Sequences_AA/", pattern = "*AAchange.csv"))
```

## Processing Code 

```{r}
for(filename_NT_NSYN in filenames_NT_NSYN){
    
  INPUT_NSYN_NT = filename_NT_NSYN
  
  replace_strs <- c("4_R_Compiled_DrugR_Sequences_NT" = "", ".csv" = "")
  
  OUTPUT_NSYN_NT = paste0("5_R_Final_DrugR_MH_Info/temp/", str_replace_all(filename_NT_NSYN, replace_strs, ""), "_Final.csv")

  ######## Part 4: Data transformation ######## 
  df_nsn <- read.csv(INPUT_NSYN_NT)
  df_nsn_1 <- as.data.frame(t(df_nsn)) 
  df_nsn_2 <- df_nsn_1 %>% row_to_names(row_number = 1) %>% tibble::rownames_to_column("SeqID") %>% filter(!str_detect(SeqID, "Ref"))
  
  # Vector of strings to search for
  search_strings <- c("crt", "dhfr", "dhps1", "dhps2", "K131", "K132", "mdr11", "mdr12", "mdr13")

  # Create an empty column called "Marker" in the data frame
  df_nsn_3 <- df_nsn_2 %>% mutate(FileNames = INPUT_NSYN_NT, Marker = NA)

  # Loop through each search string and update the "Marker" column based on the presence of the string in file names

for (marker in search_strings) {
  df_nsn_3$Marker[grepl(marker, df_nsn_3$FileNames, ignore.case = TRUE)] <- marker
}
  
  df_nsn_4 <- df_nsn_3 %>% 
    mutate(SeqID = gsub("X", "", SeqID)) %>% 
    mutate_all(~ str_replace_all(., "-", NA_character_)) %>%
    dplyr::select(SeqID, Marker, everything()) %>% 
    dplyr::select(-FileNames)
  
  ######## Part 5: Synonymous/Non-Synonymous Mutations ####### 
    
  columns_to_recode_2 <- setdiff(names(df_nsn_4), c("SeqID", "Marker"))                     # Define the columns to be recoded (exclude SeqID & Marker)
  
  df_nsn_5 <- df_nsn_4 %>% 
    unite("NT_Changes", columns_to_recode_2, -c(SeqID), sep = ";", na.rm = T) %>%   # Unite columns together 
    mutate(Reference = ifelse(str_detect(SeqID, "3D7") | str_detect(SeqID, "Dd2") | str_detect(SeqID, "HB3"), "Control", 
                       ifelse(str_detect(SeqID, "RS7"), "Repeat", "MRS_Isolate"))) %>% 
    dplyr::select(SeqID, Marker, Reference, NT_Changes, everything()) %>%  
    arrange(Reference, SeqID) 
    
  nsyn_NT_final_df <- df_nsn_5
  
   # Remove temporary variables
   remove(columns_to_recode_2)
    
  write.csv(nsyn_NT_final_df, OUTPUT_NSYN_NT, row.names = FALSE)
  
  ################### Amino Acid Data Processing   ################### 

for(filename_AA_NSYN in filenames_AA_NSYN){
    
  INPUT_NSYN_AA = filename_AA_NSYN
  
  replace_strs_AA <- c("4_R_Compiled_DrugR_Sequences_AA" = "", ".csv" = "")
  
  OUTPUT_NSYN_AA = paste0("5_R_Final_DrugR_MH_Info/temp/", str_replace_all(filename_AA_NSYN, replace_strs_AA, ""), "_Final.csv")

  ######## Part 4: Data transformation ######## 
  df_nsn_AA <- read.csv(INPUT_NSYN_AA)
  df_nsn_AA_1 <- as.data.frame(t(df_nsn_AA)) 
  df_nsn_AA_2 <- df_nsn_AA_1 %>% row_to_names(row_number = 1) %>% tibble::rownames_to_column("SeqID") %>% filter(!str_detect(SeqID, "Ref"))
   
  # Vector of strings to search for
  search_strings <- c("crt", "dhfr", "dhps1", "dhps2", "K131", "K132", "mdr11", "mdr12", "mdr13")

  # Create an empty column called "Marker" in the data frame
  df_nsn_AA_3 <- df_nsn_AA_2 %>% mutate(FileNames = INPUT_NSYN_AA, Marker = NA)

  # Loop through each search string and update the "Marker" column based on the presence of the string in file names

for (marker in search_strings) {
  df_nsn_AA_3$Marker[grepl(marker, df_nsn_AA_3$FileNames, ignore.case = TRUE)] <- marker
}
  
  df_nsn_AA_4 <- df_nsn_AA_3 %>% 
    mutate(SeqID = gsub("X", "", SeqID)) %>% 
    mutate_all(~ str_replace_all(., "-", NA_character_)) %>%
    dplyr::select(SeqID, Marker, everything()) %>% 
    dplyr::select(-FileNames)

  ######## Part 6: Synonymous/Non-Synonymous Mutations ####### 
    
  columns_to_recode_2 <- setdiff(names(df_nsn_AA_4), c("SeqID", "Marker"))                     # Define the columns to be recoded (exclude SeqID & Marker)
  df_nsn_AA_5 <- df_nsn_AA_4 %>% 
    unite("AA_Changes", columns_to_recode_2, -c(SeqID), sep = ";", na.rm = T) %>%   # Unite columns together 
    mutate(Reference = ifelse(str_detect(SeqID, "3D7") | str_detect(SeqID, "Dd2") | str_detect(SeqID, "HB3"), "Control", 
                       ifelse(str_detect(SeqID, "RS7"), "Repeat", "MRS_Isolate"))) %>% 
    dplyr::select(SeqID, Marker, Reference, AA_Changes, everything()) %>%  
    arrange(Reference, SeqID)
  
  remove(columns_to_recode_2)
  
  nsyn_AA_final_df <- df_nsn_AA_5
  
  write.csv(nsyn_AA_final_df, OUTPUT_NSYN_AA, row.names = FALSE)
  
}

}
```


# Step 4: Join YN & Change & MH

## Function 

Define a function to merge files based on the specified pattern

```{r}
merge_files <- function(directory, pattern, common_columns) {
  
  file_paths <- list.files(path = directory, pattern = pattern, full.names = TRUE)
  merged_data <- read_csv(file_paths[[1]])
  
  # Use a for loop to left join each file to the merged data
  for (i in 2:length(file_paths)) {
    current_file <-  read_csv(file_paths[[i]], col_types = cols(.default = col_character()))

    merged_data <- left_join(merged_data, current_file, by = common_columns)
  }
  
  return(merged_data)
}
```

## Merge 

```{r}
merged_crt_F35_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "crt.*Frac35.*", c("SeqID", "Marker", "Reference"))
merged_dhfr_F35_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "dhfr.*Frac35.*", c("SeqID", "Marker", "Reference"))
merged_dhps1_F35_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "dhps1.*Frac35.*", c("SeqID", "Marker", "Reference"))
merged_dhps2_F35_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "dhps2.*Frac35.*", c("SeqID", "Marker", "Reference"))
merged_K131_F35_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "K131.*Frac35.*", c("SeqID", "Marker", "Reference"))
merged_K132_F35_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "K132.*Frac35.*", c("SeqID", "Marker", "Reference"))
merged_mdr11_F35_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "mdr11.*Frac35.*", c("SeqID", "Marker", "Reference"))
merged_mdr12_F35_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "mdr12.*Frac35.*", c("SeqID", "Marker", "Reference"))
merged_mdr13_F35_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "mdr13.*Frac35.*", c("SeqID", "Marker", "Reference"))
#merged_aat1_F35_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "aat1.*Frac35.*", c("SeqID", "Marker", "Reference"))
#merged_crt_F35_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "pfs471.*Frac35.*", c("SeqID", "Marker", "Reference"))
#merged_crt_F35_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "pfs472.*Frac35.*", c("SeqID", Marker", "Reference"))

merged_crt_F5_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "crt.*Frac5.*", c("SeqID", "Marker", "Reference"))
merged_dhfr_F5_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "dhfr.*Frac5.*", c("SeqID", "Marker", "Reference"))
merged_dhps1_F5_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "dhps1.*Frac5.*", c("SeqID", "Marker", "Reference"))
merged_dhps2_F5_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "dhps2.*Frac5.*", c("SeqID", "Marker", "Reference"))
merged_K131_F5_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "K131.*Frac5.*", c("SeqID", "Marker", "Reference"))
merged_K132_F5_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "K132.*Frac5.*", c("SeqID", "Marker", "Reference"))
merged_mdr11_F5_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "mdr11.*Frac5.*", c("SeqID", "Marker", "Reference"))
merged_mdr12_F5_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "mdr12.*Frac5.*", c("SeqID", "Marker", "Reference"))
merged_mdr13_F5_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "mdr13.*Frac5.*", c("SeqID", "Marker", "Reference"))
#merged_aat1_F5_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "aat1.*Frac.5*", c("SeqID", "Marker", "Reference"))
#merged_crt_F5_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "pfs471.*Frac.5*", c("SeqID", "Marker", "Reference"))
#merged_crt_F5_data <- merge_files("5_R_Final_DrugR_MH_Info/temp", "pfs472.*Frac.5*", c("SeqID", "Marker", "Reference"))
```

## Changes & Isolate Type

### Calculate Synonymous/Non-Synonymous Mutations 

```{r}
###### Function ###### 
syn_nonsyn_diffs <- function(df) {
  
  NT_Diffs = as.numeric(df$NT_Diffs)
  AA_Diffs = as.numeric(df$AA_Diffs)
  df$Synonymous = NT_Diffs - AA_Diffs
  names(df)[names(df) == "NT_Diffs"] <- "Differences"
  names(df)[names(df) == "AA_Diffs"] <- "NonSynonymous"
  
  return(df)
}

###### Apply to data ###### 

syn_nonsyn_crt_F35 <- syn_nonsyn_diffs(merged_crt_F35_data)
syn_nonsyn_dhfr_F35 <- syn_nonsyn_diffs(merged_dhfr_F35_data)
syn_nonsyn_dhps1_F35 <- syn_nonsyn_diffs(merged_dhps1_F35_data)
syn_nonsyn_dhps2_F35 <- syn_nonsyn_diffs(merged_dhps2_F35_data)
syn_nonsyn_K131_F35 <- syn_nonsyn_diffs(merged_K131_F35_data)
syn_nonsyn_K132_F35 <- syn_nonsyn_diffs(merged_K132_F35_data)
syn_nonsyn_mdr11_F35 <- syn_nonsyn_diffs(merged_mdr11_F35_data)
syn_nonsyn_mdr12_F35 <- syn_nonsyn_diffs(merged_mdr12_F35_data)
syn_nonsyn_mdr13_F35 <- syn_nonsyn_diffs(merged_mdr13_F35_data)
#syn_nonsyn_aat1_F35 <- syn_nonsyn_diffs(merged_aat1_F35_data)
#syn_nonsyn_pfs471_F35 <- syn_nonsyn_diffs(merged_pfs471_F35_data)
#syn_nonsyn_pfs472_F35 <- syn_nonsyn_diffs(merged_pfs472_F35_data)

syn_nonsyn_crt_F5 <- syn_nonsyn_diffs(merged_crt_F5_data)
syn_nonsyn_dhfr_F5 <- syn_nonsyn_diffs(merged_dhfr_F5_data)
syn_nonsyn_dhps1_F5 <- syn_nonsyn_diffs(merged_dhps1_F5_data)
syn_nonsyn_dhps2_F5 <- syn_nonsyn_diffs(merged_dhps2_F5_data)
syn_nonsyn_K131_F5 <- syn_nonsyn_diffs(merged_K131_F5_data)
syn_nonsyn_K132_F5 <- syn_nonsyn_diffs(merged_K132_F5_data)
syn_nonsyn_mdr11_F5 <- syn_nonsyn_diffs(merged_mdr11_F5_data)
syn_nonsyn_mdr12_F5 <- syn_nonsyn_diffs(merged_mdr12_F5_data)
syn_nonsyn_mdr13_F5 <- syn_nonsyn_diffs(merged_mdr13_F5_data)
#syn_nonsyn_aat1_F5 <- syn_nonsyn_diffs(merged_aat1_F5_data)
#syn_nonsyn_pfs471_F5 <- syn_nonsyn_diffs(merged_pfs471_F5_data)
#syn_nonsyn_pfs472_F5 <- syn_nonsyn_diffs(merged_pfs472_F5_data)
```

### Separate files based on Isolate Type (MRS_Isolate, Control, Repeat)

```{r}
###### Function ###### 
sep_by_Pf_type <- function(df, filename_1, filename_2, filename_3) {
  
  isolates <- df %>% filter(Reference!="Control") %>% filter(Reference!="Repeat") %>% filter(!str_detect(SeqID, "SY"))
  controls <- df %>% filter(Reference=="Control")
  repeats <- df %>% filter(Reference=="Repeat")
  
  write.csv(isolates, paste0(filename_1, ".csv"), row.names = F)
  write.csv(controls, paste0(filename_2, ".csv"), row.names = F)
  write.csv(repeats, paste0(filename_3, ".csv"), row.names = F)

}

###### Apply to data ###### 
sep_by_Pf_type(syn_nonsyn_crt_F35, 
               "5_R_Final_DrugR_MH_Info/Isolates/S7_crt_Frac35_Isolates_Final_MH", 
               "5_R_Final_DrugR_MH_Info/Controls/S7_crt_Frac35_Controls_Final_MH",
               "5_R_Final_DrugR_MH_Info/Repeats/S7_crt_Frac35_Repeats_Final_MH")
sep_by_Pf_type(syn_nonsyn_dhfr_F35, 
               "5_R_Final_DrugR_MH_Info/Isolates/S7_dhfr_Frac35_Isolates_Final_MH", 
               "5_R_Final_DrugR_MH_Info/Controls/S7_dhfr_Frac35_Controls_Final_MH",
               "5_R_Final_DrugR_MH_Info/Repeats/S7_dhfr_Frac35_Repeats_Final_MH")
sep_by_Pf_type(syn_nonsyn_dhps1_F35, 
               "5_R_Final_DrugR_MH_Info/Isolates/S7_dhps1_Frac35_Isolates_Final_MH", 
               "5_R_Final_DrugR_MH_Info/Controls/S7_dhps1_Frac35_Controls_Final_MH",
               "5_R_Final_DrugR_MH_Info/Repeats/S7_dhps1_Frac35_Repeats_Final_MH")
sep_by_Pf_type(syn_nonsyn_dhps2_F35, 
               "5_R_Final_DrugR_MH_Info/Isolates/S7_dhps2_Frac35_Isolates_Final_MH", 
               "5_R_Final_DrugR_MH_Info/Controls/S7_dhps2_Frac35_Controls_Final_MH",
               "5_R_Final_DrugR_MH_Info/Repeats/S7_dhps2_Frac35_Repeats_Final_MH")
sep_by_Pf_type(syn_nonsyn_K131_F35, 
               "5_R_Final_DrugR_MH_Info/Isolates/S7_K131_Frac35_Isolates_Final_MH", 
               "5_R_Final_DrugR_MH_Info/Controls/S7_K131_Frac35_Controls_Final_MH",
               "5_R_Final_DrugR_MH_Info/Repeats/S7_K131_Frac35_Repeats_Final_MH")
sep_by_Pf_type(syn_nonsyn_K132_F35, 
               "5_R_Final_DrugR_MH_Info/Isolates/S7_K132_Frac35_Isolates_Final_MH", 
               "5_R_Final_DrugR_MH_Info/Controls/S7_K132_Frac35_Controls_Final_MH",
               "5_R_Final_DrugR_MH_Info/Repeats/S7_K132_Frac35_Repeats_Final_MH")
sep_by_Pf_type(syn_nonsyn_mdr11_F35, 
               "5_R_Final_DrugR_MH_Info/Isolates/S7_mdr11_Frac35_Isolates_Final_MH", 
               "5_R_Final_DrugR_MH_Info/Controls/S7_mdr11_Frac35_Controls_Final_MH",
               "5_R_Final_DrugR_MH_Info/Repeats/S7_mdr11_Frac35_Repeats_Final_MH")
sep_by_Pf_type(syn_nonsyn_mdr12_F35, 
               "5_R_Final_DrugR_MH_Info/Isolates/S7_mdr12_Frac35_Isolates_Final_MH", 
               "5_R_Final_DrugR_MH_Info/Controls/S7_mdr12_Frac35_Controls_Final_MH",
               "5_R_Final_DrugR_MH_Info/Repeats/S7_mdr12_Frac35_Repeats_Final_MH")
sep_by_Pf_type(syn_nonsyn_mdr13_F35, 
               "5_R_Final_DrugR_MH_Info/Isolates/S7_mdr13_Frac35_Isolates_Final_MH", 
               "5_R_Final_DrugR_MH_Info/Controls/S7_mdr13_Frac35_Controls_Final_MH",
               "5_R_Final_DrugR_MH_Info/Repeats/S7_mdr13_Frac35_Repeats_Final_MH")
#sep_by_Pf_type(syn_nonsyn_aat1_F35, 
#               "5_R_Final_DrugR_MH_Info/Isolates/S7_aat1_Frac35_Isolates_Final_MH", 
#               "5_R_Final_DrugR_MH_Info/Controls/S7_aat1_Frac35_Controls_Final_MH",
#               "5_R_Final_DrugR_MH_Info/Repeats/S7_aat1_Frac35_Repeats_Final_MH")
#sep_by_Pf_type(syn_nonsyn_pfs471_F35, 
#               "5_R_Final_DrugR_MH_Info/Isolates/S7_pfs471_Frac35_Isolates_Final_MH", 
#               "5_R_Final_DrugR_MH_Info/Controls/S7_pfs471_Frac35_Controls_Final_MH",
#               "5_R_Final_DrugR_MH_Info/Repeats/S7_pfs471_Frac35_Repeats_Final_MH")
#sep_by_Pf_type(syn_nonsyn_pfs472_F35, 
#               "5_R_Final_DrugR_MH_Info/Isolates/S7_pfs472_Frac35_Isolates_Final_MH", 
#               "5_R_Final_DrugR_MH_Info/Controls/S7_pfs472_Frac35_Controls_Final_MH",
#               "5_R_Final_DrugR_MH_Info/Repeats/S7_pfs472_Frac35_Repeats_Final_MH")

sep_by_Pf_type(syn_nonsyn_crt_F5, 
               "5_R_Final_DrugR_MH_Info/Isolates/S7_crt_Frac5_Isolates_Final_MH", 
               "5_R_Final_DrugR_MH_Info/Controls/S7_crt_Frac5_Controls_Final_MH",
               "5_R_Final_DrugR_MH_Info/Repeats/S7_crt_Frac5_Repeats_Final_MH")
sep_by_Pf_type(syn_nonsyn_dhfr_F5, 
               "5_R_Final_DrugR_MH_Info/Isolates/S7_dhfr_Frac5_Isolates_Final_MH", 
               "5_R_Final_DrugR_MH_Info/Controls/S7_dhfr_Frac5_Controls_Final_MH",
               "5_R_Final_DrugR_MH_Info/Repeats/S7_dhfr_Frac5_Repeats_Final_MH")
sep_by_Pf_type(syn_nonsyn_dhps1_F5, 
               "5_R_Final_DrugR_MH_Info/Isolates/S7_dhps1_Frac5_Isolates_Final_MH", 
               "5_R_Final_DrugR_MH_Info/Controls/S7_dhps1_Frac5_Controls_Final_MH",
               "5_R_Final_DrugR_MH_Info/Repeats/S7_dhps1_Frac5_Repeats_Final_MH")
sep_by_Pf_type(syn_nonsyn_dhps2_F5, 
               "5_R_Final_DrugR_MH_Info/Isolates/S7_dhps2_Frac5_Isolates_Final_MH", 
               "5_R_Final_DrugR_MH_Info/Controls/S7_dhps2_Frac5_Controls_Final_MH",
               "5_R_Final_DrugR_MH_Info/Repeats/S7_dhps2_Frac5_Repeats_Final_MH")
sep_by_Pf_type(syn_nonsyn_K131_F5, 
               "5_R_Final_DrugR_MH_Info/Isolates/S7_K131_Frac5_Isolates_Final_MH", 
               "5_R_Final_DrugR_MH_Info/Controls/S7_K131_Frac5_Controls_Final_MH",
               "5_R_Final_DrugR_MH_Info/Repeats/S7_K131_Frac5_Repeats_Final_MH")
sep_by_Pf_type(syn_nonsyn_K132_F5, 
               "5_R_Final_DrugR_MH_Info/Isolates/S7_K132_Frac5_Isolates_Final_MH", 
               "5_R_Final_DrugR_MH_Info/Controls/S7_K132_Frac5_Controls_Final_MH",
               "5_R_Final_DrugR_MH_Info/Repeats/S7_K132_Frac5_Repeats_Final_MH")
sep_by_Pf_type(syn_nonsyn_mdr11_F5, 
               "5_R_Final_DrugR_MH_Info/Isolates/S7_mdr11_Frac5_Isolates_Final_MH", 
               "5_R_Final_DrugR_MH_Info/Controls/S7_mdr11_Frac5_Controls_Final_MH",
               "5_R_Final_DrugR_MH_Info/Repeats/S7_mdr11_Frac5_Repeats_Final_MH")
sep_by_Pf_type(syn_nonsyn_mdr12_F5, 
               "5_R_Final_DrugR_MH_Info/Isolates/S7_mdr12_Frac5_Isolates_Final_MH", 
               "5_R_Final_DrugR_MH_Info/Controls/S7_mdr12_Frac5_Controls_Final_MH",
               "5_R_Final_DrugR_MH_Info/Repeats/S7_mdr12_Frac5_Repeats_Final_MH")
sep_by_Pf_type(syn_nonsyn_mdr13_F5, 
               "5_R_Final_DrugR_MH_Info/Isolates/S7_mdr13_Frac5_Isolates_Final_MH", 
               "5_R_Final_DrugR_MH_Info/Controls/S7_mdr13_Frac5_Controls_Final_MH",
               "5_R_Final_DrugR_MH_Info/Repeats/S7_mdr13_Frac5_Repeats_Final_MH")
#sep_by_Pf_type(syn_nonsyn_aat1_F5, 
#               "5_R_Final_DrugR_MH_Info/Isolates/S7_aat1_Frac5_Isolates_Final_MH", 
#               "5_R_Final_DrugR_MH_Info/Controls/S7_aat1_Frac5_Controls_Final_MH",
#               "5_R_Final_DrugR_MH_Info/Repeats/S7_aat1_Frac5_Repeats_Final_MH")
#sep_by_Pf_type(syn_nonsyn_pfs471_F5, 
#               "5_R_Final_DrugR_MH_Info/Isolates/S7_pfs471_Frac5_Isolates_Final_MH", 
#               "5_R_Final_DrugR_MH_Info/Controls/S7_pfs471_Frac5_Controls_Final_MH",
#               "5_R_Final_DrugR_MH_Info/Repeats/S7_pfs471_Frac5_Repeats_Final_MH")
#sep_by_Pf_type(syn_nonsyn_pfs472_F5, 
#               "5_R_Final_DrugR_MH_Info/Isolates/S7_pfs472_Frac5_Isolates_Final_MH", 
#               "5_R_Final_DrugR_MH_Info/Controls/S7_pfs472_Frac5_Controls_Final_MH",
#               "5_R_Final_DrugR_MH_Info/Repeats/S7_pfs472_Frac5_Repeats_Final_MH")
```
