---
title: "Step_04_Specific_mdr12_dhps2_Cleaning"
format: html
---

Both dhps-2 and mdr1-2 contain isolates that appear to have SAME genotype but are said to be in different clusters. I found that this is because for mdr1-2, either the last one or last six nucleotide differ. For dhps-2 there is a string of A's that are variable. 

- dhps-2: Remove codons 520 and 520.1

| Nucleotides 50 to 75      | n |
| ----------- | ----------- |
| TTAAAAAAAAAAAAAAACAAATTCT      | 2       |
| TTAAAAAAAAAAAAAACAAATTCTA   | 204        |
| TTAAAAAAAAAAAAACAAATTCTAT   | 18        |
| TTAAAAAAAAAAAACAAATTCTATA      | 1790       |
| TTAAAAAAAAAACAAATTCTATAGT   | 964        |
| TTAAAAAAAAAAACAAATTCTATAG   | 1        |

- NOTE: MID 47 (TGTGAGTAGT, [Roche](https://www.yumpu.com/en/document/read/24794008/roche-mid-adapters)) exists within the dhps-2 amplicon sequence. Our current SeekDeep parameters account for this by only searching for the MID in the first 8 nucleotides of the amplicon.  

- mdr1-2: Remove codons 1077, 1078, 1079

| Last 6 nucleotides      | n |
| ----------- | ----------- |
| AAAATG      | 1088       |
| AAAATT   | 981        |
| TTTCTG   | 550        |

- This R Notebook contains the re-formatting of mdr1-2 and dhps-2. 

```{r}
library(tidyverse)
library(janitor)
```

# Load Data

Load your final dhps2 and mdr12 haplotype ("_Isolates_Final_Haps.csv"), amino acid sequence ("_final_AA_Aligned_Isolates_Final_Haps.csv") and nucleotide sequence ("_final_NT_Aligned_Isolates_Final_Haps.csv") files as below: 

```{r}
######## dhps-2 ######### 
dhps2_isolates_Haps_final   <- read.csv("05_All_DrugR_Haplotypes/Isolates/Surveys_dhps2_Isolates_Final_Haps.csv")
dhps2_isolates_seq_AA_final <- read.csv("05_All_DrugR_AA_Sequences/Isolates/Surveys_dhps2_final_AA_Aligned_Isolates_Final_Haps.csv", colClasses = "character")
dhps2_isolates_seq_NT_final <- read.csv("05_All_DrugR_NT_Sequences/Isolates/Surveys_dhps2_final_NT_Aligned_Isolates_Final_Haps.csv", colClasses = "character")
######### mdr1-2 ######### 
mdr12_isolates_Haps_final   <- read.csv("05_All_DrugR_Haplotypes/Isolates/Surveys_mdr12_Isolates_Final_Haps.csv")
mdr12_isolates_seq_AA_final <- read.csv("05_All_DrugR_AA_Sequences/Isolates/Surveys_mdr12_final_AA_Aligned_Isolates_Final_Haps.csv", colClasses = "character")
mdr12_isolates_seq_NT_final <- read.csv("05_All_DrugR_NT_Sequences/Isolates/Surveys_mdr12_final_NT_Aligned_Isolates_Final_Haps.csv", colClasses = "character")
```

# dhps-2: Codons 520 and 520.1

- Here we are going to remove codons 520 and 520.1 by `  select(-c(X3D7_520, X3D7_520.1)) %>% `
- And recalculate the frequency and read count of the new list of codons: ` %>%  summarise(Freq_new = sum(Frequency), ReadCnt_new = sum(c_ReadCnt)) %>% mutate(Freq_new = ifelse(Freq_new==0.9999997, 1, Freq_new))`

```{r}
new_freqs_seq_NT <- dhps2_isolates_seq_NT_final %>% 
  select(-c(X3D7_520, X3D7_520.1)) %>% 
  mutate(Frequency = as.numeric(Frequency), c_ReadCnt = as.numeric(c_ReadCnt)) %>%
  unite("Sequence", X3D7_501:X3D7_654, sep = "") %>% 
  group_by(StudyID, Survey, Marker, Sequence) %>% 
  summarise(Freq_new = sum(Frequency), ReadCnt_new = sum(c_ReadCnt)) %>%
  mutate(Freq_new = ifelse(Freq_new==0.9999997, 1, Freq_new))

dhps2_isolates_seq_NT_CLEANED <- dhps2_isolates_seq_NT_final %>% 
  select(-c(X3D7_520, X3D7_520.1)) %>% 
  mutate(Frequency = as.numeric(Frequency), c_ReadCnt = as.numeric(c_ReadCnt)) %>%
  unite("Sequence", X3D7_501:X3D7_654, sep = "") %>% 
  distinct(StudyID, Sequence, .keep_all = T) %>% select(-c(Frequency, c_ReadCnt)) %>%
  left_join(new_freqs_seq_NT, by = c("StudyID", "Survey", "Marker", "Sequence")) %>% 
  left_join(dhps2_isolates_seq_NT_final, by = c("SeqID", "StudyID", "Haplotype","Survey", "Marker", "Reference", "Haps_NT")) %>% 
  select(-c(Sequence, Frequency, c_ReadCnt, X3D7_520.1)) %>% 
  rename(Frequency = Freq_new, c_ReadCnt = ReadCnt_new) %>%
  select(SeqID, StudyID, Haplotype, Frequency, c_ReadCnt, everything()) 

dhps2_isolates_CLEANED_data <- dhps2_isolates_seq_NT_CLEANED %>% select(-c(X3D7_501:X3D7_654, Haps_NT))

dhps2_isolates_seq_AA_CLEANED <- dhps2_isolates_seq_AA_final %>% 
  select(-c(Frequency, c_ReadCnt)) %>%
  right_join(dhps2_isolates_CLEANED_data, by = c("SeqID", "StudyID", "Haplotype", "Survey", "Marker", "Reference")) %>%
  select(SeqID, StudyID, Haplotype, Frequency, c_ReadCnt, everything()) 
    
dhps2_isolates_Haps_final_CLEANED <- dhps2_isolates_Haps_final %>% 
  select(-c(Frequency, c_ReadCnt, Haplotype)) %>%
  right_join(dhps2_isolates_CLEANED_data, by = c("SeqID", "StudyID", "Survey", "Marker", "Reference")) %>%
  select(SeqID, StudyID, Haplotype, Frequency, c_ReadCnt, everything())
```

# mdr1-2: Remove X3D7_1077, X3D7_1078, X3D7_1079

- Here we are going to remove codons 1077, 1078 and 1079 and recalculate the frequency and read count of the new list of codons: ` %>% mutate(Frequency = as.numeric(Frequency), c_ReadCnt = as.numeric(c_ReadCnt)) %>% unite("Sequence", X3D7_939:X3D7_1076, sep = "")`

```{r}
mdr12_new_freqs_seq_NT <- mdr12_isolates_seq_NT_final %>% 
  mutate(Frequency = as.numeric(Frequency), c_ReadCnt = as.numeric(c_ReadCnt)) %>%
  unite("Sequence", X3D7_939:X3D7_1076, sep = "") %>% 
  group_by(StudyID, Survey, Marker, Sequence) %>% 
  summarise(Freq_new = sum(Frequency), ReadCnt_new = sum(c_ReadCnt)) %>%
  mutate(Freq_new = ifelse(Freq_new>0.99999, 1, Freq_new))

mdr12_isolates_seq_NT_CLEANED <- mdr12_isolates_seq_NT_final %>% 
  mutate(Frequency = as.numeric(Frequency), c_ReadCnt = as.numeric(c_ReadCnt)) %>%
  unite("Sequence", X3D7_939:X3D7_1076, sep = "") %>% 
  distinct(StudyID, Sequence, .keep_all = T) %>% select(-c(Frequency, c_ReadCnt)) %>%
  left_join(mdr12_new_freqs_seq_NT, by = c("StudyID", "Survey", "Marker", "Sequence")) %>% 
  left_join(mdr12_isolates_seq_NT_final, by = c("SeqID", "StudyID", "Haplotype","Survey", "Marker", "Reference", "Haps_NT")) %>% 
  select(-c(Sequence, Frequency, c_ReadCnt)) %>% 
  rename(Frequency = Freq_new, c_ReadCnt = ReadCnt_new) %>%
  select(SeqID, StudyID, Haplotype, Frequency, c_ReadCnt, everything()) 

mdr12_isolates_CLEANED_data <- mdr12_isolates_seq_NT_CLEANED %>% select(-c(X3D7_939:X3D7_1076, Haps_NT))

mdr12_isolates_seq_AA_CLEANED <- mdr12_isolates_seq_AA_final %>% 
  select(-c(Frequency, c_ReadCnt)) %>%
  right_join(mdr12_isolates_CLEANED_data, by = c("SeqID", "StudyID", "Haplotype", "Survey", "Marker", "Reference")) %>%
  select(SeqID, StudyID, Haplotype, Frequency, c_ReadCnt, everything()) 
    
mdr12_isolates_Haps_final_CLEANED <- mdr12_isolates_Haps_final %>% 
  select(-c(Frequency, c_ReadCnt, Haplotype)) %>%
  right_join(mdr12_isolates_CLEANED_data, by = c("SeqID", "StudyID", "Survey", "Marker", "Reference")) %>%
  select(SeqID, StudyID, Haplotype, Frequency, c_ReadCnt, everything())
```

# Write new files

Now write these new files and continue with your R journey! 

```{r}
write.csv(dhps2_isolates_seq_NT_CLEANED, "05_All_DrugR_NT_Sequences/Isolates/Surveys_dhps2_final_NT_Aligned_Isolates_Final_Adjusted_Haps.csv", row.names = F)
write.csv(dhps2_isolates_seq_AA_CLEANED, "05_All_DrugR_AA_Sequences/Isolates/Surveys_dhps2_final_AA_Aligned_Isolates_Final_Adjusted_Haps.csv", row.names = F)
write.csv(dhps2_isolates_Haps_final_CLEANED, "05_All_DrugR_Haplotypes/Isolates/Surveys_dhps2_Isolates_Final_Adjusted_Haps.csv", row.names = F)

write.csv(dhps2_repeats_seq_NT_CLEANED, "05_All_DrugR_NT_Sequences/Repeats/Surveys_dhps2_final_NT_Aligned_Repeats_Final_Adjusted_Haps.csv", row.names = F)
write.csv(dhps2_repeats_seq_AA_CLEANED, "05_All_DrugR_AA_Sequences/Repeats/Surveys_dhps2_final_AA_Aligned_Repeats_Final_Adjusted_Haps.csv", row.names = F)
write.csv(dhps2_repeats_Haps_final_CLEANED, "05_All_DrugR_Haplotypes/Repeats/Surveys_dhps2_Repeats_Final_Adjusted_Haps.csv", row.names = F)
```
