---
title: "Step_02_MACSE_to_R"
---

This `.qmd` Notebook contains the post-processing of MACSE data (.fasta) into a data frame (.csv) for manipulation in R.

```{r Load packages, include=F}
library(tidyverse)
library(janitor)
library(data.table)
library(seqinr)
library(Biostrings)
```

# MACSE Processing

First we wish to process the MACSE output data with the suffix "_NT.fasta" and "_AA.fasta". This involves: 

1. Aligning to the reference genome e.g., 3D7, Dd2 and HB3 to gain the **position** of each codon. 
2. For **nucleotide** processing: Ensure columns are created for each codon: each 3 characters, i.e., nucleotides. 
3. For **amino acid** processing: Ensure columns are created for each codon: each 1 character, i.e., amino acid. 

## Nucleotide (NT) Processing

INPUTS: 

- "_NT.fasta" files: MACSE output 

OUTPUTS: 

- "_NT_Aligned.csv": Data frame of aligned sequences to the 3D7 reference genome where each codon represented by a unique column. 
- "_NT_Aligned_YorN.csv": Data frame of whether the codon MATCHES YES (Y) or NO (N) the 3D7 reference. 
- "_NT_Aligned_NTchange.csv": Data frame of when codons DID NOT MATCH 3D7 reference and what the nature of the change is.

What YOU need to do:

1. Enter the FOLDER NAME in `line 28` where you have stored all "_NT.fasta" MACSE output files. 
2. Enter the SAME FOLDER NAME in `line 37` where have stored all "_NT.fasta" MACSE output files. 
3. Enter a NEW FOLDER NAME in `line 40`, `line 41` and `line 42` where you would like to SAVE these processed files. 
4. Run Chunk! 


```{r}
filenames = Sys.glob(file.path("02_MACSE_Output_NT", pattern = "*.fasta")) # REPLACE "02_MACSE_Output_NT" WITH YOUR FOLDER NAME. 

for (filename in filenames) {
  
  INPUT_NT = filename
  
  replace_strs <- c("02_MACSE_Output_NT" = "", ".fasta" = "") # REPLACE "02_MACSE_Output_NT" WITH YOUR FOLDER NAME. 

  OUTPUT_1_NT = paste0("03_DrugR_NT_Sequences", str_replace_all(filename, replace_strs, ""), "_Aligned.csv") # REPLACE "03_DrugR_NT_Sequences" WITH YOUR FOLDER NAME 
  OUTPUT_2_NT = paste0("03_DrugR_NT_Changes", str_replace_all(filename, replace_strs, ""), "_Aligned_YorN.csv") # REPLACE "03_DrugR_NT_Changes" WITH YOUR FOLDER NAME
  OUTPUT_3_NT = paste0("03_DrugR_NT_Changes", str_replace_all(filename, replace_strs, ""), "_Aligned_NTchange.csv") # REPLACE "03_DrugR_NT_Changes" WITH YOUR FOLDER NAME 
  
  ########## PART 1: General processing of input file  ########## 
  
  fastaFile <- readBStringSet(INPUT_NT)

  # Data manipulation 
    seq_name = names(fastaFile)
    sequence = paste(fastaFile)
    df_NT <- data.frame(seq_name, sequence)

  # Rename reference file
    df_NT1 <- df_NT %>% mutate(seq_name = ifelse(str_detect(seq_name, "Plasmodium falciparum 3D7"), "3D7_Ref",
                                          ifelse(str_detect(seq_name, "Plasmodium falciparum Dd2"), "Dd2_Ref",
                                          ifelse(str_detect(seq_name, "Plasmodium falciparum HB3"), "HB3_Ref", 
                                          df_NT$seq_name))))

    # Separate by codon
    n <- ((max(nchar(df_NT1$sequence)) - 1) %/% 3) + 1
    df_NT2 <- df_NT1 %>% separate(sequence, into=paste0("NT", seq(n)), sep=seq(n-1)*3, remove = F)

    # Data Manipulation 2
    df_NT3 <- df_NT2 %>% dplyr::select(-sequence) %>% mutate(seq_name = gsub("-", "\\.",seq_name)) %>% mutate_all(~ str_replace_all(., "!", NA_character_)) 
    df_NT4 <- df_NT3 %>% separate(seq_name, c("Isolate", "MID", "Pool", "Date", "Haplotype"), sep="\\.", extra = "merge", fill = "right") %>% arrange(Isolate)

    # Transform and put 3D7_Ref to first column
    df_NT5 <- as.data.frame(t(df_NT3))
    df_NT6 <- df_NT5 %>% row_to_names(1) %>% dplyr::select(`3D7_Ref`, everything())

  ########## PART 2: Relabel codons based on reference 3D7 positions  ########## 
  # - Loop through rows of transposed df
  # - If find != “---”, label and increment position by 1
  # - If find “---”, label as something else to indicate alignment positions w/o ref in 3D7

    # Add initial NT value as column for unique values for merging
    df_NT6 <- df_NT6 %>% mutate(rownames=row.names(.))

    # Rearrange columns
    df_NT6 <- df_NT6 %>% select("rownames", "3D7_Ref", everything())

    # Initialise first positions and df object
    pos <- 0
    tmp <- 0
    new_df <- data.frame()

    # Loop every row to relabel 3D7 positions
    
for (row in 1:nrow(df_NT6)) {
  # get 3D7 codon triplet
  codon <- df_NT6[row, "3D7_Ref"]
  row_name <- row.names(df_NT6[row, ])
  # if 3D7 codon triplet is valid, relabel and increment count
  # if 3D7 codon triplet is invalid (i.e., "---"), relabel with appended temp value
  if (codon != "---") {
    pos <- pos + 1  # increments pos by 1 if codon is valid
    tmp <- 0  # resets tmp to 1 if the codon is valid
    label <- paste0("3D7_", pos)
  } else {
    tmp <- tmp + 1  # increments tmp by 1 if codon is invalid
    label <- paste0("3D7_", pos, ".", tmp)
  }
  new_df <- rbind(new_df, cbind(row_name, codon, label))
}

    # Rename new_df columns
    colnames(new_df) <- c("row_name", "codon", "3D7_Pos")

    # Join new labels with previous df with data, remove rownames column
    df_NT6 <- left_join(df_NT6, new_df, by=c("rownames" = "row_name", "3D7_Ref" = "codon"))
    df_NT6 <- df_NT6 %>% select("3D7_Pos", "3D7_Ref", everything()) %>% select(-c(rownames))

    # Remove temporary variables
    remove(row, pos, tmp, new_df, label)

    # Write data frame
    write.csv(df_NT6, OUTPUT_1_NT, row.names = F)

  ########## PART 3: Compare isolates to 3D7 and generate multiple output files  ########## 
  # - Output 1: “Y/N” to refer to codons compared to ref
  # - Output 2: Nucleotide changes (e.g., ATC -> ATG): DA changed " > ", to substring(df_NT6$`3D7_Pos`, 5),

    df_NT7 <- df_NT6 # Y/N output
    df_NT8 <- df_NT6 # NT change output

    # Save list of isolates
    isolatelist <- df_NT6 %>% colnames() %>% data.frame() %>% filter(.!=c("3D7_Pos", "3D7_Ref"))

    # Loop through list of isolates (i.e., columns)
for (row in 1:nrow(isolatelist)) {
  
  # Get isolate ID
  id <- isolatelist[row, "."]
  
  ### Compare isolate codon triplets to 3D7_Ref codon triplets
  
  # "Y": same codons, "N": different codons, "---": gap in alignment
  # codon triplets are kept if 3D7_Ref is "---"
  df_NT7[[id]] <- ifelse(df_NT6[[id]] == "---", "-",
                         ifelse(df_NT6$`3D7_Ref` == "---", df_NT6[[id]],
                                ifelse(df_NT6[[id]] == df_NT6$`3D7_Ref`, "Y", "N")))
  
  # "NA": same codons, "xxx > xxx": different codons, "-": gap in alignment
  # codon triplets are kept if 3D7_Ref is "---"
  df_NT8[[id]] <- ifelse(df_NT6[[id]] == "---", "-",
                         ifelse(df_NT6$`3D7_Ref` == "---", paste0("- > ", df_NT6[[id]]),
                                ifelse(df_NT6[[id]] == df_NT6$`3D7_Ref`, NA, paste0(df_NT6$`3D7_Ref`, substring(df_NT6$`3D7_Pos`, 5), df_NT6[[id]]))))
}
    
    # Remove temporary variables
    remove(row, id)

    # Write data frame
    write.csv(df_NT7, OUTPUT_2_NT, row.names = F)
    write.csv(df_NT8, OUTPUT_3_NT, row.names = F)

} 

```

## Amino Acid (AA) Processing

INPUTS: 

- "_AA.fasta" files: MACSE output 

OUTPUTS: 

- "_AA_Aligned.csv": Data frame of aligned sequences to the 3D7 reference genome where each codon represented by a unique column. 
- "_AA_Aligned_YorN.csv": Data frame of whether the codon MATCHES YES (Y) or NO (N) the 3D7 reference. 
- "_AA_Aligned_NTchange.csv": Data frame of when codons DID NOT MATCH 3D7 reference and what the nature of the change is.


What YOU need to do:

1. Enter the FOLDER NAME in `line 173` where you have stored all "_AA.fasta" MACSE output files. 
2. Enter the SAME FOLDER NAME in `line 179` where have stored all "_AA.fasta" MACSE output files. 
3. Enter a NEW FOLDER NAME in `line 181`, `line 182` and `line 183` where you would like to SAVE these processed files. 
4. Run Chunk! 

```{r}
filenames_AA = Sys.glob(file.path("02_MACSE_Output_AA", pattern = "*.fasta")) # REPLACE "02_MACSE_Output_AA" WITH YOUR FOLDER NAME. 

for (filename in filenames_AA) {
  
  INPUT_AA = filename
  
  replace_strs <- c("02_MACSE_Output_AA" = "", ".fasta" = "") # REPLACE "02_MACSE_Output_AA" WITH YOUR FOLDER NAME. 

  OUTPUT_1_AA = paste0("03_DrugR_AA_Sequences", str_replace_all(filename, replace_strs, ""), "_Aligned.csv")# REPLACE "03_DrugR_AA_Sequences" WITH YOUR FOLDER NAME
  OUTPUT_2_AA = paste0("03_DrugR_AA_Changes", str_replace_all(filename, replace_strs, ""), "_Aligned_YorN.csv")# REPLACE "03_DrugR_AA_Sequences" WITH YOUR FOLDER NAME
  OUTPUT_3_AA = paste0("03_DrugR_AA_Changes", str_replace_all(filename, replace_strs, ""), "_Aligned_AAchange.csv")# REPLACE "03_DrugR_AA_Sequences" WITH YOUR FOLDER NAME

  ########## PART 1: General processing of input file  ########## 
  
    fastaFile <- readBStringSet(INPUT_AA)

    # Data manipulation 
    seq_name = names(fastaFile)
    sequence = paste(fastaFile)
    df_AA <- data.frame(seq_name, sequence)

    # Rename reference file
    df_AA1 <- df_AA %>% mutate(seq_name = ifelse(str_detect(seq_name, "Plasmodium falciparum 3D7"), "3D7_Ref",
                                          ifelse(str_detect(seq_name, "Plasmodium falciparum Dd2"), "Dd2_Ref",
                                          ifelse(str_detect(seq_name, "Plasmodium falciparum HB3"), "HB3_Ref", 
                                          df_AA$seq_name))))
    # Separate by codon
    df_AA2 <- df_AA1 %>% data.frame(do.call("rbind", strsplit(as.character(df_AA$sequence), "", fixed = TRUE)), stringsAsFactors = F)

    # Data Manipulation 2
    df_AA3 <- df_AA2 %>% dplyr::select(-sequence) %>% mutate(seq_name = gsub("-", "\\.",seq_name)) %>% mutate_all(~ str_replace_all(., "!", NA_character_)) 
    df_AA4 <- df_AA3 %>% separate(seq_name, c("Isolate", "MID", "Pool", "Date", "Haplotype"), 
                              sep="\\.", extra = "merge", fill = "right") %>% arrange(Isolate)

    # Transform and put 3D7_Ref to first column
    df_AA5 <- as.data.frame(t(df_AA3))
    df_AA6 <- df_AA5 %>% row_to_names(1) %>% dplyr::select(`3D7_Ref`, everything())

      ########## PART 2: Relabel codons based on reference 3D7 positions  ########## 
      # - Loop through rows of transposed df
      # - If find != “---”, label and increment position by 1
      # - If find “---”, label as something else to indicate alignment positions w/o ref in 3D7
    
    # Add initial AA value as column for unique values for merging
    df_AA6 <- df_AA6 %>% mutate(rownames=row.names(.))

    # Rearrange columns
    df_AA6 <- df_AA6 %>% select("rownames", "3D7_Ref", everything())

    # Initialise first positions and df object
    pos <- 0
    tmp <- 0
    new_df <- data.frame()

# Loop every row to relabel 3D7 positions
for (row in 1:nrow(df_AA6)) {
  # get 3D7 codon triplet
  codon <- df_AA6[row, "3D7_Ref"]
  row_name <- row.names(df_AA6[row, ])
  # if 3D7 codon triplet is valid, relabel and increment count
  # if 3D7 codon triplet is invalid (i.e., "-"), relabel with appended temp value ## DCA CHANGED "---" to "-"
  if (codon != "-") { ## DCA CHANGED "---" to "-"
    pos <- pos + 1  # increments pos by 1 if codon is valid
    tmp <- 0  # resets tmp to 1 if the codon is valid
    label <- paste0("3D7_", pos)
  } else {
    tmp <- tmp + 1  # increments tmp by 1 if codon is invalid
    label <- paste0("3D7_", pos, ".", tmp)
  }
  new_df <- rbind(new_df, cbind(row_name, codon, label))
}

    # Rename new_df columns
    colnames(new_df) <- c("row_name", "codon", "3D7_Pos")

    # Join new labels with previous df with data, remove rownames column
    df_AA6 <- left_join(df_AA6, new_df, by=c("rownames" = "row_name", "3D7_Ref" = "codon"))
    df_AA6 <- df_AA6 %>% select("3D7_Pos", "3D7_Ref", everything()) %>% select(-c(rownames))

    # Remove temporary variables
    remove(row, pos, tmp, new_df, label)

    # Write data frame
    write.csv(df_AA6, OUTPUT_1_AA, row.names = F)

    ########## PART 3: Compare isolates to 3D7 and generate multiple output files  ########## 
    # - Output 1: “Y/N” to refer to codons compared to ref
    # - Output 2: Nucleotide changes (e.g., ATC -> ATG): DA changed " > ", to substring(df_AA6$`3D7_Pos`, 5),
    df_AA7 <- df_AA6 # Y/N output
    df_AA8 <- df_AA6 # AA change output

    # Save list of isolates
    isolatelist <- df_AA6 %>% colnames() %>% data.frame() %>% filter(.!=c("3D7_Pos", "3D7_Ref"))

    # Loop through list of isolates (i.e., columns)
    for (row in 1:nrow(isolatelist)) {
  
      # Get isolate ID
      id <- isolatelist[row, "."]
  
      ### Compare isolate codon triplets to 3D7_Ref codon triplets
  
      # "Y": same codons, "N": different codons, "---": gap in alignment
      # codon triplets are kept if 3D7_Ref is "---"; ## DCA CHANGED "---" to "-"
      df_AA7[[id]] <- ifelse(df_AA6[[id]] == "-", "-", ## DCA CHANGED "---" to "-"
                         ifelse(df_AA6$`3D7_Ref` == "-", df_AA6[[id]], ## DCA CHANGED "---" to "-"
                                ifelse(df_AA6[[id]] == df_AA6$`3D7_Ref`, "Y", "N")))

      # "NA": same codons, "xxx > xxx": different codons, "-": gap in alignment ## This is where I think we need to modify! 
      # codon triplets are kept if 3D7_Ref is "---"
      df_AA8[[id]] <- ifelse(df_AA6[[id]] == "-", "-",
                         ifelse(df_AA6$`3D7_Ref` == "-", paste0("- > ", df_AA6[[id]]),
                                ifelse(df_AA6[[id]] == df_AA6$`3D7_Ref`, NA, paste0(df_AA6$`3D7_Ref`, substring(df_AA6$`3D7_Pos`, 5), df_AA6[[id]]))))  
}

      # Remove temporary variables
    remove(row, id)

    # Write data frame
    write.csv(df_AA7, OUTPUT_2_AA, row.names = F)
    write.csv(df_AA8, OUTPUT_3_AA, row.names = F)
} 
```

# Join by Marker

After we have processed each .fasta file to a .csv file, we can (1, 2) merge all .csv files per marker for NT (nucleotide) sequences and AA (amino acid) sequences and (3) write the .csv file for these merged files. 

## 1. Load Function to Merge 

```{r}
merge_files <- function(directory, pattern, common_columns) {
  file_paths <- list.files(path = directory, pattern = pattern, full.names = TRUE)
  merged_data <- read_csv(file_paths[[1]])
  for (i in 2:length(file_paths)) {
    current_file <-  read_csv(file_paths[[i]], col_types = cols(.default = col_character()))
    merged_data <- left_join(merged_data, current_file, by = common_columns)
  }
  return(merged_data)
} 
```

## 2. Merge Together 

- Note: REPLACE "03_DrugR_NT_Sequences" with YOUR OUTPUT FOLDER for "_NT_Aligned.csv" files in `lines 324-336`
- Note: REPLACE "03_DrugR_NT_Changes" with YOUR OUTPUT FOLDER for "_NT_Aligned_YorN.csv"  files in `lines 338-349`
- Note: REPLACE "03_DrugR_NT_Changes" with YOUR OUTPUT FOLDER for "_NT_Aligned_NTchange.csv" files in `lines 351-362`

- Note: REPLACE "03_DrugR_AA_Sequences" with YOUR OUTPUT FOLDER for "_NT_Aligned.csv" files in `lines 367-378`
- Note: REPLACE "03_DrugR_AA_Changes" with YOUR OUTPUT FOLDER for "_NT_Aligned_YorN.csv"  files in `lines 380-391`
- Note: REPLACE "03_DrugR_AA_Changes" with YOUR OUTPUT FOLDER for "_NT_Aligned_NTchange.csv" files in `lines 393-404`

```{r}
# Merge files with pattern "[marker]" and common columns "3D7_Pos" and "3D7_Ref"

Survey_merged_crt_data <- merge_files("03_DrugR_NT_Sequences", "Survey.*crt_final_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_dhfr_data <- merge_files("03_DrugR_NT_Sequences", "Survey.*dhfr_final_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_dhps1_data <- merge_files("03_DrugR_NT_Sequences", "Survey.*dhps1_final_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_dhps2_data <- merge_files("03_DrugR_NT_Sequences", "Survey.*dhps2_final_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_K131_data <- merge_files("03_DrugR_NT_Sequences", "Survey.*K131_final_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_K132_data <- merge_files("03_DrugR_NT_Sequences", "Survey.*K132_final_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_mdr11_data <- merge_files("03_DrugR_NT_Sequences", "Survey.*mdr11_final_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_mdr12_data <- merge_files("03_DrugR_NT_Sequences", "Survey.*mdr12_final_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_mdr13_data <- merge_files("03_DrugR_NT_Sequences", "Survey.*mdr13_final_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_aat1_data <- merge_files("03_DrugR_NT_Sequences", "Survey.*aat1_final_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_pfs471_data <- merge_files("03_DrugR_NT_Sequences", "Survey.*pfs471_final_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_pfs472_data <- merge_files("03_DrugR_NT_Sequences", "Survey.*pfs472_final_NT_Aligned.csv", c("3D7_Pos", "3D7_Ref"))

Survey_merged_crt_YN <- merge_files("03_DrugR_NT_Changes", "Survey.*crt_final_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_dhfr_YN <- merge_files("03_DrugR_NT_Changes", "Survey.*dhfr_final_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_dhps1_YN <- merge_files("03_DrugR_NT_Changes", "Survey.*dhps1_final_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_dhps2_YN <- merge_files("03_DrugR_NT_Changes", "Survey.*dhps2_final_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_K131_YN <- merge_files("03_DrugR_NT_Changes", "Survey.*K131_final_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_K132_YN <- merge_files("03_DrugR_NT_Changes", "Survey.*K132_final_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_mdr11_YN <- merge_files("03_DrugR_NT_Changes", "Survey.*mdr11_final_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_mdr12_YN <- merge_files("03_DrugR_NT_Changes", "Survey.*mdr12_final_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_mdr13_YN <- merge_files("03_DrugR_NT_Changes", "Survey.*mdr13_final_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_aat1_YN <- merge_files("03_DrugR_NT_Changes", "Survey.*aat1_final_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_pfs471_YN <- merge_files("03_DrugR_NT_Changes", "Survey.*pfs471_final_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_pfs472_YN <- merge_files("03_DrugR_NT_Changes", "Survey.*pfs472_final_NT_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))

Survey_merged_crt_Change <- merge_files("03_DrugR_NT_Changes", "Survey.*crt_final_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_dhfr_Change <- merge_files("03_DrugR_NT_Changes", "Survey.*dhfr_final_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_dhps1_Change <- merge_files("03_DrugR_NT_Changes", "Survey.*dhps1_final_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_dhps2_Change <- merge_files("03_DrugR_NT_Changes", "Survey.*dhps2_final_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_K131_Change <- merge_files("03_DrugR_NT_Changes", "Survey.*K131_final_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_K132_Change <- merge_files("03_DrugR_NT_Changes", "Survey.*K132_final_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_mdr11_Change <- merge_files("03_DrugR_NT_Changes", "Survey.*mdr11_final_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_mdr12_Change <- merge_files("03_DrugR_NT_Changes", "Survey.*mdr12_final_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_mdr13_Change <- merge_files("03_DrugR_NT_Changes", "Survey.*mdr13_final_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_aat1_Change <- merge_files("03_DrugR_NT_Changes", "Survey.*aat1_final_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_pfs471_Change <- merge_files("03_DrugR_NT_Changes", "Survey.*pfs471_final_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_pfs472_Change <- merge_files("03_DrugR_NT_Changes", "Survey.*pfs472_final_NT_Aligned_NTchange.csv", c("3D7_Pos", "3D7_Ref"))

Survey_merged_AA_crt_data <- merge_files("03_DrugR_AA_Sequences", "Survey.*crt_final_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_dhfr_data <- merge_files("03_DrugR_AA_Sequences", "Survey.*dhfr_final_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_dhps1_data <- merge_files("03_DrugR_AA_Sequences", "Survey.*dhps1_final_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_dhps2_data <- merge_files("03_DrugR_AA_Sequences", "Survey.*dhps2_final_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_K131_data <- merge_files("03_DrugR_AA_Sequences", "Survey.*K131_final_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_K132_data <- merge_files("03_DrugR_AA_Sequences", "Survey.*K132_final_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_mdr11_data <- merge_files("03_DrugR_AA_Sequences", "Survey.*mdr11_final_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_mdr12_data <- merge_files("03_DrugR_AA_Sequences", "Survey.*mdr12_final_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_mdr13_data <- merge_files("03_DrugR_AA_Sequences", "Survey.*mdr13_final_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_aat1_data <- merge_files("03_DrugR_AA_Sequences", "Survey.*aat1_final_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_pfs471_data <- merge_files("03_DrugR_AA_Sequences", "Survey.*pfs471_final_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_pfs472_data <- merge_files("03_DrugR_AA_Sequences", "Survey.*pfs472_final_AA_Aligned.csv", c("3D7_Pos", "3D7_Ref"))

Survey_merged_AA_crt_YN <- merge_files("03_DrugR_AA_Changes", "Survey.*crt_final_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_dhfr_YN <- merge_files("03_DrugR_AA_Changes", "Survey.*dhfr_final_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_dhps1_YN <- merge_files("03_DrugR_AA_Changes", "Survey.*dhps1_final_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_dhps2_YN <- merge_files("03_DrugR_AA_Changes", "Survey.*dhps2_final_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_K131_YN <- merge_files("03_DrugR_AA_Changes", "Survey.*K131_final_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_K132_YN <- merge_files("03_DrugR_AA_Changes", "Survey.*K132_final_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_mdr11_YN <- merge_files("03_DrugR_AA_Changes", "Survey.*mdr11_final_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_mdr12_YN <- merge_files("03_DrugR_AA_Changes", "Survey.*mdr12_final_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_mdr13_YN <- merge_files("03_DrugR_AA_Changes", "Survey.*mdr13_final_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_aat1_YN <- merge_files("03_DrugR_AA_Changes", "Survey.*aat1_final_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_pfs471_YN <- merge_files("03_DrugR_AA_Changes", "Survey.*pfs471_final_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_pfs472_YN <- merge_files("03_DrugR_AA_Changes", "Survey.*pfs472_final_AA_Aligned_YorN.csv", c("3D7_Pos", "3D7_Ref"))

Survey_merged_AA_crt_Change <- merge_files("03_DrugR_AA_Changes", "Survey.*crt_final_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_dhfr_Change <- merge_files("03_DrugR_AA_Changes", "Survey.*dhfr_final_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_dhps1_Change <- merge_files("03_DrugR_AA_Changes", "Survey.*dhps1_final_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_dhps2_Change <- merge_files("03_DrugR_AA_Changes", "Survey.*dhps2_final_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_K131_Change <- merge_files("03_DrugR_AA_Changes", "Survey.*K131_final_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_K132_Change <- merge_files("03_DrugR_AA_Changes", "Survey.*K132_final_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_mdr11_Change <- merge_files("03_DrugR_AA_Changes", "Survey.*mdr11_final_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_mdr12_Change <- merge_files("03_DrugR_AA_Changes", "Survey.*mdr12_final_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_mdr13_Change <- merge_files("03_DrugR_AA_Changes", "Survey.*mdr13_final_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_aat1_Change <- merge_files("03_DrugR_AA_Changes", "Survey.*aat1_final_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_pfs471_Change <- merge_files("03_DrugR_AA_Changes", "Survey.*pfs471_final_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
Survey_merged_AA_pfs472_Change <- merge_files("03_DrugR_AA_Changes", "Survey.*pfs472_final_AA_Aligned_AAchange.csv", c("3D7_Pos", "3D7_Ref"))
```

## 3. Write New Merged .csv Files 

- Note: REPLACE "04_Compiled_NT_Sequences" with YOUR OUTPUT FOLDER
- Note: REPLACE "04_Compiled_AA_Sequences" with YOUR OUTPUT FOLDER

```{r}
write.csv(Survey_merged_crt_data, "04_Compiled_NT_Sequences/Survey_crt_final_NT_Aligned.csv", row.names = F)
write.csv(Survey_merged_dhfr_data, "04_Compiled_NT_Sequences/Survey_dhfr_final_NT_Aligned.csv", row.names = F)
write.csv(Survey_merged_dhps1_data, "04_Compiled_NT_Sequences/Survey_dhps1_final_NT_Aligned.csv", row.names = F)
write.csv(Survey_merged_dhps2_data, "04_Compiled_NT_Sequences/Survey_dhps2_final_NT_Aligned.csv", row.names = F)
write.csv(Survey_merged_K131_data, "04_Compiled_NT_Sequences/Survey_K131_final_NT_Aligned.csv", row.names = F)
write.csv(Survey_merged_K132_data, "04_Compiled_NT_Sequences/Survey_K132_final_NT_Aligned.csv", row.names = F)
write.csv(Survey_merged_mdr11_data, "04_Compiled_NT_Sequences/Survey_mdr11_final_NT_Aligned.csv", row.names = F)
write.csv(Survey_merged_mdr12_data, "04_Compiled_NT_Sequences/Survey_mdr12_final_NT_Aligned.csv", row.names = F)
write.csv(Survey_merged_mdr13_data, "04_Compiled_NT_Sequences/Survey_mdr13_final_NT_Aligned.csv", row.names = F)
write.csv(Survey_merged_aat1_data, "04_Compiled_NT_Sequences/Survey_aat1_final_NT_Aligned.csv", row.names = F)
write.csv(Survey_merged_pfs471_data, "04_Compiled_NT_Sequences/Survey_pfs471_final_NT_Aligned.csv", row.names = F)
write.csv(Survey_merged_pfs472_data, "04_Compiled_NT_Sequences/Survey_pfs472_final_NT_Aligned.csv", row.names = F)

write.csv(Survey_merged_crt_YN, "04_Compiled_NT_Sequences/Survey_crt_final_NT_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_dhfr_YN, "04_Compiled_NT_Sequences/Survey_dhfr_final_NT_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_dhps1_YN, "04_Compiled_NT_Sequences/Survey_dhps1_final_NT_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_dhps2_YN, "04_Compiled_NT_Sequences/Survey_dhps2_final_NT_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_K131_YN, "04_Compiled_NT_Sequences/Survey_K131_final_NT_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_K132_YN, "04_Compiled_NT_Sequences/Survey_K132_final_NT_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_mdr11_YN, "04_Compiled_NT_Sequences/Survey_mdr11_final_NT_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_mdr12_YN, "04_Compiled_NT_Sequences/Survey_mdr12_final_NT_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_mdr13_YN, "04_Compiled_NT_Sequences/Survey_mdr13_final_NT_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_aat1_YN, "04_Compiled_NT_Sequences/Survey_aat1_final_NT_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_pfs471_YN, "04_Compiled_NT_Sequences/Survey_pfs471_final_NT_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_pfs472_YN, "04_Compiled_NT_Sequences/Survey_pfs472_final_NT_Aligned_YorN.csv", row.names = F)

write.csv(Survey_merged_crt_Change, "04_Compiled_NT_Sequences/Survey_crt_final_NT_Aligned_NTchange.csv", row.names = F)
write.csv(Survey_merged_dhfr_Change, "04_Compiled_NT_Sequences/Survey_dhfr_final_NT_Aligned_NTchange.csv", row.names = F)
write.csv(Survey_merged_dhps1_Change, "04_Compiled_NT_Sequences/Survey_dhps1_final_NT_Aligned_NTchange.csv", row.names = F)
write.csv(Survey_merged_dhps2_Change, "04_Compiled_NT_Sequences/Survey_dhps2_final_NT_Aligned_NTchange.csv", row.names = F)
write.csv(Survey_merged_K131_Change, "04_Compiled_NT_Sequences/Survey_K131_final_NT_Aligned_NTchange.csv", row.names = F)
write.csv(Survey_merged_K132_Change, "04_Compiled_NT_Sequences/Survey_K132_final_NT_Aligned_NTchange.csv", row.names = F)
write.csv(Survey_merged_mdr11_Change, "04_Compiled_NT_Sequences/Survey_mdr11_final_NT_Aligned_NTchange.csv", row.names = F)
write.csv(Survey_merged_mdr12_Change, "04_Compiled_NT_Sequences/Survey_mdr12_final_NT_Aligned_NTchange.csv", row.names = F)
write.csv(Survey_merged_mdr13_Change, "04_Compiled_NT_Sequences/Survey_mdr13_final_NT_Aligned_NTchange.csv", row.names = F)
write.csv(Survey_merged_aat1_Change, "04_Compiled_NT_Sequences/Survey_aat1_final_NT_Aligned_NTchange.csv", row.names = F)
write.csv(Survey_merged_pfs471_Change, "04_Compiled_NT_Sequences/Survey_pfs471_final_NT_Aligned_NTchange.csv", row.names = F)
write.csv(Survey_merged_pfs472_Change, "04_Compiled_NT_Sequences/Survey_pfs472_final_NT_Aligned_NTchange.csv", row.names = F)

write.csv(Survey_merged_AA_crt_data, "04_Compiled_AA_Sequences/Survey_crt_final_AA_Aligned.csv", row.names = F)
write.csv(Survey_merged_AA_dhfr_data, "04_Compiled_AA_Sequences/Survey_dhfr_final_AA_Aligned.csv", row.names = F)
write.csv(Survey_merged_AA_dhps1_data, "04_Compiled_AA_Sequences/Survey_dhps1_final_AA_Aligned.csv", row.names = F)
write.csv(Survey_merged_AA_dhps2_data, "04_Compiled_AA_Sequences/Survey_dhps2_final_AA_Aligned.csv", row.names = F)
write.csv(Survey_merged_AA_K131_data, "04_Compiled_AA_Sequences/Survey_K131_final_AA_Aligned.csv", row.names = F)
write.csv(Survey_merged_AA_K132_data, "04_Compiled_AA_Sequences/Survey_K132_final_AA_Aligned.csv", row.names = F)
write.csv(Survey_merged_AA_mdr11_data, "04_Compiled_AA_Sequences/Survey_mdr11_final_AA_Aligned.csv", row.names = F)
write.csv(Survey_merged_AA_mdr12_data, "04_Compiled_AA_Sequences/Survey_mdr12_final_AA_Aligned.csv", row.names = F)
write.csv(Survey_merged_AA_mdr13_data, "04_Compiled_AA_Sequences/Survey_mdr13_final_AA_Aligned.csv", row.names = F)
write.csv(Survey_merged_AA_aat1_data, "04_Compiled_AA_Sequences/Survey_aat1_final_AA_Aligned.csv", row.names = F)
write.csv(Survey_merged_AA_pfs471_data, "04_Compiled_AA_Sequences/Survey_pfs471_final_AA_Aligned.csv", row.names = F)
write.csv(Survey_merged_AA_pfs472_data, "04_Compiled_AA_Sequences/Survey_pfs472_final_AA_Aligned.csv", row.names = F)

write.csv(Survey_merged_AA_crt_YN, "04_Compiled_AA_Sequences/Survey_crt_final_AA_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_AA_dhfr_YN, "04_Compiled_AA_Sequences/Survey_dhfr_final_AA_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_AA_dhps1_YN, "04_Compiled_AA_Sequences/Survey_dhps1_final_AA_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_AA_dhps2_YN, "04_Compiled_AA_Sequences/Survey_dhps2_final_AA_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_AA_K131_YN, "04_Compiled_AA_Sequences/Survey_K131_final_AA_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_AA_K132_YN, "04_Compiled_AA_Sequences/Survey_K132_final_AA_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_AA_mdr11_YN, "04_Compiled_AA_Sequences/Survey_mdr11_final_AA_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_AA_mdr12_YN, "04_Compiled_AA_Sequences/Survey_mdr12_final_AA_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_AA_mdr13_YN, "04_Compiled_AA_Sequences/Survey_mdr13_final_AA_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_AA_aat1_YN, "04_Compiled_AA_Sequences/Survey_aat1_final_AA_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_AA_pfs471_YN, "04_Compiled_AA_Sequences/Survey_pfs471_final_AA_Aligned_YorN.csv", row.names = F)
write.csv(Survey_merged_AA_pfs472_YN, "04_Compiled_AA_Sequences/Survey_pfs472_final_AA_Aligned_YorN.csv", row.names = F)

write.csv(Survey_merged_AA_crt_Change, "04_Compiled_AA_Sequences/Survey_crt_final_AA_Aligned_AAchange.csv", row.names = F)
write.csv(Survey_merged_AA_dhfr_Change, "04_Compiled_AA_Sequences/Survey_dhfr_final_AA_Aligned_AAchange.csv", row.names = F)
write.csv(Survey_merged_AA_dhps1_Change, "04_Compiled_AA_Sequences/Survey_dhps1_final_AA_Aligned_AAchange.csv", row.names = F)
write.csv(Survey_merged_AA_dhps2_Change, "04_Compiled_AA_Sequences/Survey_dhps2_final_AA_Aligned_AAchange.csv", row.names = F)
write.csv(Survey_merged_AA_K131_Change, "04_Compiled_AA_Sequences/Survey_K131_final_AA_Aligned_AAchange.csv", row.names = F)
write.csv(Survey_merged_AA_K132_Change, "04_Compiled_AA_Sequences/Survey_K132_final_AA_Aligned_AAchange.csv", row.names = F)
write.csv(Survey_merged_AA_mdr11_Change, "04_Compiled_AA_Sequences/Survey_mdr11_final_AA_Aligned_AAchange.csv", row.names = F)
write.csv(Survey_merged_AA_mdr12_Change, "04_Compiled_AA_Sequences/Survey_mdr12_final_AA_Aligned_AAchange.csv", row.names = F)
write.csv(Survey_merged_AA_mdr13_Change, "04_Compiled_AA_Sequences/Survey_mdr13_final_AA_Aligned_AAchange.csv", row.names = F)
write.csv(Survey_merged_AA_aat1_Change, "04_Compiled_AA_Sequences/Survey_aat1_final_AA_Aligned_AAchange.csv", row.names = F)
write.csv(Survey_merged_AA_pfs471_Change, "04_Compiled_AA_Sequences/Survey_pfs471_final_AA_Aligned_AAchange.csv", row.names = F)
write.csv(Survey_merged_AA_pfs472_Change, "04_Compiled_AA_Sequences/Survey_pfs472_final_AA_Aligned_AAchange.csv", row.names = F)
```
