---
title: "Antimalarial Drug Resistance"
bibliography: DrugR_About.bib
---

## Antifolates (*pfdhfr*, *pfdhps*)
Treatment failure of antifolate drugs such as sulfadoxine and pyrimethamine (collectively referred to as SP) have been associated with point mutations in *P. falciparum* dihydrofolate reductase (*pfdhfr*) and *P. falciparum* dihydropteroate synthase (*pfdhps*) genes on chromosome 4 and 8 respectively. These genes encode for enzymes in the folate biosynthesis pathway; *pfdhfr* reduces 7,8-dihydrofolate into 5,6,7,8-tetrahydrofolate and *pfdhps* aids in the synthesis of folate from GTPC. These enzymes are targeted by pyrimethamine and sulfadoxine respectively. 

The *pfdhfr* S108N substitution is critical for pyrimethamine resistance, with N51I and C59R modulating the strength of resistance conferring sub-optimal activity and reduced binding affinity to the drug (reviewed in @Naidoo2013). Additional mutations at A16V and C50A have been shown to confer resistance to cycloguanil and enhance pyrimethamine in Bolivia [@Foote1990; @Plowe1997]. The I164L substitution has been found alone or in combination with the triple *pfdhfr* mutant haplotype as an important resistance-conferring mutation. Haplotypes are biologically meaningful, since they determine the resistance properties of parasites that are exposed to drugs at the time of treatment. A triple mutant *pfdhfr* haplotype with 51I-59R-108N has a 1.5- to 3-fold higher pyrimethamine resistance in vitro than either 51I-108N or 59-108N double mutant haplotypes [@Sirawaraporn1997]. 

Sulfadoxine resistance first arises from the A437G substitution with compensatory mutations at I431V, S436A/F, K540E, A581G, A613S further increasing resistance. There are location-specific differences in *pfdhps* haplotypes in Africa, with East & Southern African isolates containing both 437G and 540E, while only 437G is predominantly observed West and Central Africa [@Naidoo2013]. The 540E allele has been found to work together with 437G to raise the sulfadoxine tolerance of sensitive *pfdhps* by 200-fold compared to just 10-fold with only 437G substitution [@Triglia1997]. The A581G substitution was found in northern Tanzania [@Gesase2009] to reduce the efficacy of IPTp-SP [@Minja2013], and is also rare in West and Central Africa. I431V was identified from UK imported malaria infections originating Nigeria [@Sutherland2009] and pregnant women in Cameroon (@Chauvin2015).

Naidoo and Roper @Naidoo2013 have classified *pfdhfr* and *pfdhps* mediated resistance into three categories: 


(i) “partial resistant” haplotypes (*pfdhfr* 51I, 59R, 108N; *pfdhps* 473G), 
(ii) “full resistant” haplotypes also containing *pfdhps* 540E, associated with clinical failure of SP treatment, and 
(iii) “super-resistant” haplotypes additionally harbouring 581G. More recently, the local emergence and clonal expansion of a VAGKGS mutant 


haplotype was observed in north-western Central Africa (Cameroon and Nigeria) and there was a mutually exclusive relationship with the presence of the 431V mutation and the 540E mutation in countries further south in Central Africa (Republic of Congo, Democratic Republic of Congo, Gabon, Central African Republic and Angola) [@Guemas2023]. The study also found the 431V mutant found in 15 countries in Central and West Africa, with the majority also having the VAGKGS haplotype. Given the additive nature of *pfdhps* mutations leading to increased resistance, this haplotype would confer an increased level of resistance to SP. It was therefore theorised that this haplotype is the Central and West African equivalent of a “super-resistant” mutant [@Guemas2023].

## Aminoquinolines and Arylaminoalcohols (*pfcrt*, *pfmdr1*, *pfaat1*)

4-aminoquinoline and arylaminoalcohol antimalarial drugs such as chloroquine, amodiaquine and lumefantrine accumulate in the digestive vacuole and reduce haemoglobin degradation, resulting in parasite death (reviewed in @Wicht2020). 


Non-synonymous point mutations in *P. falciparum* chloroquine resistance transporter (*pfcrt*) on chromosome 7 mediate drug resistance by preventing drug accumulation in the digestive vacuole. The K76T substitution in *pfcrt* results in the export of drugs from the digestive vacuole and additional region-specific mutations exist at codon positions 72-75 line the central drug binding cavity of *pfcrt* (reviewed in @Wicht2020). For example, the SVMNT haplotype is predominantly observed in South American isolates as opposed to the CVIET haplotypes found in Africa and Asian isolates @Veiga2016. Both haplotypes have been demonstrated to be equally associated with AQR [@Wicht2020]. 


Mutations in *pfcrt* alone cannot halt drug accumulation in the digestive vacuole via passive diffusion. This is supplemented by mutations in *P. falciparum* multidrug resistance protein 1 (*pfmdr1*) on chromosome 5 and *P. falciparum* putative amino acid transporter 1 (*pfaat1*) on chromosome 6. Substitutions at *pfmdr1* codons 86, 184, 1034, 1042 and 1246 result in a strong epistatic interaction with *pfcrt* 76T and act to augment drug resistance. Amino-terminal substitutions (N86Y and Y184F) are common in Asian and African parasites, while carboxy-terminal mutations (S1034C, N1042D and D1246Y) are found at a higher frequency in South American isolates [@Veiga2016; @Wicht2020]. Mutations at 1034C and 1042D have been shown to be involved in the binding pocket of 4-aminoquinolines [@Sidhu2005]. ASAQ and AL are the most commonly used ACTs in Africa [@WorldHealthOrganisation2020] and exert opposing selection pressures on *pfcrt* and *pfmdr1*. ASAQ selects for the *pfmdr1* 86Y, Y184 and 1246Y alleles (or the *pfmdr1* YYY haplotype) while the absence of aminoquinoline resistance mutations is preferentially selected for by AL (i.e., N86, 184F and D1246 alleles; or the *pfmdr1* NFD haplotype) [@Ochong2003; @Dokomajilar2006; @Happi2006; @Holmgren2007; @Humphreys2007; @Tinto2008; @Froberg2012; @Baraka2014; @Venkatesan2014; @Otienoburu2016; @Sondo2016; @Okell2018]. ASAQ also preferentially selects for *pfcrt* 76T allele, while AL has been associated with increased prevalence of the *pfcrt* K76 allele [@Echeverry2007; @Folarin2011; @Venkatesan2014]. Therefore it has been proposed that either rotating or deploying b ASAQ and AL simultaneously may reduce the spread of mutations that confer resistance the ACT and/or ACT partner drugs in the population [@Valderramos2010; @Nguyen2015; @Yeka2015]. 


Recently, *pfaat1* was found to have an epistatic interaction with 76T and parasite fitness [@Amambua-Ngwa2023]. The combined *pfaat1* single mutant 258L and *pfcrt* 76T types result in high chloroquine resistance but at a very high fitness cost and was found in African isolates, while the *pfaat1* double mutant (258L and 313S) and *pfcrt* 76T causes chloroquine resistance and an increase in fitness (to almost wild-type levels) and is found in southeast Asian isolates. 

## Artemisinins (*pfk13*)

Artemisinin resistance is defined as delayed parasite clearance following treatment with an artemisinin derivative drug (reviewed in @Pandit2023). This is a form of ‘partial resistance’ where the phenotype manifests as reduced drug susceptibility in ring-stage parasites. Single point mutations in the Broad-Complex, Tramtrack and Bric-à-bac/Pox virus and Zinc Finger domain (BTB/POZ; codons 350-437) and the propeller domain (codons 350-726) of *pfk13* (encoded by *pfk13* on chromosome 13) have been found to confer this phenotype. There are currently 13 validated *pfk13* markers and nine that are candidate/associated with artemisinin resistance. The *pfk13* mutations that have originated in southeast Asia (C580Y, R539T, Y493H, I543T) confer the slow-clearance phenotype in vitro following exposure to artemisinin [@Ariey2014; @Conrad2014; @Zhu2018; @Moser2020; @Uwimana2020; @Kirby2022; @Fola2023; @Juliano2023). 


Validated *pfk13* mutations have been identified in Africa at very low frequencies (<1%) [@Owoloye2021]. De novo non-synonymous mutations with local flanking haplotypes (SNPs or microsatellites) have arisen at low frequency such as R561H (Rwanda, Tanzania), R622I (Ethiopia), and A675V (Uganda) in East Africa. However, 622I has not yet been validated to cause artemisinin resistance in vitro nor in vivo [@Fola2023]. The introduction and spread of ART resistant strains in Tanzania [@Juliano2023), Uganda [@Conrad2023], Rwanda [@Uwimana2020] and Ethiopia [@Fola2023] raise concerns for the rest of Africa and its first-line therapeutic. Other markers in different genes have been observed to act as background and secondary determinants of resistance such as PfCoronin, PfUbp1, PfAP2μ, *pfmdr1* copy number, PfTrx1 and PfSpp (Pandit2023).
