---
title: "Multiplex 2: crt, mdr1 (1034, 1042, 1246)"
bibliography: DrugR_About.bib
---

```{r}
#| echo: false
library(tidyverse)
library(kableExtra)
options(knitr.table.format = "html") 
knitr::opts_chunk$set(cache=TRUE)
theme_set(theme_bw())

alarmcoding_PCR <- read.csv("Data/ALARMcoding_PCR.csv")
alarmcoding_PCR_multiplex_2 <- alarmcoding_PCR %>% filter(Multiplex == 2)
```

## Primers

Round 1 PCR primers were designed using Geneious Prime R9 to target larger fragments that surrounded the codons of interest per marker. The forward and reverse primers for pfmdr1.1.2 and pfmdr1.1.3 and the forward primer for pfcrt from [@Nag2017] were used as the round 2 PCR primers, and a novel pfcrt reverse primer was developed using Geneious Prime R9. 


```{r}
#| echo: false

colnames(alarmcoding_PCR_multiplex_2)[colnames(alarmcoding_PCR_multiplex_2)=="Nucleotide.position"] <- "Nucleotide position"
colnames(alarmcoding_PCR_multiplex_2)[colnames(alarmcoding_PCR_multiplex_2)=="Sequence"] <- "Sequence (5' to 3')"
names(alarmcoding_PCR_multiplex_2)[6] <- paste0(names(alarmcoding_PCR_multiplex_2)[6],  footnote_marker_alphabet(1))

alarmcoding_PCR_multiplex_2 %>%
  select(-c(Gene, Length:Codons)) %>% 
  knitr::kable(caption = "Details about primers used in Multiplex 2 of the <em>ALARM</em>coding PCR genotyping protocol", align = "lcccc", escape = FALSE) %>% 
      # Remember this escape = F
  kable_styling(bootstrap_options = c("condensed", "responsive"), full_width = T) %>% 
  group_rows("Multidrug resistance gene 1 (<em>pfmdr1</em>: Pf3D7_0523000)", 1, 6, escape = FALSE) %>% 
  group_rows("Chloroquine resistance transporter (<em>pfcrt</em>: PF3D7_070900)", 7, 10, escape = FALSE) %>%
  footnote(general = "Blank references denote that the primer was created by this author. Manuscript in preparation.",
           number = c(" ", "Nag et al., 2017"),
           alphabet = c("Nucleotide position relative ot the 3D7 v3 reference genome"),
           footnote_as_chunk = T)
```


## PCR Instructions 

### Round 1. 
The first round PCR was carried out in a total volume of 25 μL with 1 x buffer, 500 uM dNTPs mix (Promega), 5 mM MgCl2, 200 nM of each primer (forward (F) and reverse (R)), 1.3 units of GoTaq Flexi Polymerase (Promega) and only 2 μL of genomic DNA. 

```{r}
#| echo: false
multiplex_2_round_1 <- data.frame(
    Reagent = c("Water", "5X Buffer", "MgCl2 (25mM)", "dNTP", "mdr1-2-out-F", "mdr1-2-out-R", "crt-out-F", "crt-out-R", "G2 Flexi Taq (5U/μL)"), 
    Concentration = c("", "1x", "5 mM", "0.5 μM", "0.2 μM", "0.2 μM", "0.2 μM", "0.2 μM", "1.3 U/μL"), 
    Volume = c(9.49, 5, 5, 1.25, 0.5, 0.5, 0.5, 0.5, 0.26)
  ) %>% 
  mutate(Volume_x8 = Volume*8)

colnames(multiplex_2_round_1)[colnames(multiplex_2_round_1)=="Concentration"] <- "Final Concentration"
colnames(multiplex_2_round_1)[colnames(multiplex_2_round_1)=="Volume"] <- "Volume x1 (μL)"
colnames(multiplex_2_round_1)[colnames(multiplex_2_round_1)=="Volume_x8"] <- "Volume x8 (μL)"
multiplex_2_round_1  %>% 
  knitr::kable(align = "lcccc") %>% 
  kable_styling(bootstrap_options = c("condensed", "responsive"), full_width = T)  
```

Cycling conditions: 

* 95°C/2 min 
* 40 cycles of: 
  * 95°C/20 sec, 
  * 58°C/2 min, 
  * 72/2 min;
* 72°C/10 min
* Resting at 4°C. 


### Round 2. 

::: panel-tabset

#### mdr1-2

For both pfmdr1 markers (mdr1-2/mdr1-3), the simplex PCR was performed in separate reactions for each inner primer pair with a total volume of 25μL, with a master mix that contained 1x Buffer, 500 uM dNTPs mix, 5 mM MgCl2, 300 nM per primer (F/R), 1.3 units of GoTaq Flexi Polymerase and 1 μL of PCR product from the first round.

```{r}
#| echo: false
multiplex_2_round_2_mdr12 <- data.frame(
    Reagent = c("Water", "5X Buffer", "MgCl2 (25mM)", "dNTP", "mdr1-2-in-F", "mdr1-2-in-R", "G2 Flexi Taq (5U/μL)"), 
    Concentration = c("", "1x", "5 mM", "0.5 μM", "0.3 μM", "0.3 μM", "1.3 U/μL"), 
    Volume = c(10.99, 5, 5, 1.25, 0.75, 0.75, 0.26)
  ) %>% 
  mutate(Volume_x8 = Volume*8)

colnames(multiplex_2_round_2_mdr12)[colnames(multiplex_2_round_2_mdr12)=="Concentration"] <- "Final Concentration"
colnames(multiplex_2_round_2_mdr12)[colnames(multiplex_2_round_2_mdr12)=="Volume"] <- "Volume x1 (μL)"
colnames(multiplex_2_round_2_mdr12)[colnames(multiplex_2_round_2_mdr12)=="Volume_x8"] <- "Volume x8 (μL)"
multiplex_2_round_2_mdr12  %>% 
  knitr::kable(align = "lcccc") %>% 
  kable_styling(bootstrap_options = c("condensed", "responsive"), full_width = T)  
```

Cycling conditions:

* 95°C/2 min
* 40 cycles of: 
  * 95°C/20 sec
  * 58°C/2 min
  * 72/2 min;
* 72°C/10 min
* Resting at 4°C. 

#### mdr1-3

For both pfmdr1 markers (mdr1-2/mdr1-3), the simplex PCR was performed in separate reactions for each inner primer pair with a total volume of 25μL, with a master mix that contained 1x Buffer, 500 uM dNTPs mix, 5 mM MgCl2, 300 nM per primer (F/R), 1.3 units of GoTaq Flexi Polymerase and 1 μL of PCR product from the first round.

```{r}
#| echo: false
multiplex_2_round_2_mdr13 <- data.frame(
    Reagent = c("Water", "5X Buffer", "MgCl2 (25mM)", "dNTP", "mdr1-3-in-F", "mdr1-3-in-R", "G2 Flexi Taq (5U/μL)"), 
    Concentration = c("", "1x", "5 mM", "0.5 μM", "0.3 μM", "0.3 μM", "1.3 U/μL"), 
    Volume = c(10.99, 5, 5, 1.25, 0.75, 0.75, 0.26)
  ) %>% 
  mutate(Volume_x8 = Volume*8)

colnames(multiplex_2_round_2_mdr13)[colnames(multiplex_2_round_2_mdr13)=="Concentration"] <- "Final Concentration"
colnames(multiplex_2_round_2_mdr13)[colnames(multiplex_2_round_2_mdr13)=="Volume"] <- "Volume x1 (μL)"
colnames(multiplex_2_round_2_mdr13)[colnames(multiplex_2_round_2_mdr13)=="Volume_x8"] <- "Volume x8 (μL)"
multiplex_2_round_2_mdr13  %>% 
  knitr::kable(align = "lcccc") %>% 
  kable_styling(bootstrap_options = c("condensed", "responsive"), full_width = T)  
```

Cycling conditions:

* 95°C/2 min
* 40 cycles of: 
  * 95°C/20 sec
  * 58°C/2 min
  * 72/2 min;
* 72°C/10 min
* Resting at 4°C. 

#### crt

For pfcrt, the master mix contained 1x Buffer, 500 uM dNTPs mix, 5 mM MgCl2, 200 nM per primer (F/R), 1.3 units of GoTaq Flexi Polymerase and 1 μL of PCR product from the first round. 

```{r}
#| echo: false
multiplex_2_round_2_crt <- data.frame(
    Reagent = c("Water", "5X Buffer", "MgCl2 (25mM)", "dNTP", "crt-in-F", "crt-in-R", "G2 Flexi Taq (5U/μL)"), 
    Concentration = c("", "1x", "5 mM", "0.5 μM", "0.2 μM", "0.2 μM", "1.3 U/μL"), 
    Volume = c(11.49, 5, 5, 1.25, 0.5, 0.5, 0.26)
  ) %>% 
  mutate(Volume_x8 = Volume*8)

colnames(multiplex_2_round_2_crt)[colnames(multiplex_2_round_2_crt)=="Concentration"] <- "Final Concentration"
colnames(multiplex_2_round_2_crt)[colnames(multiplex_2_round_2_crt)=="Volume"] <- "Volume x1 (μL)"
colnames(multiplex_2_round_2_crt)[colnames(multiplex_2_round_2_crt)=="Volume_x8"] <- "Volume x8 (μL)"
multiplex_2_round_2_crt  %>% 
  knitr::kable(align = "lcccc") %>% 
  kable_styling(bootstrap_options = c("condensed", "responsive"), full_width = T)  
```


Cycling conditions:

* 95°C/2 min
* 40 cycles of: 
  * 95°C/20 sec
  * **62**°C/2 min
  * 72/2 min;
* 72°C/10 min
* Resting at 4°C. 

:::


## Final Sequences 

```{r}
#| echo: false
alarmcoding_PCR_multiplex_2 %>%
  distinct(Length, PCR_Round, Codons) %>% 
  mutate(Marker = case_when(Length==1468 ~ "mdr1-2 Round 1 Product",
                            Length==488 ~ "mdr1-2",
                            Length==552 ~ "mdr1-3",
                            Length==777 ~ "crt Round 1 Product",
                            Length==214 ~ "crt", .default = "CHECK")) %>%
  select(Marker, Length, PCR_Round, Codons) %>% 
  knitr::kable(caption = "Table 2. Details about sequences genotyped in Multiplex 2 of the <em>ALARM</em>coding protocol", align = "lcccc", escape = FALSE) %>% 
  kable_styling(bootstrap_options = c("condensed", "responsive"), full_width = T) %>% 
  group_rows("Multidrug resistance gene 1 (<em>pfmdr1</em>: Pf3D7_0523000)", 1, 3, escape = FALSE) %>% 
  group_rows("Chloroquine resistance transporter (<em>pfcrt</em>: PF3D7_070900)", 4, 5, escape = FALSE)
```

