---
title: "Multiplex 3: dhfr, dhps"
bibliography: DrugR_About.bib
---

```{r}
#| echo: false
library(tidyverse)
library(kableExtra)
options(knitr.table.format = "html") 
knitr::opts_chunk$set(cache=TRUE)
theme_set(theme_bw())

alarmcoding_PCR <- read.csv("Data/ALARMcoding_PCR.csv")
alarmcoding_PCR_multiplex_3 <- alarmcoding_PCR %>% filter(Multiplex == 3)
```

## Primers

The round 1 PCR primers were designed using Geneious Prime R9 to target larger fragments that encapsulated the primers described in [@Nag2017] that were used as the round 2 PCR primers.

```{r}
#| echo: false

colnames(alarmcoding_PCR_multiplex_3)[colnames(alarmcoding_PCR_multiplex_3)=="Nucleotide.position"] <- "Nucleotide position"
colnames(alarmcoding_PCR_multiplex_3)[colnames(alarmcoding_PCR_multiplex_3)=="Sequence"] <- "Sequence (5' to 3')"
names(alarmcoding_PCR_multiplex_3)[6] <- paste0(names(alarmcoding_PCR_multiplex_3)[6],  footnote_marker_alphabet(1))

alarmcoding_PCR_multiplex_3 %>%
  select(-c(Gene, Length:Codons)) %>% 
  knitr::kable(caption = "Table 6. Details about primers used in Multiplex 3 of the <em>ALARM</em>coding PCR genotyping protocol", align = "lcccc", escape = FALSE) %>% 
      # Remember this escape = F
  kable_styling(bootstrap_options = c("condensed", "responsive"), full_width = T) %>% 
  group_rows("Dihydropteroate synthase (<em>pfdhps</em>: PF3D7_0810800)", 1, 6, escape = FALSE) %>% 
  group_rows("Dihydrofolate reductase (<em>pfdhfr</em>: PF3D7_0417200)", 7, 10, escape = FALSE) %>%
  footnote(general = "Blank references denote that the primer was created by this author. Manuscript in preparation.",
           number = c(" ", "Nag et al., 2017"),
           alphabet = c("Nucleotide position relative ot the 3D7 v3 reference genome"),
           footnote_as_chunk = T)
```

## PCR Instructions

### Round 1.

The first round PCR was carried out in a total volume of 25 μL with 1 x buffer, 500 uM dNTPs mix (Promega), 5 mM MgCl2, 200 nM of each primer (forward (F) and reverse (R)), 1.3 units of GoTaq Flexi Polymerase (Promega) and only 2 μL of genomic DNA.

```{r}
#| echo: false
multiplex_3_round_1 <- data.frame(
    Reagent = c("Water", "5X Buffer", "MgCl2 (25mM)", "dNTP", "K13-out-F", "K13-out-R", "mdr1-1-out-F", "mdr1-1-out-R", "G2 Flexi Taq (5U/μL)"), 
    Concentration = c("", "1x", "5 mM", "0.5 μM", "0.2 μM", "0.2 μM", "0.2 μM", "0.2 μM", "1.3 U/μL"), 
    Volume = c(9.49, 5, 5, 1.25, 0.5, 0.5, 0.5, 0.5, 0.26)
  ) %>% 
  mutate(Volume_x8 = Volume*8)

colnames(multiplex_3_round_1)[colnames(multiplex_3_round_1)=="Concentration"] <- "Final Concentration"
colnames(multiplex_3_round_1)[colnames(multiplex_3_round_1)=="Volume"] <- "Volume x1 (μL)"
colnames(multiplex_3_round_1)[colnames(multiplex_3_round_1)=="Volume_x8"] <- "Volume x8 (μL)"
multiplex_3_round_1  %>% 
  knitr::kable(align = "lcccc") %>% 
  kable_styling(bootstrap_options = c("condensed", "responsive"), full_width = T)  
```

Cycling conditions:

-   95°C/2 min
-   40 cycles of:
    -   95°C/20 sec,
    -   58°C/2 min,
    -   72/2 min;
-   72°C/10 min
-   Resting at 4°C.

### Round 2.

The second round of the nested PCR was carried out in separate reactions containing a single inner fragment primer pair and with a total volume of 25 μL containing 1x Buffer, 500 uM dNTPs mix, 5 mM MgCl2, 200 nM of each primer (F/R), 1.3 units of GoTaq Flexi Polymerase and 1 μL of PCR product from round 1.

::: panel-tabset
#### dhps-1

```{r}
#| echo: false
multiplex_3_round_2_dhps1 <- data.frame(
    Reagent = c("Water", "5X Buffer", "MgCl2 (25mM)", "dNTP", "dhps-1-in-F", "dhps-1-in-R", "G2 Flexi Taq (5U/μL)"), 
    Concentration = c("", "1x", "5 mM", "0.5 μM", "0.2 μM", "0.2 μM", "1.3 U/μL"), 
    Volume = c(11.49, 5, 5, 1.25, 0.5, 0.5, 0.26)
  ) %>% 
  mutate(Volume_x8 = Volume*8)

colnames(multiplex_3_round_2_dhps1)[colnames(multiplex_3_round_2_dhps1)=="Concentration"] <- "Final Concentration"
colnames(multiplex_3_round_2_dhps1)[colnames(multiplex_3_round_2_dhps1)=="Volume"] <- "Volume x1 (μL)"
colnames(multiplex_3_round_2_dhps1)[colnames(multiplex_3_round_2_dhps1)=="Volume_x8"] <- "Volume x8 (μL)"
multiplex_3_round_2_dhps1  %>% 
  knitr::kable(align = "lcccc") %>% 
  kable_styling(bootstrap_options = c("condensed", "responsive"), full_width = T)  
```

#### dhps-2

```{r}
#| echo: false
multiplex_3_round_2_dhps2 <- data.frame(
    Reagent = c("Water", "5X Buffer", "MgCl2 (25mM)", "dNTP", "dhps-2-in-F", "dhps-2-in-R", "G2 Flexi Taq (5U/μL)"), 
    Concentration = c("", "1x", "5 mM", "0.5 μM", "0.2 μM", "0.2 μM", "1.3 U/μL"), 
    Volume = c(11.49, 5, 5, 1.25, 0.5, 0.5, 0.26)
  ) %>% 
  mutate(Volume_x8 = Volume*8)

colnames(multiplex_3_round_2_dhps2)[colnames(multiplex_3_round_2_dhps2)=="Concentration"] <- "Final Concentration"
colnames(multiplex_3_round_2_dhps2)[colnames(multiplex_3_round_2_dhps2)=="Volume"] <- "Volume x1 (μL)"
colnames(multiplex_3_round_2_dhps2)[colnames(multiplex_3_round_2_dhps2)=="Volume_x8"] <- "Volume x8 (μL)"
multiplex_3_round_2_dhps2  %>% 
  knitr::kable(align = "lcccc") %>% 
  kable_styling(bootstrap_options = c("condensed", "responsive"), full_width = T)  
```

#### dhfr

```{r}
#| echo: false
multiplex_3_round_2_dhfr <- data.frame(
    Reagent = c("Water", "5X Buffer", "MgCl2 (25mM)", "dNTP", "dhfr-in-F", "dhfr-in-R", "G2 Flexi Taq (5U/μL)"), 
    Concentration = c("", "1x", "5 mM", "0.5 μM", "0.2 μM", "0.2 μM", "1.3 U/μL"), 
    Volume = c(11.49, 5, 5, 1.25, 0.5, 0.5, 0.26)
  ) %>% 
  mutate(Volume_x8 = Volume*8)

colnames(multiplex_3_round_2_dhfr)[colnames(multiplex_3_round_2_dhfr)=="Concentration"] <- "Final Concentration"
colnames(multiplex_3_round_2_dhfr)[colnames(multiplex_3_round_2_dhfr)=="Volume"] <- "Volume x1 (μL)"
colnames(multiplex_3_round_2_dhfr)[colnames(multiplex_3_round_2_dhfr)=="Volume_x8"] <- "Volume x8 (μL)"
multiplex_3_round_2_dhfr  %>% 
  knitr::kable(align = "lcccc") %>% 
  kable_styling(bootstrap_options = c("condensed", "responsive"), full_width = T)  
```
:::

Cycling conditions for each simplex reaction:

-   95°C/2 min
-   40 cycles of:
    -   95°C/20 sec
    -   58°C/2 min
    -   72/2 min;
-   72°C/10 min
-   Resting at 4°C.

## Final Sequences

```{r}
#| echo: false

alarmcoding_PCR_multiplex_3 %>%
  distinct(Length, PCR_Round, Codons) %>% 
  mutate(Marker = case_when(Length==1189 ~ "dhps Round 1 Product",
                            Length==493 ~ "dhps-1",
                            Length==521 ~ "dhps-2",
                            Length==766 ~ "dhfr Round 1 Product",
                            Length==523 ~ "dhfr", .default = "CHECK")) %>%
  select(Marker, Length, PCR_Round, Codons) %>% 
  knitr::kable(caption = "Table 7. Details about sequences genotyped in Multiplex 3 of the <em>ALARM</em>coding protocol", align = "lcccc", escape = FALSE) %>% 
  kable_styling(bootstrap_options = c("condensed", "responsive"), full_width = T) %>% 
  group_rows("Dihydropteroate synthase (<em>pfdhps</em>: PF3D7_0810800)", 1, 3, escape = FALSE) %>% 
  group_rows("Dihydrofolate reductase (<em>pfdhfr</em>: PF3D7_0417200)", 4, 5, escape = FALSE)
```
