---
title: "Multiplex 1: K13 and mdr1 (86, 184)"
bibliography: DrugR_About.bib
---

```{r}
#| echo: false
library(tidyverse)
library(kableExtra)
options(knitr.table.format = "html") 
knitr::opts_chunk$set(cache=TRUE)
theme_set(theme_bw())

alarmcoding_PCR <- read.csv("Data/ALARMcoding_PCR.csv")
alarmcoding_PCR_multiplex_1 <- alarmcoding_PCR %>% filter(Multiplex == 1)
```

## Primers

The first round for multiplex 1 was modified from [@Narh2020] and amplifies two regions of pfk13 and pfmdr1 fragment 1 (mdr1-1; codons 86, 184). Inner primers for each simplex reaction were used from [@Narh2020].

```{r}
#| echo: false

colnames(alarmcoding_PCR_multiplex_1)[colnames(alarmcoding_PCR_multiplex_1)=="Nucleotide.position"] <- "Nucleotide position"
colnames(alarmcoding_PCR_multiplex_1)[colnames(alarmcoding_PCR_multiplex_1)=="Sequence"] <- "Sequence (5' to 3')"
names(alarmcoding_PCR_multiplex_1)[6] <- paste0(names(alarmcoding_PCR_multiplex_1)[6],  footnote_marker_alphabet(1))

alarmcoding_PCR_multiplex_1 %>%
  select(-c(Gene, Length:Codons)) %>% 
  knitr::kable(caption = "Table 2. Details about primers used in multiplex 1 of the <em>ALARM</em>coding PCR genotyping protocol", align = "lcccc", escape = FALSE) %>% 
      # Remember this escape = F
  kable_styling(bootstrap_options = c("condensed", "responsive"), full_width = T) %>% 
  group_rows("Kelch protein (<em>pfk13</em>: PF3D7_1343700)", 1, 6, escape = FALSE) %>% 
  group_rows("Multidrug resistance gene 1 (<em>pfmdr1</em>: Pf3D7_0523000)", 7, 10, escape = FALSE) %>%
  footnote(general = "Blank references denote that the primer was created by this author. Manuscript in preparation.",
           number = c("Narh et al., 2020", "Nag et al., 2017"),
           alphabet = c("Nucleotide position relative ot the 3D7 v3 reference genome"),
           footnote_as_chunk = T)
```

## PCR Instructions

### Round 1.

The first round PCR was carried out in a total Volume of 25 μL with 1 x buffer, 500 μM dNTPs mix (Promega), 5 mM MgCl2, 200 nM of each primer (forward (F) and reverse (R)), 1.3 units of GoTaq Flexi Polymerase (Promega) and 3 μL of genomic DNA.

```{r}
#| echo: false
multiplex_1_round_1 <- data.frame(
    Reagent = c("Water", "5X Buffer", "MgCl2 (25mM)", "dNTP", "K13-out-F", "K13-out-R", "mdr1-1-out-F", "mdr1-1-out-R", "G2 Flexi Taq (5U/μL)"), 
    Concentration = c("", "1x", "5 mM", "0.5 μM", "0.2 μM", "0.2 μM", "0.2 μM", "0.2 μM", "1.3 U/μL"), 
    Volume = c(8.49, 5, 5, 1.25, 0.5, 0.5, 0.5, 0.5, 0.26)
  ) %>% 
  mutate(Volume_x8 = Volume*8)

colnames(multiplex_1_round_1)[colnames(multiplex_1_round_1)=="Concentration"] <- "Final Concentration"
colnames(multiplex_1_round_1)[colnames(multiplex_1_round_1)=="Volume"] <- "Volume x1 (μL)"
colnames(multiplex_1_round_1)[colnames(multiplex_1_round_1)=="Volume_x8"] <- "Volume x8 (μL)"
multiplex_1_round_1  %>% 
  knitr::kable(align = "lcccc") %>% 
  kable_styling(bootstrap_options = c("condensed", "responsive"), full_width = T)  
```

Cycling conditions:

-   95°C/2 min
-   40 cycles of:
    -   95°C/20 sec,
    -   58°C/2 min,
    -   72/2 min;
-   72°C/10 min
-   Resting at 4°C.

### Round 2.

::: panel-tabset
#### K13-1

The pfk13 fragment 1 (K13-1) simplex PCR was in a total Volume of 25 μL with 1x Buffer, 500 μM dNTPs mix, 1.5 mM MgCl2, 300 nM of each primer (F/R), 1 unit of GoTaq Flexi Polymerase and 2 μL of PCR product from the first round.

```{r}
#| echo: false
multiplex_1_round_2_k131 <- data.frame(
    Reagent = c("Water", "5X Buffer", "MgCl2 (25mM)", "dNTP", "K13-1-in-F", "K13-1-in-R", "G2 Flexi Taq (5U/μL)"), 
    Concentration = c("", "1x", "2 mM", "0.5 μM", "0.3 μM", "0.3 μM", "1 U/μL"), 
    Volume = c(13.05, 5, 2, 1.25, 0.75, 0.75, 0.2)
  ) %>% 
  mutate(Volume_x8 = Volume*8)

colnames(multiplex_1_round_2_k131)[colnames(multiplex_1_round_2_k131)=="Concentration"] <- "Final Concentration"
colnames(multiplex_1_round_2_k131)[colnames(multiplex_1_round_2_k131)=="Volume"] <- "Volume x1 (μL)"
colnames(multiplex_1_round_2_k131)[colnames(multiplex_1_round_2_k131)=="Volume_x8"] <- "Volume x8 (μL)"
multiplex_1_round_2_k131  %>% 
  knitr::kable(align = "lcccc") %>% 
  kable_styling(bootstrap_options = c("condensed", "responsive"), full_width = T)  
```

#### K13-2

The pfk13 fragment 2 (K13-2) simplex PCR °Ccurred with the same concentrations except that only 200 nM per primer (F/R was used).

```{r}
#| echo: false
multiplex_1_round_2_k132 <- data.frame(
    Reagent = c("Water", "5X Buffer", "MgCl2 (25mM)", "dNTP", "K13-2-in-F", "K13-2-in-R", "G2 Flexi Taq (5U/μL)"), 
    Concentration = c("", "1x", "2 mM", "0.5 μM", "0.2 μM", "0.2 μM", "1 U/μL"), 
    Volume = c(13.55, 5, 2, 1.25, 0.5, 0.5, 0.2)
  ) %>% 
  mutate(Volume_x8 = Volume*8)

colnames(multiplex_1_round_2_k132)[colnames(multiplex_1_round_2_k132)=="Concentration"] <- "Final Concentration"
colnames(multiplex_1_round_2_k132)[colnames(multiplex_1_round_2_k132)=="Volume"] <- "Volume x1 (μL)"
colnames(multiplex_1_round_2_k132)[colnames(multiplex_1_round_2_k132)=="Volume_x8"] <- "Volume x8 (μL)"
multiplex_1_round_2_k132  %>% 
  knitr::kable(align = "lcccc") %>% 
  kable_styling(bootstrap_options = c("condensed", "responsive"), full_width = T)  
```

#### mdr1-1

The simplex reaction for pfmdr1 fragment 1 (mdr1-1) had total Volume of 25 μL with 1x Buffer, 500 μM dNTPs mix, 1.5 mM MgCl2, 200 nM of each primer (F/R), 1.3 units of GoTaq Flexi Polymerase and 2 μL of PCR product from the first round.

```{r}
#| echo: false
multiplex_1_round_2_mdr11 <- data.frame(
    Reagent = c("Water", "5X Buffer", "MgCl2 (25mM)", "dNTP", "mdr1-1-in-F", "mdr1-1-in-R", "G2 Flexi Taq (5U/μL)"), 
    Concentration = c("", "1x", "1.5 mM", "0.5 μM", "0.2 μM", "0.2 μM", "1.3 U/μL"), 
    Volume = c(13.99, 5, 2, 1.25, 0.5, 0.5, 0.26)
  ) %>% 
  mutate(Volume_x8 = Volume*8)

colnames(multiplex_1_round_2_mdr11)[colnames(multiplex_1_round_2_mdr11)=="Concentration"] <- "Final Concentration"
colnames(multiplex_1_round_2_mdr11)[colnames(multiplex_1_round_2_mdr11)=="Volume"] <- "Volume x1 (μL)"
colnames(multiplex_1_round_2_mdr11)[colnames(multiplex_1_round_2_mdr11)=="Volume_x8"] <- "Volume x8 (μL)"
multiplex_1_round_2_mdr11  %>% 
  knitr::kable(align = "lcccc") %>% 
  kable_styling(bootstrap_options = c("condensed", "responsive"), full_width = T)  
```
:::

Cycling conditions for each simplex reaction:

-   95°C/2 min
-   40 cycles of:
    -   95°C/20 sec
    -   58°C/2 min
    -   72/2 min;
-   72°C/10 min
-   Resting at 4°C.

## Final Sequences

```{r}
#| echo: false

alarmcoding_PCR_multiplex_1 %>%
  distinct(Length, PCR_Round, Codons) %>% 
  mutate(Marker = case_when(Length==845 ~ "K13 Round 1 Product",
                            Length==412 ~ "K13-1",
                            Length==333 ~ "K13-2",
                            Length==579 ~ "mdr1-1 Round 1 Product",
                            Length==423 ~ "mdr1-1", .default = "CHECK")) %>%
  select(Marker, Length, PCR_Round, Codons) %>% 
  knitr::kable(caption = "Table 3. Details about sequences genotyped in the multiplex 1 of the <em>ALARM</em>coding protocol", align = "lcccc", escape = FALSE) %>% 
  kable_styling(bootstrap_options = c("condensed", "responsive"), full_width = T) %>% 
  group_rows("Kelch protein (<em>pfk13</em>: PF3D7_1343700)", 1, 3, escape = FALSE) %>% 
  group_rows("Multidrug resistance gene 1 (<em>pfmdr1</em>: Pf3D7_0523000)", 4, 5, escape = FALSE)
```
